// flowbite-vue/dist/flowbite-vue.mjs
import { reactive as _n, onBeforeMount as Cn, onBeforeUnmount as $n, defineComponent as $, openBlock as c, createElementBlock as g, unref as d, renderSlot as x, ref as I, computed as u, onMounted as Te, createCommentVNode as S, normalizeClass as w, createElementVNode as m, toRefs as N, inject as W, useSlots as Pe, useAttrs as Sn, createBlock as D, resolveDynamicComponent as ut, withCtx as H, toDisplayString as B, createTextVNode as ne, Fragment as U, Comment as Tn, toRef as He, provide as X, renderList as be, normalizeProps as Ar, guardReactiveProps as Ir, withDirectives as ye, vShow as Pn, watch as Ze, nextTick as Ot, isRef as we, getCurrentScope as zn, onScopeDispose as On, getCurrentInstance as Br, watchEffect as An, createVNode as Ce, Transition as In, normalizeStyle as Le, pushScopeId as Er, popScopeId as Lr, withModifiers as Qe, createStaticVNode as Mr, withKeys as Nr, h as se, resolveComponent as et, mergeProps as Vr, TransitionGroup as Bn, withScopeId as En, vModelDynamic as Ln, vModelCheckbox as Rr, vModelSelect as Mn, vModelText as Nn, vModelRadio as Vn } from "vue";
let Hr = (e = 21) => crypto.getRandomValues(new Uint8Array(e)).reduce((t, r) => (r &= 63, r < 36 ? t += r.toString(36) : r < 62 ? t += (r - 26).toString(36).toUpperCase() : r > 62 ? t += "-" : t += "_", t), "");
const vt = _n({});
function Ge(e, t) {
  return Cn(() => {
    var r, n;
    !e || (vt[e] = {
      id: e,
      flush: (r = t == null ? void 0 : t.flush) != null ? r : !1,
      alwaysOpen: (n = t == null ? void 0 : t.alwaysOpen) != null ? n : !1,
      panels: {}
    });
  }), $n(() => {
    !e || delete vt[e];
  }), {
    accordionsStates: vt
  };
}
const Rn = ["data-accordion-id"], p0 = /* @__PURE__ */ $({
  __name: "Accordion",
  props: {
    alwaysOpen: {
      type: Boolean,
      default: !1
    },
    flush: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = Hr();
    return Ge(r, { ...t }), (n, s) => (c(), g("div", {
      class: "vp-raw",
      "data-accordion-id": d(r)
    }, [
      x(n.$slots, "default")
    ], 8, Rn));
  }
}), Hn = ["data-panel-id"], h0 = /* @__PURE__ */ $({
  __name: "AccordionPanel",
  setup(e) {
    const { accordionsStates: t } = Ge(), r = Hr(), n = I(), s = u(() => n.value ? n.value.parentElement.dataset.accordionId : null), o = u(() => t[s.value]);
    return Te(() => {
      var l, i;
      const a = (i = Object.keys((l = o == null ? void 0 : o.value) == null ? void 0 : l.panels)) == null ? void 0 : i.length;
      o.value.panels[r] = {
        id: r,
        order: a,
        isVisible: !a
      };
    }), (a, l) => (c(), g("div", {
      "data-panel-id": d(r),
      ref_key: "panel",
      ref: n
    }, [
      d(s) ? x(a.$slots, "default", { key: 0 }) : S("", !0)
    ], 8, Hn));
  }
});
var Dr = { exports: {} };
/*!
	Copyright (c) 2018 Jed Watson.
	Licensed under the MIT License (MIT), see
	http://jedwatson.github.io/classnames
*/
(function(e) {
  (function() {
    var t = {}.hasOwnProperty;
    function r() {
      for (var n = [], s = 0; s < arguments.length; s++) {
        var o = arguments[s];
        if (!!o) {
          var a = typeof o;
          if (a === "string" || a === "number")
            n.push(o);
          else if (Array.isArray(o)) {
            if (o.length) {
              var l = r.apply(null, o);
              l && n.push(l);
            }
          } else if (a === "object") {
            if (o.toString !== Object.prototype.toString && !o.toString.toString().includes("[native code]")) {
              n.push(o.toString());
              continue;
            }
            for (var i in o)
              t.call(o, i) && o[i] && n.push(i);
          }
        }
      }
      return n.join(" ");
    }
    e.exports ? (r.default = r, e.exports = r) : window.classNames = r;
  })();
})(Dr);
const P = Dr.exports, Dn = "flex items-center p-5 w-full font-medium text-left text-gray-500 border border-gray-200 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-800 dark:border-gray-700 dark:text-gray-400 hover:bg-gray-100 dark:hover:bg-gray-800", Fn = "w-6 h-6 shrink-0";
function jn(e) {
  const t = u(() => e.value.parentElement.parentElement.dataset.accordionId), r = u(() => e.value.parentElement.dataset.panelId), { accordionsStates: n } = Ge(), s = u(() => n[t.value]), o = u(() => s.value.panels[r.value]), a = u(() => Object.keys(o.value).length), l = u(() => o.value.order !== a.value - 1), i = u(() => l.value || s.value.flush && o.value.order === a.value - 1 && !o.value.isVisible), f = u(() => P(Dn, {
    "bg-gray-100 dark:bg-gray-800": o.value.isVisible,
    "rounded-t-xl": o.value.order === 0 && !s.value.flush,
    "border-t-0": o.value.order === 0 && s.value.flush,
    "border-b-0": i.value,
    "border-x-0": s.value.flush
  })), p = u(() => P(Fn, { "rotate-180": o.value.isVisible }));
  return {
    headerClasses: f,
    arrowClasses: p
  };
}
const Wn = { class: "w-full" }, Gn = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
}, null, -1), qn = [
  Gn
], m0 = /* @__PURE__ */ $({
  __name: "AccordionHeader",
  setup(e) {
    const t = I(!1), r = I(), n = u(() => r.value.parentElement.parentElement.dataset.accordionId), s = u(() => r.value.parentElement.dataset.panelId), { accordionsStates: o } = Ge(), a = u(() => o[n.value]), l = u(() => a.value.panels[s.value]);
    let i, f;
    function p() {
      const h = l.value.isVisible;
      for (const b in a.value.panels) {
        const C = a.value.panels[b];
        C.id !== s.value ? C.isVisible = !1 : C.isVisible = !h;
      }
    }
    function v() {
      l.value.isVisible = !l.value.isVisible;
    }
    function y() {
      if (a.value.alwaysOpen)
        return v();
      p();
    }
    return Te(() => {
      const h = jn(r);
      i = h.headerClasses, f = h.arrowClasses, t.value = !0;
    }), (h, b) => (c(), g("div", {
      ref_key: "header",
      ref: r
    }, [
      t.value ? (c(), g("button", {
        key: 0,
        type: "button",
        onClick: y,
        class: w(d(i))
      }, [
        m("span", Wn, [
          x(h.$slots, "default")
        ]),
        (c(), g("svg", {
          "data-accordion-icon": "",
          class: w(d(f)),
          fill: "currentColor",
          viewBox: "0 0 20 20",
          xmlns: "http://www.w3.org/2000/svg"
        }, qn, 2))
      ], 2)) : S("", !0)
    ], 512));
  }
}), Un = "p-5 border border-gray-200 dark:border-gray-700 dark:bg-gray-900";
function Kn(e) {
  const t = u(() => e.value.parentElement.parentElement.dataset.accordionId), r = u(() => e.value.parentElement.dataset.panelId), { accordionsStates: n } = Ge(), s = u(() => n[t.value]), o = u(() => n[t.value].panels[r.value]), a = u(() => Object.keys(n[t.value].panels[r.value]).length);
  return {
    contentClasses: u(() => P(Un, {
      hidden: !o.value.isVisible,
      "border-b-0": o.value.order !== a.value - 1 || s.value.flush,
      "border-t-0": o.value.order === a.value - 1,
      "border-x-0": s.value.flush
    }))
  };
}
const b0 = /* @__PURE__ */ $({
  __name: "AccordionContent",
  setup(e) {
    const t = I(!1), r = I();
    let n;
    return Te(() => {
      n = Kn(r).contentClasses, t.value = !0;
    }), (s, o) => (c(), g("div", {
      ref_key: "content",
      ref: r
    }, [
      t.value ? (c(), g("div", {
        key: 0,
        class: w(d(n))
      }, [
        x(s.$slots, "default")
      ], 2)) : S("", !0)
    ], 512));
  }
}), Yn = {
  0: "w-0 h-0",
  "0.5": "w-0.5 h-0.5",
  1: "w-1 h-1",
  "1.5": "w-1.5 h-1.5",
  10: "w-10 h-10",
  11: "w-11 h-11",
  12: "w-12 h-12",
  2: "w-2 h-2",
  "2.5": "w-2.5 h-2.5",
  3: "w-3 h-3",
  4: "w-4 h-4",
  5: "w-5 h-5",
  6: "w-6 h-6",
  7: "w-7 h-7",
  8: "w-8 h-8",
  9: "w-9 h-9",
  px: "w-px h-px"
}, Jn = {
  blue: "fill-blue-600",
  gray: "fill-gray-600 dark:fill-gray-300",
  green: "fill-green-500",
  pink: "fill-pink-600",
  purple: "fill-purple-600",
  red: "fill-red-600",
  yellow: "fill-yellow-400",
  white: "fill-white"
};
function Qn(e) {
  const t = u(() => Yn[e.size.value]), r = u(() => Jn[e.color.value]), n = u(() => "text-gray-200 dark:text-gray-600"), s = u(() => "animate-spin");
  return {
    spinnerClasses: u(() => P(
      t.value,
      n.value,
      r.value,
      s.value
    ))
  };
}
const Xn = /* @__PURE__ */ m("path", {
  d: "M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z",
  fill: "currentColor"
}, null, -1), Zn = /* @__PURE__ */ m("path", {
  d: "M93.9676 39.0409C96.393 38.4038 97.8624 35.9116 97.0079 33.5539C95.2932 28.8227 92.871 24.3692 89.8167 20.348C85.8452 15.1192 80.8826 10.7238 75.2124 7.41289C69.5422 4.10194 63.2754 1.94025 56.7698 1.05124C51.7666 0.367541 46.6976 0.446843 41.7345 1.27873C39.2613 1.69328 37.813 4.19778 38.4501 6.62326C39.0873 9.04874 41.5694 10.4717 44.0505 10.1071C47.8511 9.54855 51.7191 9.52689 55.5402 10.0491C60.8642 10.7766 65.9928 12.5457 70.6331 15.2552C75.2735 17.9648 79.3347 21.5619 82.5849 25.841C84.9175 28.9121 86.7997 32.2913 88.1811 35.8758C89.083 38.2158 91.5421 39.6781 93.9676 39.0409Z",
  fill: "currentFill"
}, null, -1), es = [
  Xn,
  Zn
], qe = /* @__PURE__ */ $({
  __name: "Spinner",
  props: {
    size: {
      type: String,
      default: "4"
    },
    color: {
      type: String,
      default: "blue"
    }
  },
  setup(e) {
    const t = e, { spinnerClasses: r } = Qn(N(t));
    return (n, s) => (c(), g("svg", {
      role: "status",
      class: w(d(r)),
      viewBox: "0 0 100 101",
      fill: "none",
      xmlns: "http://www.w3.org/2000/svg"
    }, es, 2));
  }
}), Fr = "flowbite-themable-injection-key", _e = {
  blue: {
    background: "bg-blue-700 dark:bg-blue-600",
    disabled: "",
    hover: "hover:bg-blue-800 dark:hover:bg-blue-700",
    text: "text-blue-600 dark:text-blue-500",
    border: "border-blue-600 dark:border-blue-500",
    focus: "focus:ring-blue-300 dark:focus:ring-blue-800"
  },
  green: {
    background: "bg-green-700 dark:bg-green-600",
    disabled: "",
    hover: "hover:bg-green-800 dark:hover:bg-green-700",
    text: "text-green-600 dark:text-green-500",
    border: "border-green-600 dark:border-green-500",
    focus: "focus:ring-green-300 dark:focus:ring-green-800"
  },
  pink: {
    background: "bg-pink-700 dark:bg-pink-600",
    disabled: "",
    hover: "hover:bg-pink-800 dark:hover:bg-pink-700",
    text: "text-pink-600 dark:text-pink-500",
    border: "border-pink-600 dark:border-pink-500",
    focus: "focus:ring-pink-300 dark:focus:ring-pink-900"
  },
  purple: {
    background: "bg-purple-700 dark:bg-purple-600",
    disabled: "",
    hover: "hover:bg-purple-800 dark:hover:bg-purple-700",
    text: "text-purple-600 dark:text-purple-500",
    border: "border-purple-600 dark:border-purple-500",
    focus: "focus:ring-purple-300 dark:focus:ring-purple-900"
  },
  red: {
    background: "bg-red-700 dark:bg-red-600",
    disabled: "",
    hover: "hover:bg-red-800 dark:hover:bg-red-700",
    text: "text-red-600 dark:text-red-500",
    border: "border-red-600 dark:border-red-500",
    focus: "focus:ring-red-300 dark:focus:ring-red-900"
  }
};
function jr(e) {
  const t = W(Fr, I(null)), r = u(() => !!(t != null && t.value)), n = u(() => (t == null ? void 0 : t.value) || void 0), s = u(() => e || t.value), o = u(() => s.value ? _e[s.value].background : ""), a = u(() => s.value ? _e[s.value].disabled : ""), l = u(() => s.value ? _e[s.value].hover : ""), i = u(() => s.value ? _e[s.value].text : ""), f = u(() => s.value ? _e[s.value].border : ""), p = u(() => s.value ? _e[s.value].focus : "");
  return {
    backgroundClasses: o,
    disabledClasses: a,
    hoverClasses: l,
    textClasses: i,
    borderClasses: f,
    focusClasses: p,
    isActive: r,
    color: n
  };
}
const Gt = {
  default: {
    default: "text-white bg-blue-700 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg dark:bg-blue-600 focus:outline-none dark:focus:ring-blue-800",
    blue: "text-white bg-blue-700 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg dark:bg-blue-600 focus:outline-none dark:focus:ring-blue-800",
    alternative: "font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600",
    dark: "text-white bg-gray-800 focus:outline-none focus:ring-4 focus:ring-gray-300 font-medium rounded-lg dark:bg-gray-800 dark:focus:ring-gray-700 dark:border-gray-700",
    light: "text-gray-900 bg-white border border-gray-300 focus:outline-none focus:ring-4 focus:ring-gray-200 font-medium rounded-lg dark:bg-gray-800 dark:text-white dark:border-gray-600 dark:focus:ring-gray-700",
    green: "focus:outline-none text-white bg-green-700 focus:ring-4 focus:ring-green-300 font-medium rounded-lg dark:bg-green-600 dark:focus:ring-green-800",
    red: "focus:outline-none text-white bg-red-700 focus:ring-4 focus:ring-red-300 font-medium rounded-lg dark:bg-red-600 dark:focus:ring-red-900",
    yellow: "focus:outline-none text-white bg-yellow-400 focus:ring-4 focus:ring-yellow-300 font-medium rounded-lg dark:focus:ring-yellow-900",
    purple: "focus:outline-none text-white bg-purple-700 focus:ring-4 focus:ring-purple-300 font-medium rounded-lg dark:bg-purple-600 dark:focus:ring-purple-900",
    pink: "focus:outline-none text-white bg-pink-700 focus:ring-4 focus:ring-pink-300 font-medium rounded-lg dark:bg-pink-600 dark:focus:ring-pink-900"
  },
  hover: {
    default: "hover:bg-blue-800 dark:hover:bg-blue-700",
    blue: "hover:bg-blue-800 dark:hover:bg-blue-700",
    alternative: "hover:bg-gray-100 hover:text-blue-700 dark:hover:text-white dark:hover:bg-gray-700",
    dark: "hover:bg-gray-900 dark:hover:bg-gray-700",
    light: "hover:bg-gray-100 dark:hover:border-gray-600",
    green: "hover:bg-green-800 dark:hover:bg-green-700",
    red: "hover:bg-red-800 dark:hover:bg-red-700",
    yellow: "hover:bg-yellow-500",
    purple: "hover:bg-purple-800 dark:hover:bg-purple-700",
    pink: "hover:bg-pink-800 dark:hover:bg-pink-700"
  }
}, qt = {
  default: {
    dark: "text-gray-900 border border-gray-800 focus:ring-4 focus:outline-none focus:ring-gray-300 font-medium rounded-lg text-sm text-center dark:border-gray-600 dark:text-gray-400 dark:focus:ring-gray-800",
    default: "text-blue-700 border border-blue-700 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm text-center dark:border-blue-500 dark:text-blue-500 dark:focus:ring-blue-800",
    blue: "text-blue-700 border border-blue-700 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm text-center dark:border-blue-500 dark:text-blue-500 dark:focus:ring-blue-800",
    green: "text-green-700 border border-green-700 focus:ring-4 focus:outline-none focus:ring-green-300 font-medium rounded-lg text-sm text-center dark:border-green-500 dark:text-green-500 dark:focus:ring-green-800",
    purple: "text-purple-700 border border-purple-700 focus:ring-4 focus:outline-none focus:ring-purple-300 font-medium rounded-lg text-sm text-center dark:border-purple-400 dark:text-purple-400 dark:focus:ring-purple-900",
    pink: "text-pink-700 border border-pink-700 focus:ring-4 focus:outline-none focus:ring-pink-300 font-medium rounded-lg text-sm text-center dark:border-pink-400 dark:text-pink-400 dark:focus:ring-pink-900",
    red: "text-red-700 border border-red-700 focus:ring-4 focus:outline-none focus:ring-red-300 font-medium rounded-lg text-sm text-center dark:border-red-500 dark:text-red-500 dark:focus:ring-red-900",
    yellow: "text-yellow-400 border border-yellow-400 focus:ring-4 focus:outline-none focus:ring-yellow-300 font-medium rounded-lg text-sm text-center dark:border-yellow-300 dark:text-yellow-300 dark:focus:ring-yellow-900"
  },
  hover: {
    dark: "hover:text-white hover:bg-gray-900 dark:hover:text-white dark:hover:bg-gray-600",
    default: "hover:text-white hover:bg-blue-800 dark:hover:text-white dark:hover:bg-blue-600",
    blue: "hover:text-white hover:bg-blue-800 dark:hover:text-white dark:hover:bg-blue-600",
    green: "hover:text-white hover:bg-green-800 dark:hover:text-white dark:hover:bg-green-600",
    purple: "hover:text-white hover:bg-purple-800 dark:hover:text-white dark:hover:bg-purple-500",
    pink: "hover:text-white hover:bg-pink-800 dark:hover:text-white dark:hover:bg-pink-500",
    red: "hover:text-white hover:bg-red-800 dark:hover:text-white dark:hover:bg-red-600",
    yellow: "hover:text-white hover:bg-yellow-500 dark:hover:text-white dark:hover:bg-yellow-400"
  }
}, Ut = {
  hover: {
    "cyan-blue": "hover:bg-gradient-to-bl",
    "green-blue": "hover:bg-gradient-to-bl",
    "pink-orange": "hover:bg-gradient-to-bl",
    "purple-blue": "hover:bg-gradient-to-bl",
    "purple-pink": "hover:bg-gradient-to-l",
    "red-yellow": "hover:bg-gradient-to-bl",
    "teal-lime": "hover:bg-gradient-to-l hover:from-teal-200 hover:to-lime-200",
    blue: "hover:bg-gradient-to-br",
    cyan: "hover:bg-gradient-to-br",
    green: "hover:bg-gradient-to-br",
    lime: "hover:bg-gradient-to-br",
    pink: "hover:bg-gradient-to-br",
    purple: "hover:bg-gradient-to-br",
    red: "hover:bg-gradient-to-br",
    teal: "hover:bg-gradient-to-br"
  },
  default: {
    "cyan-blue": "text-white bg-gradient-to-r from-cyan-500 to-blue-500 focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 font-medium rounded-lg",
    "green-blue": "text-white bg-gradient-to-br from-green-400 to-blue-600 focus:ring-4 focus:outline-none focus:ring-green-200 dark:focus:ring-green-800 font-medium rounded-lg",
    "pink-orange": "text-white bg-gradient-to-br from-pink-500 to-orange-400 focus:ring-4 focus:outline-none focus:ring-pink-200 dark:focus:ring-pink-800 font-medium rounded-lg",
    "purple-blue": "text-white bg-gradient-to-br from-purple-600 to-blue-500 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:focus:ring-blue-800 font-medium rounded-lg",
    "purple-pink": "text-white bg-gradient-to-r from-purple-500 to-pink-500 focus:ring-4 focus:outline-none focus:ring-purple-200 dark:focus:ring-purple-800 font-medium rounded-lg",
    "red-yellow": "text-gray-900 bg-gradient-to-r from-red-200 via-red-300 to-yellow-200 focus:ring-4 focus:outline-none focus:ring-red-100 dark:focus:ring-red-400 font-medium rounded-lg",
    "teal-lime": "text-gray-900 bg-gradient-to-r from-teal-200 to-lime-200 focus:ring-4 focus:outline-none focus:ring-lime-200 dark:focus:ring-teal-700 font-medium rounded-lg",
    blue: "text-white bg-gradient-to-r from-blue-500 via-blue-600 to-blue-700 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:focus:ring-blue-800 rounded-lg",
    cyan: "text-white bg-gradient-to-r from-cyan-500 via-cyan-600 to-cyan-700 focus:ring-4 focus:outline-none focus:ring-cyan-300 dark:focus:ring-cyan-800 rounded-lg",
    green: "text-white bg-gradient-to-r from-green-500 via-green-600 to-green-700 focus:ring-4 focus:outline-none focus:ring-green-300 dark:focus:ring-green-800 rounded-lg",
    lime: "text-gray-900 bg-gradient-to-r from-lime-500 via-lime-600 to-lime-700 focus:ring-4 focus:outline-none focus:ring-lime-300 dark:focus:ring-lime-800 rounded-lg",
    pink: "text-white bg-gradient-to-r from-pink-500 via-pink-600 to-pink-700 focus:ring-4 focus:outline-none focus:ring-pink-300 dark:focus:ring-pink-800 rounded-lg",
    purple: "text-white bg-gradient-to-r from-purple-500 via-purple-600 to-purple-700 focus:ring-4 focus:outline-none focus:ring-purple-300 dark:focus:ring-purple-800 rounded-lg",
    red: "text-white bg-gradient-to-r from-red-500 via-red-600 to-red-700 focus:ring-4 focus:outline-none focus:ring-red-300 dark:focus:ring-red-800 rounded-lg",
    teal: "text-white bg-gradient-to-r from-teal-500 via-teal-600 to-teal-700 focus:ring-4 focus:outline-none focus:ring-teal-300 dark:focus:ring-teal-800 rounded-lg"
  }
}, Kt = {
  default: {
    "cyan-blue": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-cyan-500 to-blue-500 dark:text-white focus:ring-4 focus:outline-none focus:ring-cyan-200 dark:focus:ring-cyan-800",
    "green-blue": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-green-400 to-blue-600 dark:text-white focus:ring-4 focus:outline-none focus:ring-green-200 dark:focus:ring-green-800",
    "pink-orange": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-pink-500 to-orange-400 dark:text-white focus:ring-4 focus:outline-none focus:ring-pink-200 dark:focus:ring-pink-800",
    "purple-blue": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-purple-600 to-blue-500 dark:text-white focus:ring-4 focus:outline-none focus:ring-blue-300 dark:focus:ring-blue-800",
    "purple-pink": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-purple-500 to-pink-500 dark:text-white focus:ring-4 focus:outline-none focus:ring-purple-200 dark:focus:ring-purple-800",
    "red-yellow": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-red-200 via-red-300 to-yellow-200 dark:text-white focus:ring-4 focus:outline-none focus:ring-red-100 dark:focus:ring-red-400",
    "teal-lime": "relative inline-flex items-center justify-center overflow-hidden font-medium text-gray-900 rounded-lg group bg-gradient-to-br from-teal-300 to-lime-300 dark:text-white focus:ring-4 focus:outline-none focus:ring-lime-200 dark:focus:ring-lime-800"
  },
  hover: {
    "cyan-blue": "group-hover:from-cyan-500 group-hover:to-blue-500 hover:text-white",
    "green-blue": "group-hover:from-green-400 group-hover:to-blue-600 hover:text-white",
    "pink-orange": "group-hover:from-pink-500 group-hover:to-orange-400 hover:text-white",
    "purple-blue": "group-hover:from-purple-600 group-hover:to-blue-500 hover:text-white",
    "purple-pink": "group-hover:from-purple-500 group-hover:to-pink-500 hover:text-white",
    "red-yellow": "group-hover:from-red-200 group-hover:via-red-300 group-hover:to-yellow-200 dark:hover:text-gray-900",
    "teal-lime": "group-hover:from-teal-300 group-hover:to-lime-300 dark:hover:text-gray-900"
  }
}, ts = {
  xs: "text-xs px-2 py-1",
  sm: "text-sm px-3 py-1.5",
  md: "text-sm px-4 py-2",
  lg: "text-base px-5 py-2.5",
  xl: "text-base px-6 py-3"
}, rs = {
  xs: "text-xs p-1",
  sm: "text-sm p-1.5",
  md: "text-sm p-2",
  lg: "text-base p-2.5",
  xl: "text-base p-3"
}, Yt = {
  blue: "shadow-lg shadow-blue-500/50 dark:shadow-lg dark:shadow-blue-800/80",
  cyan: "shadow-lg shadow-cyan-500/50 dark:shadow-lg dark:shadow-cyan-800/80",
  green: "shadow-lg shadow-green-500/50 dark:shadow-lg dark:shadow-green-800/80",
  lime: "shadow-lg shadow-lime-500/50 dark:shadow-lg dark:shadow-lime-800/80",
  pink: "shadow-lg shadow-pink-500/50 dark:shadow-lg dark:shadow-pink-800/80",
  purple: "shadow-lg shadow-purple-500/50 dark:shadow-lg dark:shadow-purple-800/80",
  red: "shadow-lg shadow-red-500/50 dark:shadow-lg dark:shadow-red-800/80",
  teal: "shadow-lg shadow-teal-500/50 dark:shadow-lg dark:shadow-teal-800/80"
}, yt = ["blue", "green", "cyan", "teal", "lime", "red", "pink", "purple"], ns = ["alternative", "light"];
function ss(e) {
  const t = Pe(), r = u(() => e.square.value ? rs[e.size.value] : ts[e.size.value]), n = u(() => {
    const o = !!e.gradient.value, a = !!e.color.value, l = e.outline.value;
    let i = "", f = "";
    if (o && l)
      yt.includes(e.gradient.value) ? console.warn(`cannot use outline prop with "${e.gradient.value}" gradient`) : (f = Kt.default[e.gradient.value], e.disabled.value || (i = Kt.hover[e.gradient.value]));
    else if (o)
      f = Ut.default[e.gradient.value], e.disabled.value || (i = Ut.hover[e.gradient.value]);
    else if (a && l)
      if (ns.includes(e.color.value))
        console.warn(`cannot use outline prop with "${e.color.value}" color`);
      else {
        const v = e.color.value;
        f = qt.default[v], e.disabled.value || (i = qt.hover[v]);
      }
    else {
      const v = e.color.value;
      f = Gt.default[v], e.disabled.value || (i = Gt.hover[v]);
    }
    let p = "";
    return e.shadow.value === "" ? e.gradient.value && yt.includes(e.gradient.value) && (p = Yt[e.gradient.value]) : typeof e.shadow.value == "string" && yt.includes(e.shadow.value) && (p = Yt[e.shadow.value]), P(
      f,
      i,
      p,
      e.pill.value ? "!rounded-full" : "",
      e.disabled.value ? "cursor-not-allowed opacity-50" : "",
      o && l ? "p-0.5" : r.value,
      t.prefix || t.suffix || e.loading.value ? "inline-flex items-center" : ""
    );
  }), s = u(() => !!e.gradient.value && e.outline.value ? P(
    "relative bg-white dark:bg-gray-900 rounded-md inline-flex items-center",
    r.value,
    e.disabled.value ? "" : "group-hover:bg-opacity-0 transition-all ease-in duration-75"
  ) : "");
  return {
    wrapperClasses: n,
    spanClasses: s
  };
}
function os(e) {
  const t = {
    lg: "5",
    md: "4",
    sm: "3",
    xl: "6",
    xs: "2.5"
  }, r = u(() => t[e.size.value]), n = u(() => e.outline.value ? e.gradient.value ? e.gradient.value.includes("purple") ? "purple" : e.gradient.value.includes("blue") ? "blue" : e.gradient.value.includes("pink") ? "pink" : e.gradient.value.includes("red") ? "red" : "white" : ["alternative", "dark", "light"].includes(e.color.value) ? "white" : e.color.value === "default" ? "blue" : e.color.value : "white");
  return {
    size: r,
    color: n
  };
}
function as(e) {
  var f;
  const { textClasses: t, borderClasses: r, backgroundClasses: n, hoverClasses: s, disabledClasses: o, focusClasses: a, isActive: l } = jr((f = e.theme) == null ? void 0 : f.value);
  return {
    classes: u(() => {
      if (!l.value)
        return "";
      const p = [];
      return e.apply.value.includes("text") && p.push(t.value), e.apply.value.includes("border") && p.push(r.value), e.apply.value.includes("background") && p.push(n.value), e.apply.value.includes("hover") && p.push(s.value), e.apply.value.includes("disabled") && p.push(o.value), e.apply.value.includes("focus") && p.push(a.value), p.join(" ");
    })
  };
}
const ls = {
  border: (e) => e.substring(0, e.lastIndexOf("-"))
}, is = (e, t = ls) => {
  const n = Object.keys(t).find((s) => e.includes(s));
  return n ? t[n](e) : e.substring(0, e.indexOf("-"));
};
function j(...e) {
  return e.reduce((t, r) => {
    const n = Array.isArray(r) ? Array.from(r).map((p) => p.split(" ")).flat() : r.split(" "), s = n.map((p) => is(p)), o = s.filter((p) => !t.types.includes(p)), l = [...s.filter((p) => t.types.includes(p)), ...o], i = [.../* @__PURE__ */ new Set([...t.types, ...l])], f = i.map((p) => {
      if (l.includes(p)) {
        const y = s.indexOf(p);
        if (y >= 0)
          return n[y] || "";
      }
      const v = t.types.indexOf(p);
      return v >= 0 && t.classes[v] || "";
    }).filter((p) => !!p);
    return {
      types: i,
      classes: f
    };
  }, { types: [], classes: [] }).classes.join(" ");
}
const Wr = /* @__PURE__ */ $({
  __name: "FlowbiteThemableChild",
  props: {
    apply: {
      type: Array,
      required: !0
    },
    tag: {
      type: String,
      default: "div"
    },
    theme: {
      type: String,
      default: void 0
    }
  },
  setup(e) {
    const t = e, r = Sn(), { classes: n } = as(N(t)), s = u(() => r.class || "");
    return (o, a) => (c(), D(ut(e.tag), {
      class: w(d(j)(d(s), d(n)))
    }, {
      default: H(() => [
        x(o.$slots, "default")
      ]),
      _: 3
    }, 8, ["class"]));
  }
}), us = {
  key: 0,
  class: "mr-2"
}, ds = {
  key: 0,
  class: "mr-2"
}, cs = {
  key: 1,
  class: "ml-2"
}, fs = {
  key: 1,
  class: "ml-2"
}, gs = /* @__PURE__ */ $({
  __name: "Button",
  props: {
    color: {
      type: String,
      default: "default"
    },
    gradient: {
      type: [String, null],
      default: null
    },
    size: {
      type: String,
      default: "md"
    },
    shadow: {
      type: [String, null],
      default: null
    },
    pill: {
      type: Boolean,
      default: !1
    },
    square: {
      type: Boolean,
      default: !1
    },
    outline: {
      type: Boolean,
      default: !1
    },
    loading: {
      type: Boolean,
      default: !1
    },
    loadingPosition: {
      type: String,
      default: "prefix"
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = u(() => t.outline && t.gradient), n = u(() => t.loading && t.loadingPosition === "prefix"), s = u(() => t.loading && t.loadingPosition === "suffix"), { wrapperClasses: o, spanClasses: a } = ss(N(t)), { color: l, size: i } = os(N(t)), f = u(() => ["alternative", "light"].includes(t.color) ? [] : ["background", "hover", "focus"]);
    return (p, v) => (c(), D(Wr, {
      tag: "button",
      apply: d(f),
      class: w(d(o)),
      disabled: e.disabled
    }, {
      default: H(() => [
        !d(r) && (p.$slots.prefix || d(n)) ? (c(), g("div", us, [
          d(n) ? (c(), D(qe, {
            key: 0,
            color: d(l),
            size: d(i)
          }, null, 8, ["color", "size"])) : x(p.$slots, "prefix", { key: 1 })
        ])) : S("", !0),
        m("span", {
          class: w(d(a))
        }, [
          d(r) && (p.$slots.prefix || d(n)) ? (c(), g("div", ds, [
            d(n) ? (c(), D(qe, {
              key: 0,
              color: d(l),
              size: d(i)
            }, null, 8, ["color", "size"])) : x(p.$slots, "prefix", { key: 1 })
          ])) : S("", !0),
          x(p.$slots, "default"),
          d(r) && (p.$slots.suffix || d(s)) ? (c(), g("div", cs, [
            d(s) ? (c(), D(qe, {
              key: 0,
              color: d(l),
              size: d(i)
            }, null, 8, ["color", "size"])) : x(p.$slots, "suffix", { key: 1 })
          ])) : S("", !0)
        ], 2),
        !d(r) && (p.$slots.suffix || d(s)) ? (c(), g("div", fs, [
          d(s) ? (c(), D(qe, {
            key: 0,
            color: d(l),
            size: d(i)
          }, null, 8, ["color", "size"])) : x(p.$slots, "suffix", { key: 1 })
        ])) : S("", !0)
      ]),
      _: 3
    }, 8, ["apply", "class", "disabled"]));
  }
});
const oe = (e, t) => {
  const r = e.__vccOpts || e;
  for (const [n, s] of t)
    r[n] = s;
  return r;
}, ps = {}, hs = {
  class: "btn-group inline-flex rounded-md shadow-sm",
  role: "group"
};
function ms(e, t) {
  return c(), g("div", hs, [
    x(e.$slots, "default")
  ]);
}
const v0 = /* @__PURE__ */ oe(ps, [["render", ms]]), bs = "p-4 text-sm", vs = {
  danger: "text-red-700 dark:text-red-800",
  dark: "text-gray-700 dark:text-gray-300",
  info: "text-blue-700 dark:text-blue-800",
  success: "text-green-700 dark:text-green-800",
  warning: "text-yellow-700 dark:text-yellow-800"
}, ys = {
  danger: "bg-red-100 dark:bg-red-200",
  dark: "bg-gray-100 dark:bg-gray-700",
  info: "bg-blue-100 dark:bg-blue-200",
  success: "bg-green-100 dark:bg-green-200",
  warning: "bg-yellow-100 dark:bg-yellow-200"
}, ws = {
  danger: "border-t-4 border-red-500",
  dark: "border-t-4 border-gray-500",
  info: "border-t-4 border-blue-500",
  success: "border-t-4 border-green-500",
  warning: "border-t-4 border-yellow-500"
}, xs = "ml-auto -mx-1.5 -my-1.5 rounded-lg focus:ring-2 p-1.5 inline-flex h-8 w-8", ks = {
  danger: "hover:bg-red-200 dark:hover:bg-red-300 focus:ring-red-400 bg-red-100 dark:bg-red-200 text-red-500",
  dark: "dark:bg-gray-700 dark:text-gray-300 dark:hover:bg-gray-800 dark:hover:text-white hover:bg-gray-200 focus:ring-gray-400 bg-gray-100 text-gray-500",
  info: "hover:bg-blue-200 dark:hover:bg-blue-300 focus:ring-blue-400 bg-blue-100 dark:bg-blue-200 text-blue-500",
  success: "bg-green-100 dark:bg-green-200 text-green-500 focus:ring-green-400 hover:bg-green-200 dark:hover:bg-green-300",
  warning: "focus:ring-yellow-400 hover:bg-yellow-200 dark:hover:bg-yellow-300 bg-yellow-100 dark:bg-yellow-200 text-yellow-500"
};
function _s(e) {
  const t = u(() => P(
    bs,
    ys[e.type.value],
    r.value,
    e.border.value ? ws[e.type.value] : "rounded-lg",
    e.inline.value ? "flex" : ""
  )), r = u(() => P(
    vs[e.type.value]
  )), n = u(() => P(
    xs,
    ks[e.type.value]
  )), s = u(() => e.inline.value ? !e.icon.value && !e.title.value ? "" : P(e.title.value ? "ml-1" : "ml-3") : P("mt-2 mb-4")), o = u(() => !e.icon.value || !e.inline.value ? P(
    "font-medium",
    e.inline.value ? "" : "text-lg ml-2"
  ) : P(
    "font-medium ml-3",
    e.inline.value ? "" : "text-lg"
  ));
  return {
    alertClasses: t,
    textClasses: r,
    closeClasses: n,
    contentClasses: s,
    titleClasses: o
  };
}
const Cs = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7-4a1 1 0 11-2 0 1 1 0 012 0zM9 9a1 1 0 000 2v3a1 1 0 001 1h1a1 1 0 100-2v-3a1 1 0 00-1-1H9z",
  "clip-rule": "evenodd"
}, null, -1), $s = [
  Cs
], Ss = /* @__PURE__ */ m("span", { class: "sr-only" }, "Dismiss", -1), Ts = /* @__PURE__ */ m("svg", {
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ m("path", {
    "fill-rule": "evenodd",
    d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), Ps = [
  Ss,
  Ts
], zs = {
  key: 0,
  class: "inline-flex items-center"
}, Os = /* @__PURE__ */ m("span", { class: "sr-only" }, "Dismiss", -1), As = /* @__PURE__ */ m("svg", {
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ m("path", {
    "fill-rule": "evenodd",
    d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), Is = [
  Os,
  As
], y0 = /* @__PURE__ */ $({
  __name: "Alert",
  props: {
    type: {
      type: String,
      default: "info"
    },
    title: {
      type: String,
      default: ""
    },
    closable: {
      type: Boolean,
      default: !1
    },
    icon: {
      type: Boolean,
      default: !0
    },
    border: {
      type: Boolean,
      default: !1
    },
    inline: {
      type: Boolean,
      default: !0
    }
  },
  setup(e) {
    const t = e, { alertClasses: r, textClasses: n, closeClasses: s, contentClasses: o, titleClasses: a } = _s(N(t)), l = I(!0), i = () => {
      l.value = !1;
    };
    return (f, p) => l.value ? (c(), g("div", {
      key: 0,
      class: w(d(r)),
      role: "alert"
    }, [
      m("div", {
        class: w(e.inline ? "flex" : "flex items-center")
      }, [
        e.icon ? (c(), g("svg", {
          key: 0,
          class: w(["flex-shrink-0", "w-5", "h-5", d(n)]),
          fill: "currentColor",
          viewBox: "0 0 20 20",
          xmlns: "http://www.w3.org/2000/svg"
        }, $s, 2)) : S("", !0),
        e.title ? (c(), g("span", {
          key: 1,
          class: w(d(a))
        }, B(e.title), 3)) : S("", !0),
        !e.inline && e.closable ? (c(), g("button", {
          key: 2,
          type: "button",
          class: w(d(s)),
          "aria-label": "Close",
          onClick: i
        }, Ps, 2)) : S("", !0)
      ], 2),
      m("div", {
        class: w(d(o))
      }, [
        x(f.$slots, "default")
      ], 2),
      f.$slots.actions ? (c(), g("div", zs, [
        x(f.$slots, "actions")
      ])) : S("", !0),
      e.inline && e.closable ? (c(), g("button", {
        key: 1,
        type: "button",
        class: w(d(s)),
        "aria-label": "Close",
        onClick: i
      }, Is, 2)) : S("", !0)
    ], 2)) : S("", !0);
  }
}), Gr = "flowbite-tab-style-injection", qr = "flowbite-tab-active-name-injection", Ur = "flowbite-tab-visibility-directive-injection", Kr = "flowbite-tab-activate-func-injection";
function Bs(e) {
  const t = u(() => e.variant === "default" ? "flex flex-wrap text-sm font-medium text-center text-gray-500 border-b border-gray-200 dark:border-gray-700 dark:text-gray-400" : e.variant === "pills" ? "flex flex-wrap text-sm font-medium text-center text-gray-500 dark:text-gray-400" : e.variant === "underline" ? "flex flex-wrap -mb-px text-sm font-medium text-center text-gray-500 dark:text-gray-400" : ""), r = u(() => e.variant === "underline" ? "text-sm font-medium text-center text-gray-500 border-b border-gray-200 dark:text-gray-400 dark:border-gray-700" : "");
  return {
    ulClasses: t,
    divClasses: r
  };
}
function tt(e, t = !0, r = []) {
  return e.forEach((n) => {
    if (n !== null) {
      if (typeof n != "object") {
        (typeof n == "string" || typeof n == "number") && r.push(ne(String(n)));
        return;
      }
      if (Array.isArray(n)) {
        tt(n, t, r);
        return;
      }
      if (n.type === U) {
        if (n.children === null)
          return;
        Array.isArray(n.children) && tt(n.children, t, r);
      } else
        n.type !== Tn && r.push(n);
    }
  }), r;
}
const Es = {
  default: "cursor-pointer inline-block p-4 rounded-t-lg hover:text-gray-600 hover:bg-gray-50 dark:hover:bg-gray-800 dark:hover:text-gray-300",
  active: "cursor-pointer inline-block p-4 text-blue-600 bg-gray-100 rounded-t-lg active dark:bg-gray-800 dark:text-blue-500",
  disabled: "inline-block p-4 text-gray-400 rounded-t-lg cursor-not-allowed dark:text-gray-500"
}, Ls = {
  default: "cursor-pointer inline-block p-4 rounded-t-lg border-b-2 border-transparent hover:text-gray-600 hover:border-gray-300 dark:hover:text-gray-300",
  active: "cursor-pointer inline-block p-4 text-blue-600 rounded-t-lg border-b-2 border-blue-600 active dark:text-blue-500 dark:border-blue-500",
  disabled: "inline-block p-4 text-gray-400 rounded-t-lg cursor-not-allowed dark:text-gray-500"
}, Ms = {
  default: "cursor-pointer inline-block py-3 px-4 rounded-lg hover:text-gray-900 hover:bg-gray-100 dark:hover:bg-gray-800 dark:hover:text-white",
  active: "cursor-pointer inline-block py-3 px-4 text-white bg-blue-600 rounded-lg active",
  disabled: "inline-block py-3 px-4 text-gray-400 cursor-not-allowed dark:text-gray-500"
};
function Ns(e) {
  const t = jr();
  return {
    tabClasses: u(() => {
      const n = t.isActive.value, s = e.active.value ? "active" : e.disabled.value ? "disabled" : "default";
      return e.variant === "default" ? j(
        Es[s],
        n && s === "active" ? t.textClasses.value : ""
      ) : e.variant === "underline" ? j(
        Ls[s],
        n && s === "active" ? [t.borderClasses.value, t.textClasses.value] : ""
      ) : e.variant === "pills" ? j(
        Ms[s],
        n && s === "active" ? [t.backgroundClasses.value, "text-white"] : ""
      ) : "";
    })
  };
}
const Vs = /* @__PURE__ */ $({
  __name: "TabPane",
  props: {
    name: {
      type: String,
      required: !0
    },
    title: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: !1
    },
    active: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = W(Gr);
    r || console.warn("you can't use Tab outside of Tabs component. No tab style injection found");
    const n = W(Kr);
    n || console.warn("you can't use Tab outside of Tabs component. No tab activate injection found");
    const s = () => {
      if (!t.disabled) {
        if (!n)
          return console.warn("no onActivate");
        n(t.name);
      }
    }, { tabClasses: o } = Ns({
      active: He(t, "active"),
      disabled: He(t, "disabled"),
      variant: r
    });
    return (a, l) => (c(), g("li", null, [
      m("div", {
        class: w(d(o)),
        onClick: s
      }, B(e.title), 3)
    ]));
  }
}), Rs = {
  inheritAttrs: !1
}, w0 = /* @__PURE__ */ $({
  ...Rs,
  __name: "Tabs",
  props: {
    variant: {
      type: String,
      default: "default"
    },
    modelValue: {
      type: String,
      default: ""
    },
    directive: {
      type: String,
      default: "if"
    }
  },
  emits: ["update:modelValue", "click:pane"],
  setup(e, { emit: t }) {
    const r = e, { ulClasses: n, divClasses: s } = Bs(r);
    X(Gr, r.variant);
    const a = Pe().default, l = u(() => a ? tt(a()).filter((v) => v.type.__FLOWBITE_TAB__) : []), i = u({
      get: () => r.modelValue,
      set: (v) => t("update:modelValue", v)
    });
    X(qr, i), X(Ur, He(r, "directive"));
    const f = (v) => {
      i.value = v;
    };
    function p() {
      t("click:pane");
    }
    return X(Kr, f), (v, y) => (c(), g("div", null, [
      m("div", {
        class: w(d(s))
      }, [
        m("ul", {
          class: w(d(n))
        }, [
          (c(!0), g(U, null, be(d(l), (h, b) => {
            var C, _, k, T;
            return c(), D(Vs, {
              key: b,
              active: d(i) === ((C = h.props) == null ? void 0 : C.name),
              name: (_ = h.props) == null ? void 0 : _.name,
              disabled: (k = h.props) == null ? void 0 : k.disabled,
              title: (T = h.props) == null ? void 0 : T.title,
              onClick: p
            }, null, 8, ["active", "name", "disabled", "title"]);
          }), 128))
        ], 2)
      ], 2),
      m("div", Ar(Ir(v.$attrs)), [
        x(v.$slots, "default")
      ], 16)
    ]));
  }
}), Hs = { key: 0 }, Ds = { key: 1 }, Fs = {
  __FLOWBITE_TAB__: !0
}, x0 = /* @__PURE__ */ $({
  ...Fs,
  __name: "Tab",
  props: {
    name: {
      type: String,
      required: !0
    },
    title: {
      type: String,
      default: ""
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = W(qr, ""), r = W(Ur, "if");
    return (n, s) => (c(), g("div", null, [
      d(r) === "if" ? (c(), g(U, { key: 0 }, [
        d(t) === e.name ? (c(), g("div", Hs, [
          x(n.$slots, "default")
        ])) : S("", !0)
      ], 64)) : d(r) === "show" ? ye((c(), g("div", Ds, [
        x(n.$slots, "default")
      ], 512)), [
        [Pn, d(t) === e.name]
      ]) : S("", !0)
    ]));
  }
}), js = "absolute z-10 bg-white divide-y divide-gray-100 rounded shadow dark:bg-gray-700", Ue = 8, Ws = {
  bottom: "",
  left: "top-0",
  right: "top-0",
  top: ""
}, Gs = {
  bottom(e) {
    return `bottom: -${e.height + Ue}px;`;
  },
  left(e) {
    return `left: -${e.width + Ue}px;`;
  },
  right(e) {
    return `right: -${e.width + Ue}px;`;
  },
  top(e) {
    return `top: -${e.height + Ue}px;`;
  }
};
function qs(e) {
  Ze(e.visible, (s) => {
    s && Ot(() => r());
  });
  const t = I(""), r = () => {
    var o;
    const s = (o = e.contentRef.value) == null ? void 0 : o.getBoundingClientRect();
    if (!s)
      return t.value = "";
    t.value = Gs[e.placement.value](s);
  };
  return {
    contentClasses: u(() => P(
      js,
      Ws[e.placement.value]
    )),
    contentStyles: t
  };
}
var Jt;
const At = typeof window < "u", Us = (e) => typeof e < "u", Ks = (e) => typeof e == "function", Ys = (e) => typeof e == "string", wt = () => {
};
At && ((Jt = window == null ? void 0 : window.navigator) == null ? void 0 : Jt.userAgent) && /iP(ad|hone|od)/.test(window.navigator.userAgent);
function rt(e) {
  return typeof e == "function" ? e() : d(e);
}
function Js(e) {
  return e;
}
function Qs(e, t) {
  var r;
  if (typeof e == "number")
    return e + t;
  const n = ((r = e.match(/^-?[0-9]+\.?[0-9]*/)) == null ? void 0 : r[0]) || "", s = e.slice(n.length), o = parseFloat(n) + t;
  return Number.isNaN(o) ? e : o + s;
}
function It(e) {
  return zn() ? (On(e), !0) : !1;
}
function Xs(e) {
  return typeof e == "function" ? u(e) : I(e);
}
function Zs(e, t = !0) {
  Br() ? Te(e) : t ? e() : Ot(e);
}
function eo(e, t, r = {}) {
  const {
    immediate: n = !0
  } = r, s = I(!1);
  let o = null;
  function a() {
    o && (clearTimeout(o), o = null);
  }
  function l() {
    s.value = !1, a();
  }
  function i(...f) {
    a(), s.value = !0, o = setTimeout(() => {
      s.value = !1, o = null, e(...f);
    }, rt(t));
  }
  return n && (s.value = !0, At && i()), It(l), {
    isPending: s,
    start: i,
    stop: l
  };
}
function to(e = !1, t = {}) {
  const {
    truthyValue: r = !0,
    falsyValue: n = !1
  } = t, s = we(e), o = I(e);
  function a(l) {
    if (arguments.length)
      return o.value = l, o.value;
    {
      const i = rt(r);
      return o.value = o.value === i ? rt(n) : i, o.value;
    }
  }
  return s ? a : [o, a];
}
function Ee(e) {
  var t;
  const r = rt(e);
  return (t = r == null ? void 0 : r.$el) != null ? t : r;
}
const dt = At ? window : void 0;
function Ke(...e) {
  let t, r, n, s;
  if (Ys(e[0]) ? ([r, n, s] = e, t = dt) : [t, r, n, s] = e, !t)
    return wt;
  let o = wt;
  const a = Ze(() => Ee(t), (i) => {
    o(), i && (i.addEventListener(r, n, s), o = () => {
      i.removeEventListener(r, n, s), o = wt;
    });
  }, { immediate: !0, flush: "post" }), l = () => {
    a(), o();
  };
  return It(l), l;
}
function ro(e, t, r = {}) {
  const { window: n = dt, ignore: s, capture: o = !0, detectIframe: a = !1 } = r;
  if (!n)
    return;
  const l = I(!0);
  let i;
  const f = (h) => {
    n.clearTimeout(i);
    const b = Ee(e);
    !b || b === h.target || h.composedPath().includes(b) || !l.value || t(h);
  }, p = (h) => s && s.some((b) => {
    const C = Ee(b);
    return C && (h.target === C || h.composedPath().includes(C));
  }), v = [
    Ke(n, "click", f, { passive: !0, capture: o }),
    Ke(n, "pointerdown", (h) => {
      const b = Ee(e);
      l.value = !!b && !h.composedPath().includes(b) && !p(h);
    }, { passive: !0 }),
    Ke(n, "pointerup", (h) => {
      if (h.button === 0) {
        const b = h.composedPath();
        h.composedPath = () => b, i = n.setTimeout(() => f(h), 50);
      }
    }, { passive: !0 }),
    a && Ke(n, "blur", (h) => {
      var b;
      const C = Ee(e);
      ((b = document.activeElement) == null ? void 0 : b.tagName) === "IFRAME" && !(C != null && C.contains(document.activeElement)) && t(h);
    })
  ].filter(Boolean);
  return () => v.forEach((h) => h());
}
function no(e, t = !1) {
  const r = I(), n = () => r.value = Boolean(e());
  return n(), Zs(n, t), r;
}
function Ie(e, t = {}) {
  const { window: r = dt } = t, n = no(() => r && "matchMedia" in r && typeof r.matchMedia == "function");
  let s;
  const o = I(!1), a = () => {
    !s || ("removeEventListener" in s ? s.removeEventListener("change", l) : s.removeListener(l));
  }, l = () => {
    !n.value || (a(), s = r.matchMedia(Xs(e).value), o.value = s.matches, "addEventListener" in s ? s.addEventListener("change", l) : s.addListener(l));
  };
  return An(l), It(() => a()), o;
}
const Yr = {
  sm: 640,
  md: 768,
  lg: 1024,
  xl: 1280,
  "2xl": 1536
};
var so = Object.defineProperty, Qt = Object.getOwnPropertySymbols, oo = Object.prototype.hasOwnProperty, ao = Object.prototype.propertyIsEnumerable, Xt = (e, t, r) => t in e ? so(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, lo = (e, t) => {
  for (var r in t || (t = {}))
    oo.call(t, r) && Xt(e, r, t[r]);
  if (Qt)
    for (var r of Qt(t))
      ao.call(t, r) && Xt(e, r, t[r]);
  return e;
};
function Jr(e, t = {}) {
  function r(l, i) {
    let f = e[l];
    return i != null && (f = Qs(f, i)), typeof f == "number" && (f = `${f}px`), f;
  }
  const { window: n = dt } = t;
  function s(l) {
    return n ? n.matchMedia(l).matches : !1;
  }
  const o = (l) => Ie(`(min-width: ${r(l)})`, t), a = Object.keys(e).reduce((l, i) => (Object.defineProperty(l, i, {
    get: () => o(i),
    enumerable: !0,
    configurable: !0
  }), l), {});
  return lo({
    greater(l) {
      return Ie(`(min-width: ${r(l, 0.1)})`, t);
    },
    greaterOrEqual: o,
    smaller(l) {
      return Ie(`(max-width: ${r(l, -0.1)})`, t);
    },
    smallerOrEqual(l) {
      return Ie(`(max-width: ${r(l)})`, t);
    },
    between(l, i) {
      return Ie(`(min-width: ${r(l)}) and (max-width: ${r(i, -0.1)})`, t);
    },
    isGreater(l) {
      return s(`(min-width: ${r(l, 0.1)})`);
    },
    isGreaterOrEqual(l) {
      return s(`(min-width: ${r(l)})`);
    },
    isSmaller(l) {
      return s(`(max-width: ${r(l, -0.1)})`);
    },
    isSmallerOrEqual(l) {
      return s(`(max-width: ${r(l)})`);
    },
    isInBetween(l, i) {
      return s(`(min-width: ${r(l)}) and (max-width: ${r(i, -0.1)})`);
    }
  }, a);
}
function io(e) {
  return JSON.parse(JSON.stringify(e));
}
const Ct = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, $t = "__vueuse_ssr_handlers__";
Ct[$t] = Ct[$t] || {};
Ct[$t];
var Zt;
(function(e) {
  e.UP = "UP", e.RIGHT = "RIGHT", e.DOWN = "DOWN", e.LEFT = "LEFT", e.NONE = "NONE";
})(Zt || (Zt = {}));
var uo = Object.defineProperty, er = Object.getOwnPropertySymbols, co = Object.prototype.hasOwnProperty, fo = Object.prototype.propertyIsEnumerable, tr = (e, t, r) => t in e ? uo(e, t, { enumerable: !0, configurable: !0, writable: !0, value: r }) : e[t] = r, go = (e, t) => {
  for (var r in t || (t = {}))
    co.call(t, r) && tr(e, r, t[r]);
  if (er)
    for (var r of er(t))
      fo.call(t, r) && tr(e, r, t[r]);
  return e;
};
const po = {
  easeInSine: [0.12, 0, 0.39, 0],
  easeOutSine: [0.61, 1, 0.88, 1],
  easeInOutSine: [0.37, 0, 0.63, 1],
  easeInQuad: [0.11, 0, 0.5, 0],
  easeOutQuad: [0.5, 1, 0.89, 1],
  easeInOutQuad: [0.45, 0, 0.55, 1],
  easeInCubic: [0.32, 0, 0.67, 0],
  easeOutCubic: [0.33, 1, 0.68, 1],
  easeInOutCubic: [0.65, 0, 0.35, 1],
  easeInQuart: [0.5, 0, 0.75, 0],
  easeOutQuart: [0.25, 1, 0.5, 1],
  easeInOutQuart: [0.76, 0, 0.24, 1],
  easeInQuint: [0.64, 0, 0.78, 0],
  easeOutQuint: [0.22, 1, 0.36, 1],
  easeInOutQuint: [0.83, 0, 0.17, 1],
  easeInExpo: [0.7, 0, 0.84, 0],
  easeOutExpo: [0.16, 1, 0.3, 1],
  easeInOutExpo: [0.87, 0, 0.13, 1],
  easeInCirc: [0.55, 0, 1, 0.45],
  easeOutCirc: [0, 0.55, 0.45, 1],
  easeInOutCirc: [0.85, 0, 0.15, 1],
  easeInBack: [0.36, 0, 0.66, -0.56],
  easeOutBack: [0.34, 1.56, 0.64, 1],
  easeInOutBack: [0.68, -0.6, 0.32, 1.6]
};
go({
  linear: Js
}, po);
function ho(e, t, r, n = {}) {
  var s, o, a;
  const {
    clone: l = !1,
    passive: i = !1,
    eventName: f,
    deep: p = !1,
    defaultValue: v
  } = n, y = Br(), h = r || (y == null ? void 0 : y.emit) || ((s = y == null ? void 0 : y.$emit) == null ? void 0 : s.bind(y)) || ((a = (o = y == null ? void 0 : y.proxy) == null ? void 0 : o.$emit) == null ? void 0 : a.bind(y == null ? void 0 : y.proxy));
  let b = f;
  t || (t = "modelValue"), b = f || b || `update:${t.toString()}`;
  const C = (k) => l ? Ks(l) ? l(k) : io(k) : k, _ = () => Us(e[t]) ? C(e[t]) : v;
  if (i) {
    const k = _(), T = I(k);
    return Ze(() => e[t], (O) => T.value = C(O)), Ze(T, (O) => {
      (O !== e[t] || p) && h(b, O);
    }, { deep: p }), T;
  } else
    return u({
      get() {
        return _();
      },
      set(k) {
        h(b, k);
      }
    });
}
function mo(e, t = "default", r = void 0) {
  const n = e[t];
  if (!n)
    return console.warn("getFirstSlotVNode", `slot[${t}] is empty`), null;
  const s = tt(n(r));
  return s.length === 1 ? s[0] : (console.warn("getFirstSlotVNode", `slot[${t}] should have exactly one child`), null);
}
var bo = typeof global == "object" && global && global.Object === Object && global;
const vo = bo;
var yo = typeof self == "object" && self && self.Object === Object && self, wo = vo || yo || Function("return this")();
const Bt = wo;
var xo = Bt.Symbol;
const ue = xo;
var Qr = Object.prototype, ko = Qr.hasOwnProperty, _o = Qr.toString, Be = ue ? ue.toStringTag : void 0;
function Co(e) {
  var t = ko.call(e, Be), r = e[Be];
  try {
    e[Be] = void 0;
    var n = !0;
  } catch {
  }
  var s = _o.call(e);
  return n && (t ? e[Be] = r : delete e[Be]), s;
}
var $o = Object.prototype, So = $o.toString;
function To(e) {
  return So.call(e);
}
var Po = "[object Null]", zo = "[object Undefined]", rr = ue ? ue.toStringTag : void 0;
function Et(e) {
  return e == null ? e === void 0 ? zo : Po : rr && rr in Object(e) ? Co(e) : To(e);
}
function Lt(e) {
  return e != null && typeof e == "object";
}
var Oo = "[object Symbol]";
function Mt(e) {
  return typeof e == "symbol" || Lt(e) && Et(e) == Oo;
}
function Ao(e, t) {
  for (var r = -1, n = e == null ? 0 : e.length, s = Array(n); ++r < n; )
    s[r] = t(e[r], r, e);
  return s;
}
var Io = Array.isArray;
const ze = Io;
var Bo = 1 / 0, nr = ue ? ue.prototype : void 0, sr = nr ? nr.toString : void 0;
function Xr(e) {
  if (typeof e == "string")
    return e;
  if (ze(e))
    return Ao(e, Xr) + "";
  if (Mt(e))
    return sr ? sr.call(e) : "";
  var t = e + "";
  return t == "0" && 1 / e == -Bo ? "-0" : t;
}
function nt(e) {
  var t = typeof e;
  return e != null && (t == "object" || t == "function");
}
function Eo(e) {
  return e;
}
var Lo = "[object AsyncFunction]", Mo = "[object Function]", No = "[object GeneratorFunction]", Vo = "[object Proxy]";
function Ro(e) {
  if (!nt(e))
    return !1;
  var t = Et(e);
  return t == Mo || t == No || t == Lo || t == Vo;
}
var Ho = Bt["__core-js_shared__"];
const xt = Ho;
var or = function() {
  var e = /[^.]+$/.exec(xt && xt.keys && xt.keys.IE_PROTO || "");
  return e ? "Symbol(src)_1." + e : "";
}();
function Do(e) {
  return !!or && or in e;
}
var Fo = Function.prototype, jo = Fo.toString;
function Wo(e) {
  if (e != null) {
    try {
      return jo.call(e);
    } catch {
    }
    try {
      return e + "";
    } catch {
    }
  }
  return "";
}
var Go = /[\\^$.*+?()[\]{}|]/g, qo = /^\[object .+?Constructor\]$/, Uo = Function.prototype, Ko = Object.prototype, Yo = Uo.toString, Jo = Ko.hasOwnProperty, Qo = RegExp(
  "^" + Yo.call(Jo).replace(Go, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$"
);
function Xo(e) {
  if (!nt(e) || Do(e))
    return !1;
  var t = Ro(e) ? Qo : qo;
  return t.test(Wo(e));
}
function Zo(e, t) {
  return e == null ? void 0 : e[t];
}
function Nt(e, t) {
  var r = Zo(e, t);
  return Xo(r) ? r : void 0;
}
function ea(e, t, r) {
  switch (r.length) {
    case 0:
      return e.call(t);
    case 1:
      return e.call(t, r[0]);
    case 2:
      return e.call(t, r[0], r[1]);
    case 3:
      return e.call(t, r[0], r[1], r[2]);
  }
  return e.apply(t, r);
}
var ta = 800, ra = 16, na = Date.now;
function sa(e) {
  var t = 0, r = 0;
  return function() {
    var n = na(), s = ra - (n - r);
    if (r = n, s > 0) {
      if (++t >= ta)
        return arguments[0];
    } else
      t = 0;
    return e.apply(void 0, arguments);
  };
}
function oa(e) {
  return function() {
    return e;
  };
}
var aa = function() {
  try {
    var e = Nt(Object, "defineProperty");
    return e({}, "", {}), e;
  } catch {
  }
}();
const st = aa;
var la = st ? function(e, t) {
  return st(e, "toString", {
    configurable: !0,
    enumerable: !1,
    value: oa(t),
    writable: !0
  });
} : Eo;
const ia = la;
var ua = sa(ia);
const da = ua;
var ca = 9007199254740991, fa = /^(?:0|[1-9]\d*)$/;
function Zr(e, t) {
  var r = typeof e;
  return t = t == null ? ca : t, !!t && (r == "number" || r != "symbol" && fa.test(e)) && e > -1 && e % 1 == 0 && e < t;
}
function ga(e, t, r) {
  t == "__proto__" && st ? st(e, t, {
    configurable: !0,
    enumerable: !0,
    value: r,
    writable: !0
  }) : e[t] = r;
}
function en(e, t) {
  return e === t || e !== e && t !== t;
}
var pa = Object.prototype, ha = pa.hasOwnProperty;
function ma(e, t, r) {
  var n = e[t];
  (!(ha.call(e, t) && en(n, r)) || r === void 0 && !(t in e)) && ga(e, t, r);
}
var ar = Math.max;
function ba(e, t, r) {
  return t = ar(t === void 0 ? e.length - 1 : t, 0), function() {
    for (var n = arguments, s = -1, o = ar(n.length - t, 0), a = Array(o); ++s < o; )
      a[s] = n[t + s];
    s = -1;
    for (var l = Array(t + 1); ++s < t; )
      l[s] = n[s];
    return l[t] = r(a), ea(e, this, l);
  };
}
var va = 9007199254740991;
function ya(e) {
  return typeof e == "number" && e > -1 && e % 1 == 0 && e <= va;
}
var wa = "[object Arguments]";
function lr(e) {
  return Lt(e) && Et(e) == wa;
}
var tn = Object.prototype, xa = tn.hasOwnProperty, ka = tn.propertyIsEnumerable, _a = lr(function() {
  return arguments;
}()) ? lr : function(e) {
  return Lt(e) && xa.call(e, "callee") && !ka.call(e, "callee");
};
const rn = _a;
var Ca = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/, $a = /^\w*$/;
function Sa(e, t) {
  if (ze(e))
    return !1;
  var r = typeof e;
  return r == "number" || r == "symbol" || r == "boolean" || e == null || Mt(e) ? !0 : $a.test(e) || !Ca.test(e) || t != null && e in Object(t);
}
var Ta = Nt(Object, "create");
const De = Ta;
function Pa() {
  this.__data__ = De ? De(null) : {}, this.size = 0;
}
function za(e) {
  var t = this.has(e) && delete this.__data__[e];
  return this.size -= t ? 1 : 0, t;
}
var Oa = "__lodash_hash_undefined__", Aa = Object.prototype, Ia = Aa.hasOwnProperty;
function Ba(e) {
  var t = this.__data__;
  if (De) {
    var r = t[e];
    return r === Oa ? void 0 : r;
  }
  return Ia.call(t, e) ? t[e] : void 0;
}
var Ea = Object.prototype, La = Ea.hasOwnProperty;
function Ma(e) {
  var t = this.__data__;
  return De ? t[e] !== void 0 : La.call(t, e);
}
var Na = "__lodash_hash_undefined__";
function Va(e, t) {
  var r = this.__data__;
  return this.size += this.has(e) ? 0 : 1, r[e] = De && t === void 0 ? Na : t, this;
}
function ve(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
ve.prototype.clear = Pa;
ve.prototype.delete = za;
ve.prototype.get = Ba;
ve.prototype.has = Ma;
ve.prototype.set = Va;
function Ra() {
  this.__data__ = [], this.size = 0;
}
function ct(e, t) {
  for (var r = e.length; r--; )
    if (en(e[r][0], t))
      return r;
  return -1;
}
var Ha = Array.prototype, Da = Ha.splice;
function Fa(e) {
  var t = this.__data__, r = ct(t, e);
  if (r < 0)
    return !1;
  var n = t.length - 1;
  return r == n ? t.pop() : Da.call(t, r, 1), --this.size, !0;
}
function ja(e) {
  var t = this.__data__, r = ct(t, e);
  return r < 0 ? void 0 : t[r][1];
}
function Wa(e) {
  return ct(this.__data__, e) > -1;
}
function Ga(e, t) {
  var r = this.__data__, n = ct(r, e);
  return n < 0 ? (++this.size, r.push([e, t])) : r[n][1] = t, this;
}
function Oe(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
Oe.prototype.clear = Ra;
Oe.prototype.delete = Fa;
Oe.prototype.get = ja;
Oe.prototype.has = Wa;
Oe.prototype.set = Ga;
var qa = Nt(Bt, "Map");
const Ua = qa;
function Ka() {
  this.size = 0, this.__data__ = {
    hash: new ve(),
    map: new (Ua || Oe)(),
    string: new ve()
  };
}
function Ya(e) {
  var t = typeof e;
  return t == "string" || t == "number" || t == "symbol" || t == "boolean" ? e !== "__proto__" : e === null;
}
function ft(e, t) {
  var r = e.__data__;
  return Ya(t) ? r[typeof t == "string" ? "string" : "hash"] : r.map;
}
function Ja(e) {
  var t = ft(this, e).delete(e);
  return this.size -= t ? 1 : 0, t;
}
function Qa(e) {
  return ft(this, e).get(e);
}
function Xa(e) {
  return ft(this, e).has(e);
}
function Za(e, t) {
  var r = ft(this, e), n = r.size;
  return r.set(e, t), this.size += r.size == n ? 0 : 1, this;
}
function xe(e) {
  var t = -1, r = e == null ? 0 : e.length;
  for (this.clear(); ++t < r; ) {
    var n = e[t];
    this.set(n[0], n[1]);
  }
}
xe.prototype.clear = Ka;
xe.prototype.delete = Ja;
xe.prototype.get = Qa;
xe.prototype.has = Xa;
xe.prototype.set = Za;
var el = "Expected a function";
function Vt(e, t) {
  if (typeof e != "function" || t != null && typeof t != "function")
    throw new TypeError(el);
  var r = function() {
    var n = arguments, s = t ? t.apply(this, n) : n[0], o = r.cache;
    if (o.has(s))
      return o.get(s);
    var a = e.apply(this, n);
    return r.cache = o.set(s, a) || o, a;
  };
  return r.cache = new (Vt.Cache || xe)(), r;
}
Vt.Cache = xe;
var tl = 500;
function rl(e) {
  var t = Vt(e, function(n) {
    return r.size === tl && r.clear(), n;
  }), r = t.cache;
  return t;
}
var nl = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g, sl = /\\(\\)?/g, ol = rl(function(e) {
  var t = [];
  return e.charCodeAt(0) === 46 && t.push(""), e.replace(nl, function(r, n, s, o) {
    t.push(s ? o.replace(sl, "$1") : n || r);
  }), t;
});
const al = ol;
function ll(e) {
  return e == null ? "" : Xr(e);
}
function gt(e, t) {
  return ze(e) ? e : Sa(e, t) ? [e] : al(ll(e));
}
var il = 1 / 0;
function Rt(e) {
  if (typeof e == "string" || Mt(e))
    return e;
  var t = e + "";
  return t == "0" && 1 / e == -il ? "-0" : t;
}
function ul(e, t) {
  t = gt(t, e);
  for (var r = 0, n = t.length; e != null && r < n; )
    e = e[Rt(t[r++])];
  return r && r == n ? e : void 0;
}
function dl(e, t) {
  for (var r = -1, n = t.length, s = e.length; ++r < n; )
    e[s + r] = t[r];
  return e;
}
var ir = ue ? ue.isConcatSpreadable : void 0;
function cl(e) {
  return ze(e) || rn(e) || !!(ir && e && e[ir]);
}
function nn(e, t, r, n, s) {
  var o = -1, a = e.length;
  for (r || (r = cl), s || (s = []); ++o < a; ) {
    var l = e[o];
    t > 0 && r(l) ? t > 1 ? nn(l, t - 1, r, n, s) : dl(s, l) : n || (s[s.length] = l);
  }
  return s;
}
function fl(e) {
  var t = e == null ? 0 : e.length;
  return t ? nn(e, 1) : [];
}
function gl(e) {
  return da(ba(e, void 0, fl), e + "");
}
function pl(e, t) {
  return e != null && t in Object(e);
}
function hl(e, t, r) {
  t = gt(t, e);
  for (var n = -1, s = t.length, o = !1; ++n < s; ) {
    var a = Rt(t[n]);
    if (!(o = e != null && r(e, a)))
      break;
    e = e[a];
  }
  return o || ++n != s ? o : (s = e == null ? 0 : e.length, !!s && ya(s) && Zr(a, s) && (ze(e) || rn(e)));
}
function ml(e, t) {
  return e != null && hl(e, t, pl);
}
function bl(e, t, r, n) {
  if (!nt(e))
    return e;
  t = gt(t, e);
  for (var s = -1, o = t.length, a = o - 1, l = e; l != null && ++s < o; ) {
    var i = Rt(t[s]), f = r;
    if (i === "__proto__" || i === "constructor" || i === "prototype")
      return e;
    if (s != a) {
      var p = l[i];
      f = n ? n(p, i, l) : void 0, f === void 0 && (f = nt(p) ? p : Zr(t[s + 1]) ? [] : {});
    }
    ma(l, i, f), l = l[i];
  }
  return e;
}
function vl(e, t, r) {
  for (var n = -1, s = t.length, o = {}; ++n < s; ) {
    var a = t[n], l = ul(e, a);
    r(l, a) && bl(o, gt(a, e), l);
  }
  return o;
}
function yl(e, t) {
  return vl(e, t, function(r, n) {
    return ml(e, n);
  });
}
var wl = gl(function(e, t) {
  return e == null ? {} : yl(e, t);
});
const xl = wl, kl = {
  focus: ["onFocus", "onBlur"],
  click: ["onClick"],
  hover: ["onMouseenter", "onMouseleave"]
};
function _l(e, t) {
  Object.entries(kl).forEach(([, r]) => {
    r.forEach((n) => {
      e.props ? e.props = Object.assign({}, e.props) : e.props = {};
      const s = e.props[n], o = t[n];
      s ? e.props[n] = (...a) => {
        s(...a), o(...a);
      } : e.props[n] = o;
    });
  });
}
const ur = $({
  name: "SlotListener",
  emits: ["click", "focus", "blur", "mouseenter", "mouseleave"],
  props: {
    trigger: {
      type: String,
      default: "click"
    }
  },
  setup(e, { emit: t }) {
    return {
      handleClick: (l) => {
        t("click", l);
      },
      handleBlur: (l) => {
        t("blur", l);
      },
      handleFocus: (l) => {
        t("focus", l);
      },
      handleMouseLeave: (l) => {
        t("mouseleave", l);
      },
      handleMouseEnter: (l) => {
        t("mouseenter", l);
      }
    };
  },
  render() {
    const {
      $slots: e
    } = this, t = {
      onClick: this.handleClick,
      onMouseenter: this.handleMouseEnter,
      onMouseleave: this.handleMouseLeave,
      onFocus: this.handleFocus,
      onBlur: this.handleBlur
    };
    let r = mo(e, "default");
    const n = [
      t
    ];
    return r != null && r.props && n.push(xl(r.props, "onClick", "onMouseenter", "onMouseleave", "onFocus", "onBlur")), r && _l(
      r,
      {
        onBlur: (o) => {
          n.forEach((a) => {
            var l;
            (l = a == null ? void 0 : a.onBlur) == null || l.call(a, o);
          });
        },
        onFocus: (o) => {
          n.forEach((a) => {
            var l;
            (l = a == null ? void 0 : a.onFocus) == null || l.call(a, o);
          });
        },
        onClick: (o) => {
          n.forEach((a) => {
            var l;
            (l = a == null ? void 0 : a.onClick) == null || l.call(a, o);
          });
        },
        onMouseenter: (o) => {
          n.forEach((a) => {
            var l;
            (l = a == null ? void 0 : a.onMouseenter) == null || l.call(a, o);
          });
        },
        onMouseleave: (o) => {
          n.forEach((a) => {
            var l;
            (l = a == null ? void 0 : a.onMouseleave) == null || l.call(a, o);
          });
        }
      }
    ), r;
  }
}), Cl = (e) => (Er("data-v-cf0604df"), e = e(), Lr(), e), $l = { class: "inline-flex items-center" }, Sl = /* @__PURE__ */ Cl(() => /* @__PURE__ */ m("svg", {
  class: "w-4 h-4 ml-2",
  fill: "none",
  stroke: "currentColor",
  viewBox: "0 0 24 24",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ m("path", {
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "2",
    d: "M19 9l-7 7-7-7"
  })
], -1)), Tl = /* @__PURE__ */ $({
  __name: "Dropdown",
  props: {
    placement: {
      type: String,
      default: "bottom"
    },
    text: {
      type: String,
      default: ""
    },
    transition: {
      type: [String, null],
      default: null
    }
  },
  setup(e) {
    const t = e, r = I(!1), n = () => r.value = !1, s = () => r.value = !r.value, o = {
      bottom: "to-bottom",
      left: "to-left",
      right: "to-right",
      top: "to-top"
    }, a = u(() => t.transition === null ? o[t.placement] : t.transition), l = I(), i = I(), { contentClasses: f, contentStyles: p } = qs({
      placement: He(t, "placement"),
      visible: r,
      contentRef: l
    });
    return ro(i, () => {
      !r.value || (r.value = !1);
    }), (v, y) => (c(), g("div", {
      class: "inline-flex relative",
      ref_key: "wrapper",
      ref: i
    }, [
      m("div", $l, [
        Ce(ur, { onClick: s }, {
          default: H(() => [
            x(v.$slots, "trigger", {}, () => [
              Ce(gs, null, {
                suffix: H(() => [
                  Sl
                ]),
                default: H(() => [
                  ne(B(e.text) + " ", 1)
                ]),
                _: 1
              })
            ], !0)
          ]),
          _: 3
        })
      ]),
      Ce(In, { name: d(a) }, {
        default: H(() => [
          r.value ? (c(), g("div", {
            key: 0,
            ref_key: "content",
            ref: l,
            style: Le(d(p)),
            class: w([d(f)])
          }, [
            Ce(ur, { onClick: n }, {
              default: H(() => [
                x(v.$slots, "default", {}, void 0, !0)
              ]),
              _: 3
            })
          ], 6)) : S("", !0)
        ]),
        _: 3
      }, 8, ["name"])
    ], 512));
  }
});
const k0 = /* @__PURE__ */ oe(Tl, [["__scopeId", "data-v-cf0604df"]]), _0 = /* @__PURE__ */ $({
  __name: "FlowbiteThemable",
  props: {
    theme: {
      type: String,
      required: !0
    }
  },
  setup(e) {
    return X(Fr, He(e, "theme")), (r, n) => (c(), g("div", null, [
      x(r.$slots, "default")
    ]));
  }
}), Pl = "inline-flex items-center space-x-1 md:space-x-3", zl = {
  default: "flex",
  solid: "flex px-5 py-3 text-gray-700 border border-gray-200 rounded-lg bg-gray-50 dark:bg-gray-800 dark:border-gray-700"
};
function Ol(e) {
  const t = u(() => P(
    Pl
  )), r = u(() => P(
    zl[e.solid.value ? "solid" : "defauilt"]
  ));
  return {
    breadcrumbClasses: t,
    breadcrumbWrapperClasses: r
  };
}
const C0 = /* @__PURE__ */ $({
  __name: "Breadcrumb",
  props: {
    solid: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, { breadcrumbWrapperClasses: r, breadcrumbClasses: n } = Ol(N(t));
    return (s, o) => (c(), g("nav", {
      class: w(d(r)),
      "aria-label": "Breadcrumb"
    }, [
      m("ol", {
        class: w(d(n))
      }, [
        x(s.$slots, "default")
      ], 2)
    ], 2));
  }
}), Al = "ml-1 inline-flex items-center text-sm font-medium dark:text-gray-400", Il = "text-gray-700 hover:text-gray-900 dark:hover:text-white", Bl = "text-gray-500";
function El(e) {
  return {
    breadcrumbItemClasses: u(() => P(
      Al,
      e.href.value ? Il : Bl
    ))
  };
}
const Ll = { class: "inline-flex items-center" }, Ml = {
  key: 0,
  class: "w-6 h-6 text-gray-400 mr-1",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, Nl = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z",
  "clip-rule": "evenodd"
}, null, -1), Vl = [
  Nl
], Rl = {
  key: 0,
  class: "w-4 h-4 mr-2",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, Hl = /* @__PURE__ */ m("path", { d: "M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z" }, null, -1), Dl = [
  Hl
], $0 = /* @__PURE__ */ $({
  __name: "BreadcrumbItem",
  props: {
    href: {
      type: String,
      default: null
    },
    home: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = u(() => t.href ? "a" : "span"), { breadcrumbItemClasses: n } = El(N(t));
    return (s, o) => (c(), g("li", Ll, [
      x(s.$slots, "arrow-icon", {}, () => [
        e.home ? S("", !0) : (c(), g("svg", Ml, Vl))
      ]),
      (c(), D(ut(d(r)), {
        href: e.href,
        class: w(d(n))
      }, {
        default: H(() => [
          x(s.$slots, "home-icon", {}, () => [
            e.home ? (c(), g("svg", Rl, Dl)) : S("", !0)
          ]),
          x(s.$slots, "default")
        ]),
        _: 3
      }, 8, ["href", "class"]))
    ]));
  }
}), dr = {
  xs: "w-6 h-6",
  sm: "w-8 h-8",
  md: "w-10 h-10",
  lg: "w-20 h-20",
  xl: "w-36 h-36"
}, cr = {
  default: "rounded",
  rounded: "rounded-full"
}, Fl = "ring-2 ring-gray-300 dark:ring-gray-500 p-1", jl = "absolute h-3.5 w-3.5 rounded-full border-2 border-white dark:border-gray-800", Wl = {
  away: "bg-gray-400",
  busy: "bg-yellow-400",
  offline: "bg-red-400",
  online: "bg-green-400"
}, Gl = {
  "top-right-rounded": "top-0 -right-0.5",
  "top-right-default": "-top-1.5 -right-1.5",
  "top-left-rounded": "top-0 left-0",
  "top-left-default": "top-0 left-0 transform -translate-y-1/2 -translate-x-1/2",
  "bottom-right-rounded": "bottom-0 -right-0.5",
  "bottom-right-default": "bottom-0 -right-1.5 translate-y-1/2",
  "bottom-left-rounded": "bottom-0 left-0",
  "bottom-left-default": "-bottom-1.5 left-0 transform -translate-x-1/2 "
}, ql = "absolute w-auto h-auto text-gray-400", Ul = "inline-flex overflow-hidden relative justify-center items-center bg-gray-100 dark:bg-gray-600", Kl = "font-medium text-gray-600 dark:text-gray-300", Yl = {
  xs: "bottom-0",
  sm: "bottom-0",
  md: "-bottom-1",
  lg: "-bottom-2",
  xl: "-bottom-4"
};
function Jl(e) {
  const t = u(() => P(
    dr[e.size.value],
    cr[e.rounded.value ? "rounded" : "default"],
    e.bordered.value ? Fl : "",
    e.stacked.value ? "border-2 border-white dark:border-gray-800" : ""
  )), r = u(() => {
    const a = `${e.statusPosition.value}-${e.rounded.value ? "rounded" : "default"}`;
    return P(
      jl,
      Wl[e.status.value],
      Gl[a]
    );
  }), n = u(() => P(
    ql,
    Yl[e.size.value]
  )), s = u(() => P(
    Ul,
    dr[e.size.value],
    cr[e.rounded.value ? "rounded" : "default"]
  )), o = u(() => P(
    Kl
  ));
  return {
    avatarClasses: t,
    avatarDotClasses: r,
    avatarPlaceholderClasses: n,
    avatarPlaceholderWrapperClasses: s,
    avatarPlaceholderInitialsClasses: o
  };
}
const Ql = { class: "relative" }, Xl = ["src", "alt"], Zl = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z",
  "clip-rule": "evenodd"
}, null, -1), ei = [
  Zl
], ti = ["data-pos"], S0 = /* @__PURE__ */ $({
  __name: "Avatar",
  props: {
    alt: {
      type: String,
      default: "Avatar"
    },
    bordered: {
      type: Boolean,
      default: !1
    },
    img: {
      type: String,
      default: ""
    },
    rounded: {
      type: Boolean,
      default: !1
    },
    size: {
      type: String,
      default: "md"
    },
    stacked: {
      type: Boolean,
      default: !1
    },
    status: {
      type: String,
      default: null
    },
    statusPosition: {
      type: String,
      default: "top-right"
    },
    initials: {
      type: String,
      default: null
    }
  },
  setup(e) {
    const t = e, r = I(!1);
    function n() {
      r.value = !0;
    }
    const s = Pe(), o = u(() => s.placeholder), { avatarClasses: a, avatarDotClasses: l, avatarPlaceholderClasses: i, avatarPlaceholderWrapperClasses: f, avatarPlaceholderInitialsClasses: p } = Jl(N(t));
    return (v, y) => (c(), g("div", Ql, [
      e.img && !r.value ? (c(), g("img", {
        key: 0,
        class: w(d(a)),
        src: e.img,
        alt: e.alt,
        onError: n
      }, null, 42, Xl)) : (c(), g("div", {
        key: 1,
        class: w(d(f))
      }, [
        !e.initials && !d(o) ? (c(), g("svg", {
          key: 0,
          class: w(d(i)),
          fill: "currentColor",
          viewBox: "0 0 20 20",
          xmlns: "http://www.w3.org/2000/svg"
        }, ei, 2)) : S("", !0),
        !e.initials && d(o) ? (c(), g("div", {
          key: 1,
          class: w(d(i))
        }, [
          x(v.$slots, "placeholder")
        ], 2)) : (c(), g("div", {
          key: 2,
          class: w(d(p))
        }, B(e.initials), 3)),
        e.status ? (c(), g("span", {
          key: 3,
          class: w(d(l)),
          "data-pos": e.statusPosition
        }, null, 10, ti)) : S("", !0)
      ], 2))
    ]));
  }
}), ri = {}, ni = { class: "flex -space-x-4" };
function si(e, t) {
  return c(), g("div", ni, [
    x(e.$slots, "default")
  ]);
}
const T0 = /* @__PURE__ */ oe(ri, [["render", si]]), oi = ["href"], P0 = {
  __name: "StackedAvatarsCounter",
  props: {
    total: {
      type: Number,
      default: 1
    },
    href: {
      type: String,
      default: "#"
    }
  },
  setup(e) {
    return (t, r) => (c(), g("a", {
      class: "relative flex justify-center items-center w-10 h-10 text-xs font-medium text-white bg-gray-700 rounded-full border-2 border-white hover:bg-gray-600 dark:border-gray-800",
      href: e.href
    }, "+" + B(e.total), 9, oi));
  }
}, ai = "mr-2 px-2.5 py-0.5 rounded flex items-center justify-center", li = "bg-blue-100 hover:bg-blue-200 text-blue-800 dark:text-blue-800 dark:hover:bg-blue-300", ii = "p-1 rounded-full mr-2", ui = {
  default: "text-blue-800 dark:text-blue-800",
  dark: "text-gray-800 dark:bg-gray-700",
  red: "text-red-800 dark:text-red-900",
  green: "text-green-800 dark:text-green-900",
  yellow: "text-yellow-800 dark:text-yellow-900",
  indigo: "text-indigo-800 dark:text-indigo-900",
  purple: "text-purple-800 dark:text-purple-900",
  pink: "text-pink-800 dark:text-pink-900"
}, di = {
  default: "bg-blue-100 dark:bg-blue-200",
  dark: "bg-gray-100 dark:bg-gray-700",
  red: "bg-red-100 dark:bg-red-200",
  green: "bg-green-100 dark:bg-green-200",
  yellow: "bg-yellow-100 dark:bg-yellow-200",
  indigo: "bg-indigo-100 dark:bg-indigo-200",
  purple: "bg-purple-100 dark:bg-purple-200",
  pink: "bg-pink-100 dark:bg-pink-200"
}, ci = {
  xs: "text-xs font-semibold",
  sm: "text-sm font-medium"
};
function fi(e, t) {
  return {
    badgeClasses: u(() => P(
      ci[e.size.value],
      e.href.value ? "" : di[e.type.value],
      e.href.value ? "" : ui[e.type.value],
      e.href.value ? li : "",
      t.isContentEmpty.value ? ii : ai
    ))
  };
}
const z0 = /* @__PURE__ */ $({
  __name: "Badge",
  props: {
    type: {
      type: String,
      default: "default"
    },
    size: {
      type: String,
      default: "xs"
    },
    href: {
      type: String,
      default: null
    }
  },
  setup(e) {
    const t = e, r = Pe(), n = u(() => !r.default), s = u(() => t.href ? "a" : "span"), { badgeClasses: o } = fi(N(t), { isContentEmpty: n });
    return (a, l) => (c(), D(ut(d(s)), {
      class: w(d(o)),
      href: e.href
    }, {
      default: H(() => [
        x(a.$slots, "icon"),
        x(a.$slots, "default")
      ]),
      _: 3
    }, 8, ["class", "href"]));
  }
});
function gi(e) {
  const t = u(() => e.variant.value === "default" ? "block max-w-sm bg-white rounded-lg border border-gray-200 shadow-md hover:bg-gray-100 dark:bg-gray-800 dark:border-gray-700 dark:hover:bg-gray-700" : e.variant.value === "image" ? "max-w-sm bg-white rounded-lg border border-gray-200 shadow-md dark:bg-gray-800 dark:border-gray-700" : e.variant.value === "horizontal" ? "flex flex-col items-center bg-white rounded-lg border shadow-md md:flex-row md:max-w-xl hover:bg-gray-100 dark:border-gray-700 dark:bg-gray-800 dark:hover:bg-gray-700" : ""), r = u(() => e.variant.value === "horizontal" ? "object-cover w-full h-96 rounded-t-lg md:h-auto md:w-48 md:rounded-none md:rounded-l-lg" : "");
  return {
    cardClasses: t,
    horizontalImageClasses: r
  };
}
const pi = ["src", "alt"], hi = { class: "p-6" }, O0 = /* @__PURE__ */ $({
  __name: "TheCard",
  props: {
    href: {
      type: String,
      default: ""
    },
    imgAlt: {
      type: String,
      default: ""
    },
    imgSrc: {
      type: String,
      default: ""
    },
    variant: {
      type: String,
      default: "default"
    }
  },
  setup(e) {
    const t = e, { cardClasses: r, horizontalImageClasses: n } = gi(N(t)), s = u(() => t.href ? "a" : "div");
    return (o, a) => (c(), D(ut(d(s)), {
      href: e.href,
      class: w(d(r))
    }, {
      default: H(() => [
        e.imgSrc ? (c(), g("img", {
          key: 0,
          class: w(["rounded-t-lg", d(n)]),
          src: e.imgSrc,
          alt: e.imgAlt
        }, null, 10, pi)) : S("", !0),
        m("div", hi, [
          x(o.$slots, "default")
        ])
      ]),
      _: 3
    }, 8, ["href", "class"]));
  }
}), mi = {
  id: "default-carousel",
  class: "relative"
}, bi = { class: "overflow-hidden relative h-56 rounded-lg sm:h-64 xl:h-80 2xl:h-96" }, vi = ["src", "alt"], yi = {
  key: 0,
  class: "flex absolute bottom-5 left-1/2 z-30 space-x-3 -translate-x-1/2"
}, wi = ["aria-label", "onClick"], xi = ["onClick"], ki = /* @__PURE__ */ m("span", { class: "inline-flex justify-center items-center w-8 h-8 rounded-full sm:w-10 sm:h-10 bg-white/30 dark:bg-gray-800/30 group-hover:bg-white/50 dark:group-hover:bg-gray-800/60 group-focus:ring-4 group-focus:ring-white dark:group-focus:ring-gray-800/70 group-focus:outline-none" }, [
  /* @__PURE__ */ m("svg", {
    class: "w-5 h-5 text-white sm:w-6 sm:h-6 dark:text-gray-800",
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24",
    xmlns: "http://www.w3.org/2000/svg"
  }, [
    /* @__PURE__ */ m("path", {
      "stroke-linecap": "round",
      "stroke-linejoin": "round",
      "stroke-width": "2",
      d: "M15 19l-7-7 7-7"
    })
  ]),
  /* @__PURE__ */ m("span", { class: "hidden" }, "Previous")
], -1), _i = [
  ki
], Ci = ["onClick"], $i = /* @__PURE__ */ m("span", { class: "inline-flex justify-center items-center w-8 h-8 rounded-full sm:w-10 sm:h-10 bg-white/30 dark:bg-gray-800/30 group-hover:bg-white/50 dark:group-hover:bg-gray-800/60 group-focus:ring-4 group-focus:ring-white dark:group-focus:ring-gray-800/70 group-focus:outline-none" }, [
  /* @__PURE__ */ m("svg", {
    class: "w-5 h-5 text-white sm:w-6 sm:h-6 dark:text-gray-800",
    fill: "none",
    stroke: "currentColor",
    viewBox: "0 0 24 24",
    xmlns: "http://www.w3.org/2000/svg"
  }, [
    /* @__PURE__ */ m("path", {
      "stroke-linecap": "round",
      "stroke-linejoin": "round",
      "stroke-width": "2",
      d: "M9 5l7 7-7 7"
    })
  ]),
  /* @__PURE__ */ m("span", { class: "hidden" }, "Next")
], -1), Si = [
  $i
], A0 = /* @__PURE__ */ $({
  __name: "Carousel",
  props: {
    pictures: {
      type: Array,
      default() {
        return [
          {
            src: "https://flowbite.com/docs/images/carousel/carousel-1.svg",
            alt: "Picture 1"
          },
          {
            src: "https://flowbite.com/docs/images/carousel/carousel-2.svg",
            alt: "Picture 2"
          },
          {
            src: "https://flowbite.com/docs/images/carousel/carousel-3.svg",
            alt: "Picture 3"
          }
        ];
      }
    },
    indicators: {
      type: Boolean,
      default: !0
    },
    controls: {
      type: Boolean,
      default: !0
    },
    slide: {
      type: Boolean,
      default: !1
    },
    slideInterval: {
      type: Number,
      default: 3e3
    },
    animation: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = I(0), n = I(""), s = I(), o = () => {
      s.value = setInterval(function() {
        i();
      }, t.slideInterval);
    }, a = () => {
      clearInterval(s.value), o();
    }, l = (p) => {
      r.value = p, a();
    }, i = () => {
      r.value !== t.pictures.length - 1 ? r.value++ : r.value = 0, n.value = "right", a();
    }, f = () => {
      r.value !== 0 ? r.value-- : r.value = t.pictures.length - 1, n.value = "left", a();
    };
    return Te(() => {
      t.slide && o();
    }), (p, v) => (c(), g("div", mi, [
      m("div", bi, [
        (c(!0), g(U, null, be(e.pictures, (y, h) => (c(), g("div", {
          class: w([h === r.value ? "z-30" : "z-0", "absolute inset-0 -translate-y-0"]),
          key: h
        }, [
          m("img", {
            src: y.src,
            class: "block absolute top-1/2 left-1/2 w-full -translate-x-1/2 -translate-y-1/2",
            alt: y.alt
          }, null, 8, vi)
        ], 2))), 128))
      ]),
      e.indicators ? (c(), g("div", yi, [
        (c(!0), g(U, null, be(e.pictures, (y, h) => (c(), g("button", {
          key: h,
          type: "button",
          class: w([h === r.value ? "bg-white" : "bg-white/50", "w-3 h-3 rounded-full bg-white"]),
          "aria-current": "false",
          "aria-label": "Slide " + h,
          onClick: Qe((b) => l(h), ["prevent"])
        }, null, 10, wi))), 128))
      ])) : S("", !0),
      e.controls ? (c(), g("button", {
        key: 1,
        onClick: Qe(f, ["prevent"]),
        type: "button",
        class: "flex absolute top-0 left-0 z-30 justify-center items-center px-4 h-full cursor-pointer group focus:outline-none",
        "data-carousel-prev": ""
      }, _i, 8, xi)) : S("", !0),
      e.controls ? (c(), g("button", {
        key: 2,
        onClick: Qe(i, ["prevent"]),
        type: "button",
        class: "flex absolute top-0 right-0 z-30 justify-center items-center px-4 h-full cursor-pointer group focus:outline-none",
        "data-carousel-next": ""
      }, Si, 8, Ci)) : S("", !0)
    ]));
  }
}), Ti = { class: "p-4 bg-white rounded-lg shadow md:flex md:items-center md:justify-between md:p-6 dark:bg-gray-800" }, Pi = /* @__PURE__ */ Mr('<span class="text-sm text-gray-500 sm:text-center dark:text-gray-400">\xA9 2022 <a href="https://flowbite.com/" class="hover:underline">Flowbite\u2122</a>. All Rights Reserved. </span><ul class="flex flex-wrap items-center mt-3 text-sm text-gray-500 dark:text-gray-400 sm:mt-0"><li><a href="#" class="mr-4 hover:underline md:mr-6">About</a></li><li><a href="#" class="mr-4 hover:underline md:mr-6">Privacy Policy</a></li><li><a href="#" class="mr-4 hover:underline md:mr-6">Licensing</a></li><li><a href="#" class="hover:underline">Contact</a></li></ul>', 2), zi = [
  Pi
], I0 = /* @__PURE__ */ $({
  __name: "Footer",
  props: {
    children: {
      type: Array,
      default() {
        return [];
      }
    },
    bgDark: {
      type: Boolean,
      default: !1
    },
    container: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    return (t, r) => (c(), g("footer", Ti, zi));
  }
}), Oi = "overflow-hidden w-48 text-sm font-medium text-gray-900 bg-white border border-gray-200 rounded-lg dark:bg-gray-700 dark:border-gray-600 dark:text-white";
function Ai() {
  return {
    containerClasses: u(() => P(
      Oi
    ))
  };
}
const B0 = /* @__PURE__ */ $({
  __name: "ListGroup",
  setup(e) {
    const { containerClasses: t } = Ai();
    return (r, n) => (c(), g("ul", {
      class: w(d(t))
    }, [
      x(r.$slots, "default")
    ], 2));
  }
}), Ii = "inline-flex items-center w-full px-4 py-2 border-b border-gray-200 dark:border-gray-600", Bi = "block w-full px-4 py-2 cursor-pointer hover:bg-gray-100 hover:text-blue-700 focus:outline-none focus:ring-2 focus:ring-blue-700 focus:text-blue-700 dark:hover:bg-gray-600 dark:hover:text-white dark:focus:ring-gray-500 dark:focus:text-white", Ei = "bg-gray-100 cursor-not-allowed dark:bg-gray-600 dark:text-gray-400";
function Li(e) {
  return {
    itemClasses: u(() => j(
      Ii,
      e.disabled.value ? Ei : "",
      !e.disabled.value && e.hover.value ? Bi : ""
    ))
  };
}
const Mi = {
  key: 0,
  class: "mr-2"
}, Ni = {
  key: 1,
  class: "ml-2"
}, E0 = /* @__PURE__ */ $({
  __name: "ListGroupItem",
  props: {
    hover: {
      type: Boolean,
      default: !0
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, { itemClasses: r } = Li(N(t));
    return (n, s) => (c(), g("li", {
      class: w(d(r))
    }, [
      n.$slots.prefix ? (c(), g("div", Mi, [
        x(n.$slots, "prefix")
      ])) : S("", !0),
      x(n.$slots, "default"),
      n.$slots.suffix ? (c(), g("div", Ni, [
        x(n.$slots, "suffix")
      ])) : S("", !0)
    ], 2));
  }
}), Vi = /* @__PURE__ */ m("div", { class: "bg-gray-900 bg-opacity-50 dark:bg-opacity-80 fixed inset-0 z-40" }, null, -1), Ri = ["onKeyup", "onClick"], Hi = { class: "relative bg-white rounded-lg shadow dark:bg-gray-700" }, Di = /* @__PURE__ */ m("svg", {
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ m("path", {
    "fill-rule": "evenodd",
    d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), Fi = {
  key: 0,
  class: "p-6 rounded-b border-gray-200 border-t dark:border-gray-600"
}, L0 = /* @__PURE__ */ $({
  __name: "Modal",
  props: {
    size: { default: "2xl" },
    escapable: { type: Boolean, default: !0 },
    persistent: { type: Boolean, default: !1 }
  },
  emits: ["close", "click:outside"],
  setup(e, { emit: t }) {
    const r = e, n = {
      xs: "max-w-xs",
      sm: "max-w-sm",
      md: "max-w-md",
      lg: "max-w-lg",
      xl: "max-w-xl",
      "2xl": "max-w-2xl",
      "3xl": "max-w-3xl",
      "4xl": "max-w-4xl",
      "5xl": "max-w-5xl",
      "6xl": "max-w-6xl",
      "7xl": "max-w-7xl"
    };
    function s() {
      t("close");
    }
    function o() {
      r.persistent || (t("click:outside"), s());
    }
    function a() {
      r.escapable && !r.persistent && s();
    }
    const l = I(null);
    return Te(() => {
      l.value && l.value.focus();
    }), (i, f) => (c(), g("div", null, [
      Vi,
      m("div", {
        tabindex: "0",
        ref_key: "modalRef",
        ref: l,
        class: "overflow-y-auto overflow-x-hidden fixed top-0 right-0 left-0 z-50 w-full md:inset-0 h-modal md:h-full justify-center items-center flex",
        onKeyup: Nr(a, ["esc"]),
        onClick: Qe(o, ["self"])
      }, [
        m("div", {
          class: w(["relative p-4 w-full h-full md:h-auto", `${n[e.size]}`])
        }, [
          m("div", Hi, [
            m("div", {
              class: w(["p-4 rounded-t flex justify-between items-center", i.$slots.header ? "border-b border-gray-200 dark:border-gray-600" : ""])
            }, [
              x(i.$slots, "header"),
              e.persistent ? S("", !0) : (c(), g("button", {
                key: 0,
                onClick: s,
                "aria-label": "close",
                type: "button",
                class: "text-gray-400 bg-transparent hover:bg-gray-200 hover:text-gray-900 rounded-lg text-sm p-1.5 ml-auto inline-flex items-center dark:hover:bg-gray-600 dark:hover:text-white"
              }, [
                x(i.$slots, "close-icon", {}, () => [
                  Di
                ])
              ]))
            ], 2),
            m("div", {
              class: w(["p-6", i.$slots.header ? "" : "pt-0"])
            }, [
              x(i.$slots, "body")
            ], 2),
            i.$slots.footer ? (c(), g("div", Fi, [
              x(i.$slots, "footer")
            ])) : S("", !0)
          ])
        ], 2)
      ], 40, Ri)
    ]));
  }
}), ji = { class: "container flex flex-wrap justify-between items-center mx-auto" }, Wi = /* @__PURE__ */ m("span", { class: "sr-only" }, "Open main menu", -1), Gi = /* @__PURE__ */ m("svg", {
  class: "w-6 h-6",
  "aria-hidden": "true",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ m("path", {
    "fill-rule": "evenodd",
    d: "M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z",
    "clip-rule": "evenodd"
  })
], -1), qi = {
  key: 0,
  class: "flex md:order-2 hidden md:flex"
}, M0 = /* @__PURE__ */ $({
  __name: "Navbar",
  props: {
    sticky: {
      type: Boolean,
      default: !1
    },
    rounded: {
      type: Boolean,
      default: !1
    },
    solid: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, r = Pe(), s = Jr(Yr).smaller("md"), o = I(!1), a = to(o), l = " border-gray-200", i = "fixed w-full z-20 top-0 left-0 border-b border-gray-200 dark:border-gray-600", f = "rounded", p = "p-3 bg-gray-50 dark:bg-gray-800 dark:border-gray-700", v = "bg-white px-2 sm:px-4 py-2.5 dark:bg-gray-900", y = u(() => P(
      l,
      t.sticky ? i : "",
      t.rounded ? f : "",
      t.solid ? p : v
    )), h = u(() => s ? o.value : !0);
    return (b, C) => (c(), g("nav", {
      class: w(d(y))
    }, [
      m("div", ji, [
        x(b.$slots, "logo"),
        m("button", {
          onClick: C[0] || (C[0] = (_) => d(a)()),
          type: "button",
          class: "inline-flex items-center p-2 ml-3 text-sm text-gray-500 rounded-lg md:hidden hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-gray-200 dark:text-gray-400 dark:hover:bg-gray-700 dark:focus:ring-gray-600",
          "aria-controls": "navbar-default",
          "aria-expanded": "false"
        }, [
          Wi,
          x(b.$slots, "menu-icon", {}, () => [
            Gi
          ])
        ]),
        x(b.$slots, "default", { isShowMenu: d(h) }),
        d(r)["right-side"] ? (c(), g("div", qi, [
          x(b.$slots, "right-side")
        ])) : S("", !0)
      ])
    ], 2));
  }
}), N0 = $({
  name: "NavbarLogo",
  props: {
    link: {
      type: String,
      default: "/"
    },
    imageUrl: {
      type: String,
      default: "/assets/logo.svg"
    },
    alt: {
      type: String,
      default: "Logo"
    },
    component: {
      type: [Object, String],
      default: "a"
    },
    linkAttr: {
      type: String,
      default: "href"
    }
  },
  emits: [
    "click"
  ],
  setup(e, { slots: t }) {
    return () => se(
      e.component,
      {
        [e.linkAttr]: e.link,
        class: "flex items-center"
      },
      {
        default: () => [
          se(
            "img",
            {
              src: e.imageUrl,
              class: "mr-3 h-6 sm:h-10",
              alt: e.alt
            }
          ),
          se(
            "span",
            {
              class: "self-center text-xl font-semibold whitespace-nowrap dark:text-white"
            },
            t
          )
        ]
      }
    );
  }
}), V0 = /* @__PURE__ */ $({
  __name: "NavbarCollapse",
  props: {
    isShowMenu: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e, n = Jr(Yr).smaller("md"), s = "w-full md:block md:w-auto", o = "flex flex-col p-4 mt-4 rounded-lg border border-gray-100 md:flex-row md:space-x-8 md:mt-0 md:text-sm md:font-medium md:border-0 dark:bg-gray-800 md:dark:bg-gray-900 dark:border-gray-700", a = "bg-gray-50", l = u(() => P(s, t.isShowMenu ? "" : "hidden")), i = u(() => P(o, n.value ? a : ""));
    return (f, p) => (c(), g("div", {
      class: w(d(l))
    }, [
      m("ul", {
        class: w(d(i))
      }, [
        x(f.$slots, "default")
      ], 2)
    ], 2));
  }
}), R0 = $({
  name: "NavbarLink",
  props: {
    link: {
      type: String,
      default: "/"
    },
    isActive: {
      type: Boolean,
      default: !1
    },
    component: {
      type: [Object, String],
      default: "a"
    },
    linkAttr: {
      type: String,
      default: "href"
    },
    disabled: {
      type: Boolean,
      default: !1
    }
  },
  emits: [
    "click"
  ],
  setup(e, { slots: t, emit: r }) {
    const n = "bg-blue-700 md:bg-transparent text-white md:text-blue-700 dark:text-white", s = "text-gray-700 hover:bg-gray-100 md:hover:bg-transparent md:border-0 md:hover:text-blue-700 dark:text-gray-400 md:dark:hover:text-white dark:hover:bg-gray-700 dark:hover:text-white md:dark:hover:bg-transparent", a = P(
      "block py-2 pr-4 pl-3 rounded md:p-0",
      e.isActive ? n : s
    ), l = (i) => {
      e.disabled || r("click", i);
    };
    return () => se("li", null, se(
      et(e.component),
      {
        ...e,
        class: a,
        [e.linkAttr]: e.link,
        onClick: l
      },
      {
        default: () => t.default && t.default()
      }
    ));
  }
}), Ui = { "aria-label": "Page navigation example" }, Ki = {
  key: 0,
  class: "text-sm text-gray-700 dark:text-gray-400 mb-2"
}, Yi = { class: "font-semibold text-gray-900 dark:text-white" }, Ji = { class: "font-semibold text-gray-900 dark:text-white" }, Qi = { class: "font-semibold text-gray-900 dark:text-white" }, Xi = { class: "inline-flex -space-x-px" }, Zi = ["disabled"], eu = {
  key: 0,
  stroke: "currentColor",
  fill: "currentColor",
  "stroke-width": "0",
  viewBox: "0 0 20 20",
  "aria-hidden": "true",
  class: "h-5 w-5",
  height: "1em",
  width: "1em",
  xmlns: "http://www.w3.org/2000/svg"
}, tu = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M12.707 5.293a1 1 0 010 1.414L9.414 10l3.293 3.293a1 1 0 01-1.414 1.414l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 0z",
  "clip-rule": "evenodd"
}, null, -1), ru = [
  tu
], nu = ["disabled", "onClick"], su = ["disabled"], ou = {
  key: 0,
  stroke: "currentColor",
  fill: "currentColor",
  "stroke-width": "0",
  viewBox: "0 0 20 20",
  "aria-hidden": "true",
  class: "h-5 w-5",
  height: "1em",
  width: "1em",
  xmlns: "http://www.w3.org/2000/svg"
}, au = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z",
  "clip-rule": "evenodd"
}, null, -1), lu = [
  au
], H0 = /* @__PURE__ */ $({
  __name: "Pagination",
  props: {
    modelValue: {
      type: Number,
      default: 1
    },
    totalPages: {
      type: Number,
      default: 1
    },
    perPage: {
      type: Number,
      default: 10
    },
    totalItems: {
      type: Number,
      required: !1
    },
    layout: {
      type: String,
      default: "pagination"
    },
    showIcons: {
      type: Boolean,
      default: !1
    },
    sliceLength: {
      type: Number,
      default: 2
    },
    previousLabel: {
      type: String,
      default: "Previous"
    },
    nextLabel: {
      type: String,
      default: "Next"
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = (b) => {
      t("update:modelValue", b);
    }, s = () => {
      t("update:modelValue", r.modelValue - 1);
    }, o = () => {
      t("update:modelValue", r.modelValue + 1);
    }, a = u(() => !r.totalItems || !r.perPage ? r.totalPages : Math.ceil(r.totalItems / r.perPage)), l = u(() => r.modelValue <= 1), i = u(() => r.modelValue >= a.value), f = (b) => b === r.modelValue, p = u(() => {
      if (r.layout === "navigation")
        return [];
      if (r.layout === "table")
        return [];
      if (a.value <= r.sliceLength * 2 + 1) {
        const _ = [];
        for (let k = 1; k <= a.value; k++)
          _.push(k);
        return _;
      }
      if (r.modelValue <= r.sliceLength) {
        const _ = [], k = Math.abs(r.modelValue - r.sliceLength) + r.modelValue + r.sliceLength + 1;
        for (let T = 1; T <= k; T++)
          _.push(T);
        return _;
      }
      if (r.modelValue >= a.value - r.sliceLength) {
        const _ = [];
        for (let k = Math.abs(a.value - r.sliceLength * 2); k <= a.value; k++)
          _.push(k);
        return _;
      }
      const b = [];
      let C = r.modelValue - r.sliceLength > 0 ? r.modelValue - r.sliceLength : 1;
      for (let _ = C; _ < r.modelValue + r.sliceLength + 1 && !(_ >= a.value); _++)
        b.push(_);
      return b;
    }), v = u(() => r.modelValue * r.perPage - r.perPage + 1), y = u(() => {
      const b = r.modelValue * r.perPage + 1;
      return r.totalItems && b > r.totalItems ? r.totalItems : b;
    }), h = u(() => r.totalItems ? r.totalItems : a.value * r.perPage);
    return (b, C) => (c(), g("nav", Ui, [
      e.layout === "table" ? (c(), g("div", Ki, [
        ne(" Showing "),
        m("span", Yi, B(d(v)), 1),
        ne(" to "),
        m("span", Ji, B(d(y)), 1),
        ne(" of "),
        m("span", Qi, B(d(h)), 1)
      ])) : S("", !0),
      m("ul", Xi, [
        m("li", null, [
          m("button", {
            disabled: d(l),
            onClick: s,
            class: "flex items-center py-2 px-3 ml-0 leading-tight text-gray-500 bg-white rounded-l-lg border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
          }, [
            e.showIcons ? (c(), g("svg", eu, ru)) : S("", !0),
            ne(" " + B(e.previousLabel), 1)
          ], 8, Zi)
        ]),
        (c(!0), g(U, null, be(d(p), (_) => (c(), g("li", { key: _ }, [
          m("button", {
            disabled: f(_),
            onClick: (k) => n(_),
            class: w(["w-12 py-2 px-3 leading-tight text-gray-500 bg-white border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white", { "text-blue-600 dark:text-white bg-blue-50 dark:bg-gray-700": _ === e.modelValue }])
          }, B(_), 11, nu)
        ]))), 128)),
        m("li", null, [
          m("button", {
            disabled: d(i),
            onClick: o,
            class: "flex items-center py-2 px-3 leading-tight text-gray-500 bg-white rounded-r-lg border border-gray-300 hover:bg-gray-100 hover:text-gray-700 dark:bg-gray-800 dark:border-gray-700 dark:text-gray-400 dark:hover:bg-gray-700 dark:hover:text-white"
          }, [
            ne(B(e.nextLabel) + " ", 1),
            e.showIcons ? (c(), g("svg", ou, lu)) : S("", !0)
          ], 8, su)
        ])
      ])
    ]));
  }
}), iu = {
  default: "bg-blue-600 dark:bg-blue-600",
  blue: "bg-blue-600 dark:bg-blue-600",
  dark: "bg-gray-600 dark:bg-gray-300",
  green: "bg-green-600 dark:bg-green-500",
  red: "bg-red-600 dark:bg-red-500",
  yellow: "bg-yellow-400",
  indigo: "bg-indigo-600 dark:bg-indigo-500",
  purple: "bg-purple-600 dark:bg-purple-500"
}, uu = {
  default: "",
  blue: "text-blue-700 dark:text-blue-500",
  dark: "dark:text-white",
  green: "text-green-700 dark:text-green-500",
  red: "text-red-700 dark:text-red-500",
  yellow: "text-yellow-700 dark:text-yellow-500",
  indigo: "text-indigo-700 dark:text-indigo-500",
  purple: "text-purple-700 dark:text-purple-500"
}, fr = {
  sm: "h-1.5 text-xs leading-none",
  md: "h-2.5 text-xs leading-none",
  lg: "h-4 text-sm leading-none",
  xl: "h-6 text-base leading-tight"
};
function du(e) {
  const t = u(() => P(
    iu[e.color.value],
    fr[e.size.value]
  )), r = u(() => P(
    fr[e.size.value]
  )), n = u(() => P(
    uu[e.color.value]
  ));
  return {
    innerClasses: t,
    outerClasses: r,
    outsideLabelClasses: n
  };
}
const cu = {
  key: 0,
  class: "flex justify-between mb-1"
}, D0 = /* @__PURE__ */ $({
  __name: "Progress",
  props: {
    color: {
      type: String,
      default: "default"
    },
    label: {
      type: String,
      default: ""
    },
    labelPosition: {
      type: String,
      default: "none"
    },
    labelProgress: {
      type: Boolean,
      default: !1
    },
    progress: {
      type: Number,
      default: 0
    },
    size: {
      type: String,
      default: "md"
    }
  },
  setup(e) {
    const t = e, { innerClasses: r, outerClasses: n, outsideLabelClasses: s } = du(N(t));
    return (o, a) => (c(), g("div", null, [
      e.label || e.labelProgress && e.labelPosition === "outside" ? (c(), g("div", cu, [
        m("span", {
          class: w(["text-base font-medium", d(s)])
        }, B(e.label), 3),
        e.labelProgress && e.labelPosition === "outside" ? (c(), g("span", {
          key: 0,
          class: w(["text-sm font-medium", d(s)])
        }, B(e.progress) + "%", 3)) : S("", !0)
      ])) : S("", !0),
      m("div", {
        class: w(["w-full bg-gray-200 rounded-full dark:bg-gray-700", d(n)])
      }, [
        m("div", {
          class: w(["rounded-full font-medium text-blue-100 text-center p-0.5", d(r)]),
          style: Le({ width: e.progress + "%" })
        }, [
          e.labelProgress && e.labelPosition === "inside" ? (c(), g(U, { key: 0 }, [
            ne(B(e.progress) + "% ", 1)
          ], 64)) : S("", !0)
        ], 6)
      ], 2)
    ]));
  }
}), fu = {
  sm: "w-5 h-5",
  md: "w-7 h-7",
  lg: "w-10 h-10"
};
function gu(e) {
  return {
    sizeClasses: u(() => {
      var r;
      return P(
        (r = fu[e.size.value]) != null ? r : ""
      );
    })
  };
}
const pu = { class: "flex items-center" }, hu = /* @__PURE__ */ m("path", { d: "M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z" }, null, -1), mu = [
  hu
], bu = /* @__PURE__ */ m("path", { d: "M9.049 2.927c.3-.921 1.603-.921 1.902 0l1.07 3.292a1 1 0 00.95.69h3.462c.969 0 1.371 1.24.588 1.81l-2.8 2.034a1 1 0 00-.364 1.118l1.07 3.292c.3.921-.755 1.688-1.54 1.118l-2.8-2.034a1 1 0 00-1.175 0l-2.8 2.034c-.784.57-1.838-.197-1.539-1.118l1.07-3.292a1 1 0 00-.364-1.118L2.98 8.72c-.783-.57-.38-1.81.588-1.81h3.461a1 1 0 00.951-.69l1.07-3.292z" }, null, -1), vu = [
  bu
], yu = /* @__PURE__ */ m("span", { class: "w-1 h-1 mx-1.5 bg-gray-500 rounded-full dark:bg-gray-400" }, null, -1), wu = ["href"], F0 = /* @__PURE__ */ $({
  __name: "Rating",
  props: {
    size: {
      type: String,
      default: "sm"
    },
    rating: {
      type: Number,
      default: 3
    },
    scale: {
      type: Number,
      default: 5
    },
    reviewText: {
      type: String,
      default: ""
    },
    reviewLink: {
      type: String,
      default: ""
    }
  },
  setup(e) {
    const t = e, r = u(() => Math.floor(t.rating)), n = u(() => t.scale - r.value), { sizeClasses: s } = gu(N(t));
    return (o, a) => (c(), g("div", pu, [
      (c(!0), g(U, null, be(d(r), (l) => (c(), g("svg", {
        key: l,
        class: w(["text-yellow-400", d(s)]),
        fill: "currentColor",
        viewBox: "0 0 20 20",
        xmlns: "http://www.w3.org/2000/svg"
      }, mu, 2))), 128)),
      (c(!0), g(U, null, be(d(n), (l) => (c(), g("svg", {
        key: l,
        class: w(["text-gray-300 dark:text-gray-500", d(s)]),
        fill: "currentColor",
        viewBox: "0 0 20 20",
        xmlns: "http://www.w3.org/2000/svg"
      }, vu, 2))), 128)),
      x(o.$slots, "besideText"),
      e.reviewText && e.reviewLink ? (c(), g(U, { key: 0 }, [
        yu,
        m("a", {
          href: e.reviewLink,
          class: "text-sm font-medium text-gray-900 underline hover:no-underline dark:text-white"
        }, B(e.reviewText), 9, wu)
      ], 64)) : S("", !0)
    ]));
  }
}), xu = {
  class: "w-64",
  "aria-label": "Sidebar"
}, ku = /* @__PURE__ */ Mr('<div class="overflow-y-auto py-4 px-3 bg-gray-50 rounded dark:bg-gray-800"><ul class="space-y-2"><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M2 10a8 8 0 018-8v8h8a8 8 0 11-16 0z"></path><path d="M12 2.252A8.014 8.014 0 0117.748 8H12V2.252z"></path></svg><span class="ml-3">Dashboard</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M5 3a2 2 0 00-2 2v2a2 2 0 002 2h2a2 2 0 002-2V5a2 2 0 00-2-2H5zM5 11a2 2 0 00-2 2v2a2 2 0 002 2h2a2 2 0 002-2v-2a2 2 0 00-2-2H5zM11 5a2 2 0 012-2h2a2 2 0 012 2v2a2 2 0 01-2 2h-2a2 2 0 01-2-2V5zM11 13a2 2 0 012-2h2a2 2 0 012 2v2a2 2 0 01-2 2h-2a2 2 0 01-2-2v-2z"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Kanban</span><span class="inline-flex justify-center items-center px-2 ml-3 text-sm font-medium text-gray-800 bg-gray-200 rounded-full dark:bg-gray-700 dark:text-gray-300">Pro</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="M8.707 7.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l2-2a1 1 0 00-1.414-1.414L11 7.586V3a1 1 0 10-2 0v4.586l-.293-.293z"></path><path d="M3 5a2 2 0 012-2h1a1 1 0 010 2H5v7h2l1 2h4l1-2h2V5h-1a1 1 0 110-2h1a2 2 0 012 2v10a2 2 0 01-2 2H5a2 2 0 01-2-2V5z"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Inbox</span><span class="inline-flex justify-center items-center p-3 ml-3 w-3 h-3 text-sm font-medium text-blue-600 bg-blue-200 rounded-full dark:bg-blue-900 dark:text-blue-200">3</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z" clip-rule="evenodd"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Users</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M10 2a4 4 0 00-4 4v1H5a1 1 0 00-.994.89l-1 9A1 1 0 004 18h12a1 1 0 00.994-1.11l-1-9A1 1 0 0015 7h-1V6a4 4 0 00-4-4zm2 5V6a2 2 0 10-4 0v1h4zm-6 3a1 1 0 112 0 1 1 0 01-2 0zm7-1a1 1 0 100 2 1 1 0 000-2z" clip-rule="evenodd"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Products</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M3 3a1 1 0 00-1 1v12a1 1 0 102 0V4a1 1 0 00-1-1zm10.293 9.293a1 1 0 001.414 1.414l3-3a1 1 0 000-1.414l-3-3a1 1 0 10-1.414 1.414L14.586 9H7a1 1 0 100 2h7.586l-1.293 1.293z" clip-rule="evenodd"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Sign In</span></a></li><li><a href="#" class="flex items-center p-2 text-base font-normal text-gray-900 rounded-lg dark:text-white hover:bg-gray-100 dark:hover:bg-gray-700"><svg class="flex-shrink-0 w-6 h-6 text-gray-500 transition duration-75 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-white" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M5 4a3 3 0 00-3 3v6a3 3 0 003 3h10a3 3 0 003-3V7a3 3 0 00-3-3H5zm-1 9v-1h5v2H5a1 1 0 01-1-1zm7 1h4a1 1 0 001-1v-1h-5v2zm0-4h5V8h-5v2zM9 8H4v2h5V8z" clip-rule="evenodd"></path></svg><span class="flex-1 ml-3 whitespace-nowrap">Sign Up</span></a></li></ul></div>', 1), _u = [
  ku
], j0 = /* @__PURE__ */ $({
  __name: "Sidebar",
  props: {
    children: {
      type: Array,
      default() {
        return [];
      }
    },
    collapseBehavior: {
      type: String,
      default: "collapse"
    },
    collapsed: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    return (t, r) => (c(), g("aside", xu, _u));
  }
}), Cu = { class: "relative overflow-x-auto shadow-md sm:rounded-lg" }, $u = { class: "w-full text-sm text-left text-gray-500 dark:text-gray-400" }, W0 = /* @__PURE__ */ $({
  __name: "Table",
  props: {
    striped: {
      type: Boolean,
      default: !1
    },
    stripedColumns: {
      type: Boolean,
      default: !1
    },
    hoverable: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e;
    return X("striped", t.striped), X("hoverable", t.hoverable), X("stripedColumns", t.stripedColumns), (r, n) => (c(), g("div", Cu, [
      m("table", $u, [
        x(r.$slots, "default")
      ])
    ]));
  }
}), Su = {}, Tu = { class: "text-xs text-gray-700 uppercase bg-gray-50 dark:bg-gray-700 dark:text-gray-400" };
function Pu(e, t) {
  return c(), g("thead", Tu, [
    m("tr", null, [
      x(e.$slots, "default")
    ])
  ]);
}
const G0 = /* @__PURE__ */ oe(Su, [["render", Pu]]), zu = {};
function Ou(e, t) {
  return c(), g("tbody", null, [
    x(e.$slots, "default")
  ]);
}
const q0 = /* @__PURE__ */ oe(zu, [["render", Ou]]), Au = "px-6 py-3 text-xs uppercase", Iu = "even:bg-white even:dark:bg-gray-900 odd:dark:bg-gray-800 odd:bg-gray-50";
function Bu() {
  const e = W("stripedColumns");
  return {
    tableHeadCellClasses: u(() => P(Au, {
      [Iu]: e
    }))
  };
}
const U0 = /* @__PURE__ */ $({
  __name: "TableHeadCell",
  setup(e) {
    const { tableHeadCellClasses: t } = Bu();
    return (r, n) => (c(), g("th", {
      scope: "col",
      class: w(d(t))
    }, [
      x(r.$slots, "default")
    ], 2));
  }
}), Eu = "bg-white dark:bg-gray-800 [&:not(:last-child)]:border-b [&:not(:last-child)]:dark:border-gray-700", Lu = "odd:bg-white even:bg-gray-50 odd:dark:bg-gray-800 even:dark:bg-gray-700", Mu = "hover:bg-gray-50 dark:hover:bg-gray-600";
function Nu() {
  const e = W("striped"), t = W("hoverable");
  return {
    tableRowClasses: u(() => P(Eu, {
      [Lu]: e,
      [Mu]: t
    }))
  };
}
const K0 = /* @__PURE__ */ $({
  __name: "TableRow",
  setup(e) {
    const { tableRowClasses: t } = Nu();
    return (r, n) => (c(), g("tr", {
      class: w(d(t))
    }, [
      x(r.$slots, "default")
    ], 2));
  }
}), Vu = "px-6 py-4 first:font-medium first:text-gray-900 first:dark:text-white first:whitespace-nowrap last:text-right", Ru = "even:bg-gray-white even:dark:bg-gray-900 odd:dark:bg-gray-800 odd:bg-gray-50";
function Hu() {
  const e = W("stripedColumns");
  return {
    tableCellClasses: u(() => P(Vu, {
      [Ru]: e
    }))
  };
}
const Y0 = /* @__PURE__ */ $({
  __name: "TableCell",
  setup(e) {
    const { tableCellClasses: t } = Hu();
    return (r, n) => (c(), g("td", {
      class: w(d(t))
    }, [
      x(r.$slots, "default")
    ], 2));
  }
}), J0 = /* @__PURE__ */ $({
  __name: "Timeline",
  props: {
    horizontal: {
      type: Boolean,
      default: !1
    }
  },
  setup(e) {
    const t = e;
    X("horizontal", t.horizontal);
    const r = "relative border-gray-200 dark:border-gray-700", n = "border-l", s = "flex", o = u(() => P(r, t.horizontal ? s : n));
    return (a, l) => (c(), g("div", null, [
      m("ol", Vr({ class: d(o) }, a.$attrs), [
        x(a.$slots, "default")
      ], 16)
    ]));
  }
}), Q0 = /* @__PURE__ */ $({
  __name: "TimelineItem",
  setup(e) {
    const t = W("horizontal"), r = "mb-10", n = "mb-6 sm:mb-0 relative", s = "ml-6", o = u(() => P(r, t ? n : s));
    return (a, l) => (c(), g("li", {
      class: w(d(o))
    }, [
      x(a.$slots, "default")
    ], 2));
  }
}), X0 = /* @__PURE__ */ $({
  __name: "TimelinePoint",
  setup(e) {
    const t = Pe(), r = u(() => !!t.default), n = W("horizontal"), s = u(() => P(n ? "flex items-center" : "")), o = "h-0.5 w-full bg-gray-200 dark:bg-gray-700 sm:flex", a = u(() => P(o, { "sm:hidden hidden": !n })), l = u(() => {
      const i = "absolute rounded-full -left-1.5 border border-white dark:border-gray-900 dark:bg-gray-700", f = "mt-1.5 w-3 h-3 bg-gray-200", p = "mt-1.5 w-6 h-6 -left-3 flex justify-center items-center bg-blue-200 ring-8 ring-white dark:ring-gray-900", v = "w-3 h-3 bg-gray-200", y = "w-6 h-6 -left-3 flex justify-center items-center bg-blue-200 ring-8 ring-white dark:ring-gray-900", h = !n && !r.value, b = !n && r.value, C = n && !r.value, _ = n && r.value;
      return P(
        i,
        {
          [f]: h,
          [p]: b,
          [v]: C,
          [y]: _
        }
      );
    });
    return (i, f) => (c(), g("div", {
      class: w(d(s))
    }, [
      m("div", {
        class: w(d(l))
      }, [
        x(i.$slots, "default")
      ], 2),
      m("div", {
        class: w(d(a))
      }, null, 2)
    ], 2));
  }
}), Z0 = /* @__PURE__ */ $({
  __name: "TimelineContent",
  setup(e) {
    const t = W("horizontal"), r = u(() => P(t ? "mt-3 sm:pr-8" : ""));
    return (n, s) => (c(), g("div", {
      class: w(d(r))
    }, [
      x(n.$slots, "default")
    ], 2));
  }
}), Du = {}, Fu = { class: "mb-1 text-sm font-normal leading-none text-gray-400 dark:text-gray-500" };
function ju(e, t) {
  return c(), g("time", Fu, [
    x(e.$slots, "default")
  ]);
}
const ef = /* @__PURE__ */ oe(Du, [["render", ju]]), Wu = {}, Gu = { class: "text-lg font-semibold text-gray-900 dark:text-white" };
function qu(e, t) {
  return c(), g("h3", Gu, [
    x(e.$slots, "default")
  ]);
}
const tf = /* @__PURE__ */ oe(Wu, [["render", qu]]), Uu = {}, Ku = { class: "mb-4 text-base font-normal text-gray-500 dark:text-gray-400" };
function Yu(e, t) {
  return c(), g("p", Ku, [
    x(e.$slots, "default")
  ]);
}
const rf = /* @__PURE__ */ oe(Uu, [["render", Yu]]), Ju = {
  danger: "text-red-500 bg-red-100 dark:bg-red-800 dark:text-red-200",
  empty: "",
  success: "text-green-500 bg-green-100 dark:bg-green-800 dark:text-green-200",
  warning: "text-orange-500 bg-orange-100 dark:bg-orange-700 dark:text-orange-200"
}, Qu = {
  center: "items-center",
  end: "items-end",
  start: "items-start"
}, gr = "flex w-full max-w-xs p-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800", pr = "text-sm font-normal";
function Xu(e) {
  const t = u(() => Ju[e.type.value]), r = u(() => {
    const s = Qu[e.alignment.value];
    return e.divide.value ? j(gr, "dark:divide-gray-700 divide-x divide-gray-200", s) : j(gr, s);
  }), n = u(() => e.type.value !== "empty" && e.divide.value ? j(pr, "pl-3") : pr);
  return {
    typeClasses: t,
    wrapperClasses: r,
    contentClasses: n
  };
}
const Zu = {
  key: 1,
  "aria-hidden": "true",
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, ed = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z",
  "clip-rule": "evenodd"
}, null, -1), td = [
  ed
], rd = {
  key: 2,
  "aria-hidden": "true",
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, nd = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
  "clip-rule": "evenodd"
}, null, -1), sd = [
  nd
], od = {
  key: 3,
  "aria-hidden": "true",
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, ad = /* @__PURE__ */ m("path", {
  "fill-rule": "evenodd",
  d: "M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z",
  "clip-rule": "evenodd"
}, null, -1), ld = [
  ad
], id = /* @__PURE__ */ m("span", { class: "sr-only" }, "Close", -1), ud = /* @__PURE__ */ m("svg", {
  class: "w-5 h-5",
  fill: "currentColor",
  viewBox: "0 0 20 20",
  xmlns: "http://www.w3.org/2000/svg"
}, [
  /* @__PURE__ */ m("path", {
    "fill-rule": "evenodd",
    d: "M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z",
    "clip-rule": "evenodd"
  })
], -1), dd = [
  id,
  ud
], hr = /* @__PURE__ */ $({
  __name: "Toast",
  props: {
    type: {
      type: String,
      default: "empty"
    },
    alignment: {
      type: String,
      default: "center"
    },
    closable: {
      type: Boolean,
      default: !1
    },
    divide: {
      type: Boolean,
      default: !1
    }
  },
  emits: ["close"],
  setup(e, { emit: t }) {
    const r = e, n = I(!0), { typeClasses: s, wrapperClasses: o, contentClasses: a } = Xu(N(r)), l = () => {
      t("close"), n.value = !1;
    };
    return (i, f) => n.value ? (c(), g("div", {
      key: 0,
      id: "toast-default",
      class: w(d(o)),
      role: "alert"
    }, [
      e.type !== "empty" || i.$slots.icon ? (c(), D(Wr, {
        key: 0,
        class: w([
          "inline-flex flex-shrink-0 justify-center items-center w-8 h-8 rounded-lg",
          d(s)
        ]),
        apply: ["background", "text"]
      }, {
        default: H(() => [
          i.$slots.icon ? x(i.$slots, "icon", {
            key: 0,
            class: w({ "ml-3": e.type !== "empty" })
          }) : e.type === "success" ? (c(), g("svg", Zu, td)) : e.type === "danger" ? (c(), g("svg", rd, sd)) : e.type === "warning" ? (c(), g("svg", od, ld)) : S("", !0)
        ]),
        _: 3
      }, 8, ["class"])) : S("", !0),
      m("div", {
        class: w([d(a), { "ml-3": i.$slots.icon || e.type !== "empty" }])
      }, [
        x(i.$slots, "default")
      ], 2),
      e.closable ? (c(), g("button", {
        key: 1,
        onClick: l,
        type: "button",
        class: "border-none ml-auto -mx-1.5 -my-1.5 bg-white text-gray-400 hover:text-gray-900 rounded-lg focus:ring-2 focus:ring-gray-300 p-1.5 hover:bg-gray-100 inline-flex h-8 w-8 dark:text-gray-500 dark:hover:text-white dark:bg-gray-800 dark:hover:bg-gray-700",
        "aria-label": "Close"
      }, dd)) : S("", !0)
    ], 2)) : S("", !0);
  }
}), sn = "flowbite-toast-injection-key", cd = $({
  components: {
    Toast: hr
  },
  props: {
    transition: {
      type: String,
      default: "slide-left"
    }
  },
  setup() {
    const e = I([]), t = (o, a) => {
      eo(() => s(o), a);
    }, r = (o) => {
      const a = parseInt((new Date().getTime() * Math.random()).toString()).toString();
      return e.value.push({
        id: a,
        ...o
      }), o.time > 0 && t(a, o.time), a;
    }, n = () => {
      if (e.value.length === 0)
        return "";
      const o = e.value[e.value.length - 1].id;
      return e.value.pop(), o;
    }, s = (o) => {
      const a = e.value.findIndex((l) => l.id === o);
      return a >= 0 && e.value.splice(a, 1), a >= 0;
    };
    return X(sn, {
      add: r,
      pop: n,
      remove: s
    }), {
      toasts: e,
      removeToast: s
    };
  },
  render() {
    const {
      $props: e,
      $slots: t,
      toasts: r,
      removeToast: n
    } = this;
    return se("div", {}, [
      t.default ? t.default() : null,
      se(
        Bn,
        {
          name: e.transition,
          tag: "div",
          class: "xl:w-1/6 md:w-1/4 sm:w-1/4 fixed top-3 right-3 flex flex-col gap-2 z-50"
        },
        {
          default: () => r.map(
            (s) => s.component ? se(s.component, {
              key: s.id,
              onClose: () => n(s.id),
              ...s.componentProps ? s.componentProps : {}
            }, () => s.text) : se(hr, {
              closable: !0,
              type: s.type,
              key: s.id,
              onClose: () => n(s.id)
            }, () => s.text)
          )
        }
      )
    ]);
  }
});
const nf = /* @__PURE__ */ oe(cd, [["__scopeId", "data-v-3eb025d2"]]);
function Q(e) {
  return e.split("-")[1];
}
function Ht(e) {
  return e === "y" ? "height" : "width";
}
function Z(e) {
  return e.split("-")[0];
}
function Ae(e) {
  return ["top", "bottom"].includes(Z(e)) ? "x" : "y";
}
function mr(e, t, r) {
  let {
    reference: n,
    floating: s
  } = e;
  const o = n.x + n.width / 2 - s.width / 2, a = n.y + n.height / 2 - s.height / 2, l = Ae(t), i = Ht(l), f = n[i] / 2 - s[i] / 2, p = Z(t), v = l === "x";
  let y;
  switch (p) {
    case "top":
      y = {
        x: o,
        y: n.y - s.height
      };
      break;
    case "bottom":
      y = {
        x: o,
        y: n.y + n.height
      };
      break;
    case "right":
      y = {
        x: n.x + n.width,
        y: a
      };
      break;
    case "left":
      y = {
        x: n.x - s.width,
        y: a
      };
      break;
    default:
      y = {
        x: n.x,
        y: n.y
      };
  }
  switch (Q(t)) {
    case "start":
      y[l] -= f * (r && v ? -1 : 1);
      break;
    case "end":
      y[l] += f * (r && v ? -1 : 1);
      break;
  }
  return y;
}
const fd = async (e, t, r) => {
  const {
    placement: n = "bottom",
    strategy: s = "absolute",
    middleware: o = [],
    platform: a
  } = r, l = o.filter(Boolean), i = await (a.isRTL == null ? void 0 : a.isRTL(t));
  let f = await a.getElementRects({
    reference: e,
    floating: t,
    strategy: s
  }), {
    x: p,
    y: v
  } = mr(f, n, i), y = n, h = {}, b = 0;
  for (let C = 0; C < l.length; C++) {
    const {
      name: _,
      fn: k
    } = l[C], {
      x: T,
      y: O,
      data: L,
      reset: A
    } = await k({
      x: p,
      y: v,
      initialPlacement: n,
      placement: y,
      strategy: s,
      middlewareData: h,
      rects: f,
      platform: a,
      elements: {
        reference: e,
        floating: t
      }
    });
    if (p = T != null ? T : p, v = O != null ? O : v, h = {
      ...h,
      [_]: {
        ...h[_],
        ...L
      }
    }, A && b <= 50) {
      b++, typeof A == "object" && (A.placement && (y = A.placement), A.rects && (f = A.rects === !0 ? await a.getElementRects({
        reference: e,
        floating: t,
        strategy: s
      }) : A.rects), {
        x: p,
        y: v
      } = mr(f, y, i)), C = -1;
      continue;
    }
  }
  return {
    x: p,
    y: v,
    placement: y,
    strategy: s,
    middlewareData: h
  };
};
function ke(e, t) {
  return typeof e == "function" ? e(t) : e;
}
function gd(e) {
  return {
    top: 0,
    right: 0,
    bottom: 0,
    left: 0,
    ...e
  };
}
function on(e) {
  return typeof e != "number" ? gd(e) : {
    top: e,
    right: e,
    bottom: e,
    left: e
  };
}
function Me(e) {
  return {
    ...e,
    top: e.y,
    left: e.x,
    right: e.x + e.width,
    bottom: e.y + e.height
  };
}
async function pt(e, t) {
  var r;
  t === void 0 && (t = {});
  const {
    x: n,
    y: s,
    platform: o,
    rects: a,
    elements: l,
    strategy: i
  } = e, {
    boundary: f = "clippingAncestors",
    rootBoundary: p = "viewport",
    elementContext: v = "floating",
    altBoundary: y = !1,
    padding: h = 0
  } = ke(t, e), b = on(h), _ = l[y ? v === "floating" ? "reference" : "floating" : v], k = Me(await o.getClippingRect({
    element: (r = await (o.isElement == null ? void 0 : o.isElement(_))) == null || r ? _ : _.contextElement || await (o.getDocumentElement == null ? void 0 : o.getDocumentElement(l.floating)),
    boundary: f,
    rootBoundary: p,
    strategy: i
  })), T = v === "floating" ? {
    ...a.floating,
    x: n,
    y: s
  } : a.reference, O = await (o.getOffsetParent == null ? void 0 : o.getOffsetParent(l.floating)), L = await (o.isElement == null ? void 0 : o.isElement(O)) ? await (o.getScale == null ? void 0 : o.getScale(O)) || {
    x: 1,
    y: 1
  } : {
    x: 1,
    y: 1
  }, A = Me(o.convertOffsetParentRelativeRectToViewportRelativeRect ? await o.convertOffsetParentRelativeRectToViewportRelativeRect({
    rect: T,
    offsetParent: O,
    strategy: i
  }) : T);
  return {
    top: (k.top - A.top + b.top) / L.y,
    bottom: (A.bottom - k.bottom + b.bottom) / L.y,
    left: (k.left - A.left + b.left) / L.x,
    right: (A.right - k.right + b.right) / L.x
  };
}
const Fe = Math.min, he = Math.max;
function St(e, t, r) {
  return he(e, Fe(t, r));
}
const pd = (e) => ({
  name: "arrow",
  options: e,
  async fn(t) {
    const {
      x: r,
      y: n,
      placement: s,
      rects: o,
      platform: a,
      elements: l
    } = t, {
      element: i,
      padding: f = 0
    } = ke(e, t) || {};
    if (i == null)
      return {};
    const p = on(f), v = {
      x: r,
      y: n
    }, y = Ae(s), h = Ht(y), b = await a.getDimensions(i), C = y === "y", _ = C ? "top" : "left", k = C ? "bottom" : "right", T = C ? "clientHeight" : "clientWidth", O = o.reference[h] + o.reference[y] - v[y] - o.floating[h], L = v[y] - o.reference[y], A = await (a.getOffsetParent == null ? void 0 : a.getOffsetParent(i));
    let V = A ? A[T] : 0;
    (!V || !await (a.isElement == null ? void 0 : a.isElement(A))) && (V = l.floating[T] || o.floating[h]);
    const M = O / 2 - L / 2, K = V / 2 - b[h] / 2 - 1, re = Fe(p[_], K), Y = Fe(p[k], K), E = re, F = V - b[h] - Y, R = V / 2 - b[h] / 2 + M, G = St(E, R, F), ae = Q(s) != null && R != G && o.reference[h] / 2 - (R < E ? re : Y) - b[h] / 2 < 0 ? R < E ? E - R : F - R : 0;
    return {
      [y]: v[y] - ae,
      data: {
        [y]: G,
        centerOffset: R - G + ae
      }
    };
  }
}), hd = ["top", "right", "bottom", "left"], br = /* @__PURE__ */ hd.reduce((e, t) => e.concat(t, t + "-start", t + "-end"), []), md = {
  left: "right",
  right: "left",
  bottom: "top",
  top: "bottom"
};
function ot(e) {
  return e.replace(/left|right|bottom|top/g, (t) => md[t]);
}
function an(e, t, r) {
  r === void 0 && (r = !1);
  const n = Q(e), s = Ae(e), o = Ht(s);
  let a = s === "x" ? n === (r ? "end" : "start") ? "right" : "left" : n === "start" ? "bottom" : "top";
  return t.reference[o] > t.floating[o] && (a = ot(a)), {
    main: a,
    cross: ot(a)
  };
}
const bd = {
  start: "end",
  end: "start"
};
function at(e) {
  return e.replace(/start|end/g, (t) => bd[t]);
}
function vd(e, t, r) {
  return (e ? [...r.filter((s) => Q(s) === e), ...r.filter((s) => Q(s) !== e)] : r.filter((s) => Z(s) === s)).filter((s) => e ? Q(s) === e || (t ? at(s) !== s : !1) : !0);
}
const yd = function(e) {
  return e === void 0 && (e = {}), {
    name: "autoPlacement",
    options: e,
    async fn(t) {
      var r, n, s;
      const {
        rects: o,
        middlewareData: a,
        placement: l,
        platform: i,
        elements: f
      } = t, {
        crossAxis: p = !1,
        alignment: v,
        allowedPlacements: y = br,
        autoAlignment: h = !0,
        ...b
      } = ke(e, t), C = v !== void 0 || y === br ? vd(v || null, h, y) : y, _ = await pt(t, b), k = ((r = a.autoPlacement) == null ? void 0 : r.index) || 0, T = C[k];
      if (T == null)
        return {};
      const {
        main: O,
        cross: L
      } = an(T, o, await (i.isRTL == null ? void 0 : i.isRTL(f.floating)));
      if (l !== T)
        return {
          reset: {
            placement: C[0]
          }
        };
      const A = [_[Z(T)], _[O], _[L]], V = [...((n = a.autoPlacement) == null ? void 0 : n.overflows) || [], {
        placement: T,
        overflows: A
      }], M = C[k + 1];
      if (M)
        return {
          data: {
            index: k + 1,
            overflows: V
          },
          reset: {
            placement: M
          }
        };
      const K = V.map((E) => {
        const F = Q(E.placement);
        return [E.placement, F && p ? E.overflows.slice(0, 2).reduce((R, G) => R + G, 0) : E.overflows[0], E.overflows];
      }).sort((E, F) => E[1] - F[1]), Y = ((s = K.filter((E) => E[2].slice(
        0,
        Q(E[0]) ? 2 : 3
      ).every((F) => F <= 0))[0]) == null ? void 0 : s[0]) || K[0][0];
      return Y !== l ? {
        data: {
          index: k + 1,
          overflows: V
        },
        reset: {
          placement: Y
        }
      } : {};
    }
  };
};
function wd(e) {
  const t = ot(e);
  return [at(e), t, at(t)];
}
function xd(e, t, r) {
  const n = ["left", "right"], s = ["right", "left"], o = ["top", "bottom"], a = ["bottom", "top"];
  switch (e) {
    case "top":
    case "bottom":
      return r ? t ? s : n : t ? n : s;
    case "left":
    case "right":
      return t ? o : a;
    default:
      return [];
  }
}
function kd(e, t, r, n) {
  const s = Q(e);
  let o = xd(Z(e), r === "start", n);
  return s && (o = o.map((a) => a + "-" + s), t && (o = o.concat(o.map(at)))), o;
}
const _d = function(e) {
  return e === void 0 && (e = {}), {
    name: "flip",
    options: e,
    async fn(t) {
      var r;
      const {
        placement: n,
        middlewareData: s,
        rects: o,
        initialPlacement: a,
        platform: l,
        elements: i
      } = t, {
        mainAxis: f = !0,
        crossAxis: p = !0,
        fallbackPlacements: v,
        fallbackStrategy: y = "bestFit",
        fallbackAxisSideDirection: h = "none",
        flipAlignment: b = !0,
        ...C
      } = ke(e, t), _ = Z(n), k = Z(a) === a, T = await (l.isRTL == null ? void 0 : l.isRTL(i.floating)), O = v || (k || !b ? [ot(a)] : wd(a));
      !v && h !== "none" && O.push(...kd(a, b, h, T));
      const L = [a, ...O], A = await pt(t, C), V = [];
      let M = ((r = s.flip) == null ? void 0 : r.overflows) || [];
      if (f && V.push(A[_]), p) {
        const {
          main: E,
          cross: F
        } = an(n, o, T);
        V.push(A[E], A[F]);
      }
      if (M = [...M, {
        placement: n,
        overflows: V
      }], !V.every((E) => E <= 0)) {
        var K, re;
        const E = (((K = s.flip) == null ? void 0 : K.index) || 0) + 1, F = L[E];
        if (F)
          return {
            data: {
              index: E,
              overflows: M
            },
            reset: {
              placement: F
            }
          };
        let R = (re = M.filter((G) => G.overflows[0] <= 0).sort((G, ge) => G.overflows[1] - ge.overflows[1])[0]) == null ? void 0 : re.placement;
        if (!R)
          switch (y) {
            case "bestFit": {
              var Y;
              const G = (Y = M.map((ge) => [ge.placement, ge.overflows.filter((ae) => ae > 0).reduce((ae, kn) => ae + kn, 0)]).sort((ge, ae) => ge[1] - ae[1])[0]) == null ? void 0 : Y[0];
              G && (R = G);
              break;
            }
            case "initialPlacement":
              R = a;
              break;
          }
        if (n !== R)
          return {
            reset: {
              placement: R
            }
          };
      }
      return {};
    }
  };
};
async function Cd(e, t) {
  const {
    placement: r,
    platform: n,
    elements: s
  } = e, o = await (n.isRTL == null ? void 0 : n.isRTL(s.floating)), a = Z(r), l = Q(r), i = Ae(r) === "x", f = ["left", "top"].includes(a) ? -1 : 1, p = o && i ? -1 : 1, v = ke(t, e);
  let {
    mainAxis: y,
    crossAxis: h,
    alignmentAxis: b
  } = typeof v == "number" ? {
    mainAxis: v,
    crossAxis: 0,
    alignmentAxis: null
  } : {
    mainAxis: 0,
    crossAxis: 0,
    alignmentAxis: null,
    ...v
  };
  return l && typeof b == "number" && (h = l === "end" ? b * -1 : b), i ? {
    x: h * p,
    y: y * f
  } : {
    x: y * f,
    y: h * p
  };
}
const $d = function(e) {
  return e === void 0 && (e = 0), {
    name: "offset",
    options: e,
    async fn(t) {
      const {
        x: r,
        y: n
      } = t, s = await Cd(t, e);
      return {
        x: r + s.x,
        y: n + s.y,
        data: s
      };
    }
  };
};
function Sd(e) {
  return e === "x" ? "y" : "x";
}
const Td = function(e) {
  return e === void 0 && (e = {}), {
    name: "shift",
    options: e,
    async fn(t) {
      const {
        x: r,
        y: n,
        placement: s
      } = t, {
        mainAxis: o = !0,
        crossAxis: a = !1,
        limiter: l = {
          fn: (_) => {
            let {
              x: k,
              y: T
            } = _;
            return {
              x: k,
              y: T
            };
          }
        },
        ...i
      } = ke(e, t), f = {
        x: r,
        y: n
      }, p = await pt(t, i), v = Ae(Z(s)), y = Sd(v);
      let h = f[v], b = f[y];
      if (o) {
        const _ = v === "y" ? "top" : "left", k = v === "y" ? "bottom" : "right", T = h + p[_], O = h - p[k];
        h = St(T, h, O);
      }
      if (a) {
        const _ = y === "y" ? "top" : "left", k = y === "y" ? "bottom" : "right", T = b + p[_], O = b - p[k];
        b = St(T, b, O);
      }
      const C = l.fn({
        ...t,
        [v]: h,
        [y]: b
      });
      return {
        ...C,
        data: {
          x: C.x - r,
          y: C.y - n
        }
      };
    }
  };
}, Pd = function(e) {
  return e === void 0 && (e = {}), {
    name: "size",
    options: e,
    async fn(t) {
      const {
        placement: r,
        rects: n,
        platform: s,
        elements: o
      } = t, {
        apply: a = () => {
        },
        ...l
      } = ke(e, t), i = await pt(t, l), f = Z(r), p = Q(r), y = Ae(r) === "x", {
        width: h,
        height: b
      } = n.floating;
      let C, _;
      f === "top" || f === "bottom" ? (C = f, _ = p === (await (s.isRTL == null ? void 0 : s.isRTL(o.floating)) ? "start" : "end") ? "left" : "right") : (_ = f, C = p === "end" ? "top" : "bottom");
      const k = b - i[C], T = h - i[_], O = !t.middlewareData.shift;
      let L = k, A = T;
      if (y) {
        const M = h - i.left - i.right;
        A = p || O ? Fe(T, M) : M;
      } else {
        const M = b - i.top - i.bottom;
        L = p || O ? Fe(k, M) : M;
      }
      if (O && !p) {
        const M = he(i.left, 0), K = he(i.right, 0), re = he(i.top, 0), Y = he(i.bottom, 0);
        y ? A = h - 2 * (M !== 0 || K !== 0 ? M + K : he(i.left, i.right)) : L = b - 2 * (re !== 0 || Y !== 0 ? re + Y : he(i.top, i.bottom));
      }
      await a({
        ...t,
        availableWidth: A,
        availableHeight: L
      });
      const V = await s.getDimensions(o.floating);
      return h !== V.width || b !== V.height ? {
        reset: {
          rects: !0
        }
      } : {};
    }
  };
};
function q(e) {
  var t;
  return ((t = e.ownerDocument) == null ? void 0 : t.defaultView) || window;
}
function ee(e) {
  return q(e).getComputedStyle(e);
}
const vr = Math.min, Ne = Math.max, lt = Math.round;
function ln(e) {
  const t = ee(e);
  let r = parseFloat(t.width), n = parseFloat(t.height);
  const s = e.offsetWidth, o = e.offsetHeight, a = lt(r) !== s || lt(n) !== o;
  return a && (r = s, n = o), {
    width: r,
    height: n,
    fallback: a
  };
}
function de(e) {
  return dn(e) ? (e.nodeName || "").toLowerCase() : "";
}
let Ye;
function un() {
  if (Ye)
    return Ye;
  const e = navigator.userAgentData;
  return e && Array.isArray(e.brands) ? (Ye = e.brands.map((t) => t.brand + "/" + t.version).join(" "), Ye) : navigator.userAgent;
}
function te(e) {
  return e instanceof q(e).HTMLElement;
}
function ce(e) {
  return e instanceof q(e).Element;
}
function dn(e) {
  return e instanceof q(e).Node;
}
function yr(e) {
  if (typeof ShadowRoot > "u")
    return !1;
  const t = q(e).ShadowRoot;
  return e instanceof t || e instanceof ShadowRoot;
}
function ht(e) {
  const {
    overflow: t,
    overflowX: r,
    overflowY: n,
    display: s
  } = ee(e);
  return /auto|scroll|overlay|hidden|clip/.test(t + n + r) && !["inline", "contents"].includes(s);
}
function zd(e) {
  return ["table", "td", "th"].includes(de(e));
}
function Dt(e) {
  const t = /firefox/i.test(un()), r = ee(e), n = r.backdropFilter || r.WebkitBackdropFilter;
  return r.transform !== "none" || r.perspective !== "none" || (n ? n !== "none" : !1) || t && r.willChange === "filter" || t && (r.filter ? r.filter !== "none" : !1) || ["transform", "perspective"].some((s) => r.willChange.includes(s)) || ["paint", "layout", "strict", "content"].some((s) => {
    const o = r.contain;
    return o != null ? o.includes(s) : !1;
  });
}
function cn() {
  return !/^((?!chrome|android).)*safari/i.test(un());
}
function Ft(e) {
  return ["html", "body", "#document"].includes(de(e));
}
function fn(e) {
  return ce(e) ? e : e.contextElement;
}
const gn = {
  x: 1,
  y: 1
};
function $e(e) {
  const t = fn(e);
  if (!te(t))
    return gn;
  const r = t.getBoundingClientRect(), {
    width: n,
    height: s,
    fallback: o
  } = ln(t);
  let a = (o ? lt(r.width) : r.width) / n, l = (o ? lt(r.height) : r.height) / s;
  return (!a || !Number.isFinite(a)) && (a = 1), (!l || !Number.isFinite(l)) && (l = 1), {
    x: a,
    y: l
  };
}
function je(e, t, r, n) {
  var s, o;
  t === void 0 && (t = !1), r === void 0 && (r = !1);
  const a = e.getBoundingClientRect(), l = fn(e);
  let i = gn;
  t && (n ? ce(n) && (i = $e(n)) : i = $e(e));
  const f = l ? q(l) : window, p = !cn() && r;
  let v = (a.left + (p && ((s = f.visualViewport) == null ? void 0 : s.offsetLeft) || 0)) / i.x, y = (a.top + (p && ((o = f.visualViewport) == null ? void 0 : o.offsetTop) || 0)) / i.y, h = a.width / i.x, b = a.height / i.y;
  if (l) {
    const C = q(l), _ = n && ce(n) ? q(n) : n;
    let k = C.frameElement;
    for (; k && n && _ !== C; ) {
      const T = $e(k), O = k.getBoundingClientRect(), L = getComputedStyle(k);
      O.x += (k.clientLeft + parseFloat(L.paddingLeft)) * T.x, O.y += (k.clientTop + parseFloat(L.paddingTop)) * T.y, v *= T.x, y *= T.y, h *= T.x, b *= T.y, v += O.x, y += O.y, k = q(k).frameElement;
    }
  }
  return {
    width: h,
    height: b,
    top: y,
    right: v + h,
    bottom: y + b,
    left: v,
    x: v,
    y
  };
}
function fe(e) {
  return ((dn(e) ? e.ownerDocument : e.document) || window.document).documentElement;
}
function mt(e) {
  return ce(e) ? {
    scrollLeft: e.scrollLeft,
    scrollTop: e.scrollTop
  } : {
    scrollLeft: e.pageXOffset,
    scrollTop: e.pageYOffset
  };
}
function Od(e) {
  let {
    rect: t,
    offsetParent: r,
    strategy: n
  } = e;
  const s = te(r), o = fe(r);
  if (r === o)
    return t;
  let a = {
    scrollLeft: 0,
    scrollTop: 0
  }, l = {
    x: 1,
    y: 1
  };
  const i = {
    x: 0,
    y: 0
  };
  if ((s || !s && n !== "fixed") && ((de(r) !== "body" || ht(o)) && (a = mt(r)), te(r))) {
    const f = je(r);
    l = $e(r), i.x = f.x + r.clientLeft, i.y = f.y + r.clientTop;
  }
  return {
    width: t.width * l.x,
    height: t.height * l.y,
    x: t.x * l.x - a.scrollLeft * l.x + i.x,
    y: t.y * l.y - a.scrollTop * l.y + i.y
  };
}
function pn(e) {
  return je(fe(e)).left + mt(e).scrollLeft;
}
function Ad(e) {
  const t = fe(e), r = mt(e), n = e.ownerDocument.body, s = Ne(t.scrollWidth, t.clientWidth, n.scrollWidth, n.clientWidth), o = Ne(t.scrollHeight, t.clientHeight, n.scrollHeight, n.clientHeight);
  let a = -r.scrollLeft + pn(e);
  const l = -r.scrollTop;
  return ee(n).direction === "rtl" && (a += Ne(t.clientWidth, n.clientWidth) - s), {
    width: s,
    height: o,
    x: a,
    y: l
  };
}
function We(e) {
  if (de(e) === "html")
    return e;
  const t = e.assignedSlot || e.parentNode || yr(e) && e.host || fe(e);
  return yr(t) ? t.host : t;
}
function hn(e) {
  const t = We(e);
  return Ft(t) ? t.ownerDocument.body : te(t) && ht(t) ? t : hn(t);
}
function it(e, t) {
  var r;
  t === void 0 && (t = []);
  const n = hn(e), s = n === ((r = e.ownerDocument) == null ? void 0 : r.body), o = q(n);
  return s ? t.concat(o, o.visualViewport || [], ht(n) ? n : []) : t.concat(n, it(n));
}
function Id(e, t) {
  const r = q(e), n = fe(e), s = r.visualViewport;
  let o = n.clientWidth, a = n.clientHeight, l = 0, i = 0;
  if (s) {
    o = s.width, a = s.height;
    const f = cn();
    (f || !f && t === "fixed") && (l = s.offsetLeft, i = s.offsetTop);
  }
  return {
    width: o,
    height: a,
    x: l,
    y: i
  };
}
function Bd(e, t) {
  const r = je(e, !0, t === "fixed"), n = r.top + e.clientTop, s = r.left + e.clientLeft, o = te(e) ? $e(e) : {
    x: 1,
    y: 1
  }, a = e.clientWidth * o.x, l = e.clientHeight * o.y, i = s * o.x, f = n * o.y;
  return {
    width: a,
    height: l,
    x: i,
    y: f
  };
}
function wr(e, t, r) {
  return t === "viewport" ? Me(Id(e, r)) : ce(t) ? Me(Bd(t, r)) : Me(Ad(fe(e)));
}
function Ed(e, t) {
  const r = t.get(e);
  if (r)
    return r;
  let n = it(e).filter((l) => ce(l) && de(l) !== "body"), s = null;
  const o = ee(e).position === "fixed";
  let a = o ? We(e) : e;
  for (; ce(a) && !Ft(a); ) {
    const l = ee(a), i = Dt(a);
    (o ? !i && !s : !i && l.position === "static" && !!s && ["absolute", "fixed"].includes(s.position)) ? n = n.filter((p) => p !== a) : s = l, a = We(a);
  }
  return t.set(e, n), n;
}
function Ld(e) {
  let {
    element: t,
    boundary: r,
    rootBoundary: n,
    strategy: s
  } = e;
  const a = [...r === "clippingAncestors" ? Ed(t, this._c) : [].concat(r), n], l = a[0], i = a.reduce((f, p) => {
    const v = wr(t, p, s);
    return f.top = Ne(v.top, f.top), f.right = vr(v.right, f.right), f.bottom = vr(v.bottom, f.bottom), f.left = Ne(v.left, f.left), f;
  }, wr(t, l, s));
  return {
    width: i.right - i.left,
    height: i.bottom - i.top,
    x: i.left,
    y: i.top
  };
}
function Md(e) {
  return te(e) ? ln(e) : e.getBoundingClientRect();
}
function xr(e) {
  return !te(e) || ee(e).position === "fixed" ? null : e.offsetParent;
}
function Nd(e) {
  let t = We(e);
  for (; te(t) && !Ft(t); ) {
    if (Dt(t))
      return t;
    t = We(t);
  }
  return null;
}
function kr(e) {
  const t = q(e);
  let r = xr(e);
  for (; r && zd(r) && ee(r).position === "static"; )
    r = xr(r);
  return r && (de(r) === "html" || de(r) === "body" && ee(r).position === "static" && !Dt(r)) ? t : r || Nd(e) || t;
}
function Vd(e, t, r) {
  const n = te(t), s = fe(t), o = je(e, !0, r === "fixed", t);
  let a = {
    scrollLeft: 0,
    scrollTop: 0
  };
  const l = {
    x: 0,
    y: 0
  };
  if (n || !n && r !== "fixed")
    if ((de(t) !== "body" || ht(s)) && (a = mt(t)), te(t)) {
      const i = je(t, !0);
      l.x = i.x + t.clientLeft, l.y = i.y + t.clientTop;
    } else
      s && (l.x = pn(s));
  return {
    x: o.left + a.scrollLeft - l.x,
    y: o.top + a.scrollTop - l.y,
    width: o.width,
    height: o.height
  };
}
const Rd = {
  getClippingRect: Ld,
  convertOffsetParentRelativeRectToViewportRelativeRect: Od,
  isElement: ce,
  getDimensions: Md,
  getOffsetParent: kr,
  getDocumentElement: fe,
  getScale: $e,
  async getElementRects(e) {
    let {
      reference: t,
      floating: r,
      strategy: n
    } = e;
    const s = this.getOffsetParent || kr, o = this.getDimensions;
    return {
      reference: Vd(t, await s(r), n),
      floating: {
        x: 0,
        y: 0,
        ...await o(r)
      }
    };
  },
  getClientRects: (e) => Array.from(e.getClientRects()),
  isRTL: (e) => ee(e).direction === "rtl"
}, Hd = (e, t, r) => {
  const n = /* @__PURE__ */ new Map(), s = {
    platform: Rd,
    ...r
  }, o = {
    ...s.platform,
    _c: n
  };
  return fd(e, t, {
    ...s,
    platform: o
  });
}, me = {
  disabled: !1,
  distance: 5,
  skidding: 0,
  container: "body",
  boundary: void 0,
  instantMove: !1,
  disposeTimeout: 5e3,
  popperTriggers: [],
  strategy: "absolute",
  preventOverflow: !0,
  flip: !0,
  shift: !0,
  overflowPadding: 0,
  arrowPadding: 0,
  arrowOverflow: !0,
  themes: {
    tooltip: {
      placement: "top",
      triggers: ["hover", "focus", "touch"],
      hideTriggers: (e) => [...e, "click"],
      delay: {
        show: 200,
        hide: 0
      },
      handleResize: !1,
      html: !1,
      loadingContent: "..."
    },
    dropdown: {
      placement: "bottom",
      triggers: ["click"],
      delay: 0,
      handleResize: !0,
      autoHide: !0
    },
    menu: {
      $extend: "dropdown",
      triggers: ["hover", "focus"],
      popperTriggers: ["hover", "focus"],
      delay: {
        show: 0,
        hide: 400
      }
    }
  }
};
function Tt(e, t) {
  let r = me.themes[e] || {}, n;
  do
    n = r[t], typeof n > "u" ? r.$extend ? r = me.themes[r.$extend] || {} : (r = null, n = me[t]) : r = null;
  while (r);
  return n;
}
function Dd(e) {
  const t = [e];
  let r = me.themes[e] || {};
  do
    r.$extend && !r.$resetCss ? (t.push(r.$extend), r = me.themes[r.$extend] || {}) : r = null;
  while (r);
  return t.map((n) => `v-popper--theme-${n}`);
}
function _r(e) {
  const t = [e];
  let r = me.themes[e] || {};
  do
    r.$extend ? (t.push(r.$extend), r = me.themes[r.$extend] || {}) : r = null;
  while (r);
  return t;
}
let Se = !1;
if (typeof window < "u") {
  Se = !1;
  try {
    const e = Object.defineProperty({}, "passive", {
      get() {
        Se = !0;
      }
    });
    window.addEventListener("test", null, e);
  } catch {
  }
}
let mn = !1;
typeof window < "u" && typeof navigator < "u" && (mn = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream);
const Fd = ["auto", "top", "bottom", "left", "right"].reduce((e, t) => e.concat([
  t,
  `${t}-start`,
  `${t}-end`
]), []), Cr = {
  hover: "mouseenter",
  focus: "focus",
  click: "click",
  touch: "touchstart",
  pointer: "pointerdown"
}, $r = {
  hover: "mouseleave",
  focus: "blur",
  click: "click",
  touch: "touchend",
  pointer: "pointerup"
};
function Sr(e, t) {
  const r = e.indexOf(t);
  r !== -1 && e.splice(r, 1);
}
function kt() {
  return new Promise((e) => requestAnimationFrame(() => {
    requestAnimationFrame(e);
  }));
}
const J = [];
let pe = null;
const Tr = {};
function Pr(e) {
  let t = Tr[e];
  return t || (t = Tr[e] = []), t;
}
let Pt = function() {
};
typeof window < "u" && (Pt = window.Element);
function z(e) {
  return function(t) {
    return Tt(t.theme, e);
  };
}
const _t = "__floating-vue__popper", bn = () => $({
  name: "VPopper",
  provide() {
    return {
      [_t]: {
        parentPopper: this
      }
    };
  },
  inject: {
    [_t]: { default: null }
  },
  props: {
    theme: {
      type: String,
      required: !0
    },
    targetNodes: {
      type: Function,
      required: !0
    },
    referenceNode: {
      type: Function,
      default: null
    },
    popperNode: {
      type: Function,
      required: !0
    },
    shown: {
      type: Boolean,
      default: !1
    },
    showGroup: {
      type: String,
      default: null
    },
    ariaId: {
      default: null
    },
    disabled: {
      type: Boolean,
      default: z("disabled")
    },
    positioningDisabled: {
      type: Boolean,
      default: z("positioningDisabled")
    },
    placement: {
      type: String,
      default: z("placement"),
      validator: (e) => Fd.includes(e)
    },
    delay: {
      type: [String, Number, Object],
      default: z("delay")
    },
    distance: {
      type: [Number, String],
      default: z("distance")
    },
    skidding: {
      type: [Number, String],
      default: z("skidding")
    },
    triggers: {
      type: Array,
      default: z("triggers")
    },
    showTriggers: {
      type: [Array, Function],
      default: z("showTriggers")
    },
    hideTriggers: {
      type: [Array, Function],
      default: z("hideTriggers")
    },
    popperTriggers: {
      type: Array,
      default: z("popperTriggers")
    },
    popperShowTriggers: {
      type: [Array, Function],
      default: z("popperShowTriggers")
    },
    popperHideTriggers: {
      type: [Array, Function],
      default: z("popperHideTriggers")
    },
    container: {
      type: [String, Object, Pt, Boolean],
      default: z("container")
    },
    boundary: {
      type: [String, Pt],
      default: z("boundary")
    },
    strategy: {
      type: String,
      validator: (e) => ["absolute", "fixed"].includes(e),
      default: z("strategy")
    },
    autoHide: {
      type: [Boolean, Function],
      default: z("autoHide")
    },
    handleResize: {
      type: Boolean,
      default: z("handleResize")
    },
    instantMove: {
      type: Boolean,
      default: z("instantMove")
    },
    eagerMount: {
      type: Boolean,
      default: z("eagerMount")
    },
    popperClass: {
      type: [String, Array, Object],
      default: z("popperClass")
    },
    computeTransformOrigin: {
      type: Boolean,
      default: z("computeTransformOrigin")
    },
    autoMinSize: {
      type: Boolean,
      default: z("autoMinSize")
    },
    autoSize: {
      type: [Boolean, String],
      default: z("autoSize")
    },
    autoMaxSize: {
      type: Boolean,
      default: z("autoMaxSize")
    },
    autoBoundaryMaxSize: {
      type: Boolean,
      default: z("autoBoundaryMaxSize")
    },
    preventOverflow: {
      type: Boolean,
      default: z("preventOverflow")
    },
    overflowPadding: {
      type: [Number, String],
      default: z("overflowPadding")
    },
    arrowPadding: {
      type: [Number, String],
      default: z("arrowPadding")
    },
    arrowOverflow: {
      type: Boolean,
      default: z("arrowOverflow")
    },
    flip: {
      type: Boolean,
      default: z("flip")
    },
    shift: {
      type: Boolean,
      default: z("shift")
    },
    shiftCrossAxis: {
      type: Boolean,
      default: z("shiftCrossAxis")
    },
    noAutoFocus: {
      type: Boolean,
      default: z("noAutoFocus")
    },
    disposeTimeout: {
      type: Number,
      default: z("disposeTimeout")
    }
  },
  emits: [
    "show",
    "hide",
    "update:shown",
    "apply-show",
    "apply-hide",
    "close-group",
    "close-directive",
    "auto-hide",
    "resize",
    "dispose"
  ],
  data() {
    return {
      isShown: !1,
      isMounted: !1,
      skipTransition: !1,
      classes: {
        showFrom: !1,
        showTo: !1,
        hideFrom: !1,
        hideTo: !0
      },
      result: {
        x: 0,
        y: 0,
        placement: "",
        strategy: this.strategy,
        arrow: {
          x: 0,
          y: 0,
          centerOffset: 0
        },
        transformOrigin: null
      },
      shownChildren: /* @__PURE__ */ new Set(),
      lastAutoHide: !0
    };
  },
  computed: {
    popperId() {
      return this.ariaId != null ? this.ariaId : this.randomId;
    },
    shouldMountContent() {
      return this.eagerMount || this.isMounted;
    },
    slotData() {
      return {
        popperId: this.popperId,
        isShown: this.isShown,
        shouldMountContent: this.shouldMountContent,
        skipTransition: this.skipTransition,
        autoHide: typeof this.autoHide == "function" ? this.lastAutoHide : this.autoHide,
        show: this.show,
        hide: this.hide,
        handleResize: this.handleResize,
        onResize: this.onResize,
        classes: {
          ...this.classes,
          popperClass: this.popperClass
        },
        result: this.positioningDisabled ? null : this.result,
        attrs: this.$attrs
      };
    },
    parentPopper() {
      var e;
      return (e = this[_t]) == null ? void 0 : e.parentPopper;
    },
    hasPopperShowTriggerHover() {
      var e, t;
      return ((e = this.popperTriggers) == null ? void 0 : e.includes("hover")) || ((t = this.popperShowTriggers) == null ? void 0 : t.includes("hover"));
    }
  },
  watch: {
    shown: "$_autoShowHide",
    disabled(e) {
      e ? this.dispose() : this.init();
    },
    async container() {
      this.isShown && (this.$_ensureTeleport(), await this.$_computePosition());
    },
    ...[
      "triggers",
      "positioningDisabled"
    ].reduce((e, t) => (e[t] = "$_refreshListeners", e), {}),
    ...[
      "placement",
      "distance",
      "skidding",
      "boundary",
      "strategy",
      "overflowPadding",
      "arrowPadding",
      "preventOverflow",
      "shift",
      "shiftCrossAxis",
      "flip"
    ].reduce((e, t) => (e[t] = "$_computePosition", e), {})
  },
  created() {
    this.$_isDisposed = !0, this.randomId = `popper_${[Math.random(), Date.now()].map((e) => e.toString(36).substring(2, 10)).join("_")}`, this.autoMinSize && console.warn('[floating-vue] `autoMinSize` option is deprecated. Use `autoSize="min"` instead.'), this.autoMaxSize && console.warn("[floating-vue] `autoMaxSize` option is deprecated. Use `autoBoundaryMaxSize` instead.");
  },
  mounted() {
    this.init(), this.$_detachPopperNode();
  },
  activated() {
    this.$_autoShowHide();
  },
  deactivated() {
    this.hide();
  },
  beforeUnmount() {
    this.dispose();
  },
  methods: {
    show({ event: e = null, skipDelay: t = !1, force: r = !1 } = {}) {
      var n, s;
      (n = this.parentPopper) != null && n.lockedChild && this.parentPopper.lockedChild !== this || (this.$_pendingHide = !1, (r || !this.disabled) && (((s = this.parentPopper) == null ? void 0 : s.lockedChild) === this && (this.parentPopper.lockedChild = null), this.$_scheduleShow(e, t), this.$emit("show"), this.$_showFrameLocked = !0, requestAnimationFrame(() => {
        this.$_showFrameLocked = !1;
      })), this.$emit("update:shown", !0));
    },
    hide({ event: e = null, skipDelay: t = !1 } = {}) {
      var r;
      if (!this.$_hideInProgress) {
        if (this.shownChildren.size > 0) {
          this.$_pendingHide = !0;
          return;
        }
        if (this.hasPopperShowTriggerHover && this.$_isAimingPopper()) {
          this.parentPopper && (this.parentPopper.lockedChild = this, clearTimeout(this.parentPopper.lockedChildTimer), this.parentPopper.lockedChildTimer = setTimeout(() => {
            this.parentPopper.lockedChild === this && (this.parentPopper.lockedChild.hide({ skipDelay: t }), this.parentPopper.lockedChild = null);
          }, 1e3));
          return;
        }
        ((r = this.parentPopper) == null ? void 0 : r.lockedChild) === this && (this.parentPopper.lockedChild = null), this.$_pendingHide = !1, this.$_scheduleHide(e, t), this.$emit("hide"), this.$emit("update:shown", !1);
      }
    },
    init() {
      var t;
      var e;
      this.$_isDisposed && (this.$_isDisposed = !1, this.isMounted = !1, this.$_events = [], this.$_preventShow = !1, this.$_referenceNode = (t = (e = this.referenceNode) == null ? void 0 : e.call(this)) != null ? t : this.$el, this.$_targetNodes = this.targetNodes().filter((r) => r.nodeType === r.ELEMENT_NODE), this.$_popperNode = this.popperNode(), this.$_innerNode = this.$_popperNode.querySelector(".v-popper__inner"), this.$_arrowNode = this.$_popperNode.querySelector(".v-popper__arrow-container"), this.$_swapTargetAttrs("title", "data-original-title"), this.$_detachPopperNode(), this.triggers.length && this.$_addEventListeners(), this.shown && this.show());
    },
    dispose() {
      this.$_isDisposed || (this.$_isDisposed = !0, this.$_removeEventListeners(), this.hide({ skipDelay: !0 }), this.$_detachPopperNode(), this.isMounted = !1, this.isShown = !1, this.$_updateParentShownChildren(!1), this.$_swapTargetAttrs("data-original-title", "title"), this.$emit("dispose"));
    },
    async onResize() {
      this.isShown && (await this.$_computePosition(), this.$emit("resize"));
    },
    async $_computePosition() {
      var n;
      if (this.$_isDisposed || this.positioningDisabled)
        return;
      const e = {
        strategy: this.strategy,
        middleware: []
      };
      (this.distance || this.skidding) && e.middleware.push($d({
        mainAxis: this.distance,
        crossAxis: this.skidding
      }));
      const t = this.placement.startsWith("auto");
      if (t ? e.middleware.push(yd({
        alignment: (n = this.placement.split("-")[1]) != null ? n : ""
      })) : e.placement = this.placement, this.preventOverflow && (this.shift && e.middleware.push(Td({
        padding: this.overflowPadding,
        boundary: this.boundary,
        crossAxis: this.shiftCrossAxis
      })), !t && this.flip && e.middleware.push(_d({
        padding: this.overflowPadding,
        boundary: this.boundary
      }))), e.middleware.push(pd({
        element: this.$_arrowNode,
        padding: this.arrowPadding
      })), this.arrowOverflow && e.middleware.push({
        name: "arrowOverflow",
        fn: ({ placement: s, rects: o, middlewareData: a }) => {
          let l;
          const { centerOffset: i } = a.arrow;
          return s.startsWith("top") || s.startsWith("bottom") ? l = Math.abs(i) > o.reference.width / 2 : l = Math.abs(i) > o.reference.height / 2, {
            data: {
              overflow: l
            }
          };
        }
      }), this.autoMinSize || this.autoSize) {
        const s = this.autoSize ? this.autoSize : this.autoMinSize ? "min" : null;
        e.middleware.push({
          name: "autoSize",
          fn: ({ rects: o, placement: a, middlewareData: l }) => {
            var i;
            if ((i = l.autoSize) != null && i.skip)
              return {};
            let f, p;
            return a.startsWith("top") || a.startsWith("bottom") ? f = o.reference.width : p = o.reference.height, this.$_innerNode.style[s === "min" ? "minWidth" : s === "max" ? "maxWidth" : "width"] = f != null ? `${f}px` : null, this.$_innerNode.style[s === "min" ? "minHeight" : s === "max" ? "maxHeight" : "height"] = p != null ? `${p}px` : null, {
              data: {
                skip: !0
              },
              reset: {
                rects: !0
              }
            };
          }
        });
      }
      (this.autoMaxSize || this.autoBoundaryMaxSize) && (this.$_innerNode.style.maxWidth = null, this.$_innerNode.style.maxHeight = null, e.middleware.push(Pd({
        boundary: this.boundary,
        padding: this.overflowPadding,
        apply: ({ availableWidth: s, availableHeight: o }) => {
          this.$_innerNode.style.maxWidth = s != null ? `${s}px` : null, this.$_innerNode.style.maxHeight = o != null ? `${o}px` : null;
        }
      })));
      const r = await Hd(this.$_referenceNode, this.$_popperNode, e);
      Object.assign(this.result, {
        x: r.x,
        y: r.y,
        placement: r.placement,
        strategy: r.strategy,
        arrow: {
          ...r.middlewareData.arrow,
          ...r.middlewareData.arrowOverflow
        }
      });
    },
    $_scheduleShow(e = null, t = !1) {
      if (this.$_updateParentShownChildren(!0), this.$_hideInProgress = !1, clearTimeout(this.$_scheduleTimer), pe && this.instantMove && pe.instantMove && pe !== this.parentPopper) {
        pe.$_applyHide(!0), this.$_applyShow(!0);
        return;
      }
      t ? this.$_applyShow() : this.$_scheduleTimer = setTimeout(this.$_applyShow.bind(this), this.$_computeDelay("show"));
    },
    $_scheduleHide(e = null, t = !1) {
      if (this.shownChildren.size > 0) {
        this.$_pendingHide = !0;
        return;
      }
      this.$_updateParentShownChildren(!1), this.$_hideInProgress = !0, clearTimeout(this.$_scheduleTimer), this.isShown && (pe = this), t ? this.$_applyHide() : this.$_scheduleTimer = setTimeout(this.$_applyHide.bind(this), this.$_computeDelay("hide"));
    },
    $_computeDelay(e) {
      const t = this.delay;
      return parseInt(t && t[e] || t || 0);
    },
    async $_applyShow(e = !1) {
      clearTimeout(this.$_disposeTimer), clearTimeout(this.$_scheduleTimer), this.skipTransition = e, !this.isShown && (this.$_ensureTeleport(), await kt(), await this.$_computePosition(), await this.$_applyShowEffect(), this.positioningDisabled || this.$_registerEventListeners([
        ...it(this.$_referenceNode),
        ...it(this.$_popperNode)
      ], "scroll", () => {
        this.$_computePosition();
      }));
    },
    async $_applyShowEffect() {
      if (this.$_hideInProgress)
        return;
      if (this.computeTransformOrigin) {
        const t = this.$_referenceNode.getBoundingClientRect(), r = this.$_popperNode.querySelector(".v-popper__wrapper"), n = r.parentNode.getBoundingClientRect(), s = t.x + t.width / 2 - (n.left + r.offsetLeft), o = t.y + t.height / 2 - (n.top + r.offsetTop);
        this.result.transformOrigin = `${s}px ${o}px`;
      }
      this.isShown = !0, this.$_applyAttrsToTarget({
        "aria-describedby": this.popperId,
        "data-popper-shown": ""
      });
      const e = this.showGroup;
      if (e) {
        let t;
        for (let r = 0; r < J.length; r++)
          t = J[r], t.showGroup !== e && (t.hide(), t.$emit("close-group"));
      }
      J.push(this), document.body.classList.add("v-popper--some-open");
      for (const t of _r(this.theme))
        Pr(t).push(this), document.body.classList.add(`v-popper--some-open--${t}`);
      this.$emit("apply-show"), this.classes.showFrom = !0, this.classes.showTo = !1, this.classes.hideFrom = !1, this.classes.hideTo = !1, await kt(), this.classes.showFrom = !1, this.classes.showTo = !0, this.noAutoFocus || this.$_popperNode.focus();
    },
    async $_applyHide(e = !1) {
      if (this.shownChildren.size > 0) {
        this.$_pendingHide = !0, this.$_hideInProgress = !1;
        return;
      }
      if (clearTimeout(this.$_scheduleTimer), !this.isShown)
        return;
      this.skipTransition = e, Sr(J, this), J.length === 0 && document.body.classList.remove("v-popper--some-open");
      for (const r of _r(this.theme)) {
        const n = Pr(r);
        Sr(n, this), n.length === 0 && document.body.classList.remove(`v-popper--some-open--${r}`);
      }
      pe === this && (pe = null), this.isShown = !1, this.$_applyAttrsToTarget({
        "aria-describedby": void 0,
        "data-popper-shown": void 0
      }), clearTimeout(this.$_disposeTimer);
      const t = this.disposeTimeout;
      t !== null && (this.$_disposeTimer = setTimeout(() => {
        this.$_popperNode && (this.$_detachPopperNode(), this.isMounted = !1);
      }, t)), this.$_removeEventListeners("scroll"), this.$emit("apply-hide"), this.classes.showFrom = !1, this.classes.showTo = !1, this.classes.hideFrom = !0, this.classes.hideTo = !1, await kt(), this.classes.hideFrom = !1, this.classes.hideTo = !0;
    },
    $_autoShowHide() {
      this.shown ? this.show() : this.hide();
    },
    $_ensureTeleport() {
      if (this.$_isDisposed)
        return;
      let e = this.container;
      if (typeof e == "string" ? e = window.document.querySelector(e) : e === !1 && (e = this.$_targetNodes[0].parentNode), !e)
        throw new Error("No container for popover: " + this.container);
      e.appendChild(this.$_popperNode), this.isMounted = !0;
    },
    $_addEventListeners() {
      const e = (r) => {
        this.isShown && !this.$_hideInProgress || (r.usedByTooltip = !0, !this.$_preventShow && this.show({ event: r }));
      };
      this.$_registerTriggerListeners(this.$_targetNodes, Cr, this.triggers, this.showTriggers, e), this.$_registerTriggerListeners([this.$_popperNode], Cr, this.popperTriggers, this.popperShowTriggers, e);
      const t = (r) => {
        r.usedByTooltip || this.hide({ event: r });
      };
      this.$_registerTriggerListeners(this.$_targetNodes, $r, this.triggers, this.hideTriggers, t), this.$_registerTriggerListeners([this.$_popperNode], $r, this.popperTriggers, this.popperHideTriggers, t);
    },
    $_registerEventListeners(e, t, r) {
      this.$_events.push({ targetNodes: e, eventType: t, handler: r }), e.forEach((n) => n.addEventListener(t, r, Se ? {
        passive: !0
      } : void 0));
    },
    $_registerTriggerListeners(e, t, r, n, s) {
      let o = r;
      n != null && (o = typeof n == "function" ? n(o) : n), o.forEach((a) => {
        const l = t[a];
        l && this.$_registerEventListeners(e, l, s);
      });
    },
    $_removeEventListeners(e) {
      const t = [];
      this.$_events.forEach((r) => {
        const { targetNodes: n, eventType: s, handler: o } = r;
        !e || e === s ? n.forEach((a) => a.removeEventListener(s, o)) : t.push(r);
      }), this.$_events = t;
    },
    $_refreshListeners() {
      this.$_isDisposed || (this.$_removeEventListeners(), this.$_addEventListeners());
    },
    $_handleGlobalClose(e, t = !1) {
      this.$_showFrameLocked || (this.hide({ event: e }), e.closePopover ? this.$emit("close-directive") : this.$emit("auto-hide"), t && (this.$_preventShow = !0, setTimeout(() => {
        this.$_preventShow = !1;
      }, 300)));
    },
    $_detachPopperNode() {
      this.$_popperNode.parentNode && this.$_popperNode.parentNode.removeChild(this.$_popperNode);
    },
    $_swapTargetAttrs(e, t) {
      for (const r of this.$_targetNodes) {
        const n = r.getAttribute(e);
        n && (r.removeAttribute(e), r.setAttribute(t, n));
      }
    },
    $_applyAttrsToTarget(e) {
      for (const t of this.$_targetNodes)
        for (const r in e) {
          const n = e[r];
          n == null ? t.removeAttribute(r) : t.setAttribute(r, n);
        }
    },
    $_updateParentShownChildren(e) {
      let t = this.parentPopper;
      for (; t; )
        e ? t.shownChildren.add(this.randomId) : (t.shownChildren.delete(this.randomId), t.$_pendingHide && t.hide()), t = t.parentPopper;
    },
    $_isAimingPopper() {
      const e = this.$_referenceNode.getBoundingClientRect();
      if (Ve >= e.left && Ve <= e.right && Re >= e.top && Re <= e.bottom) {
        const t = this.$_popperNode.getBoundingClientRect(), r = Ve - le, n = Re - ie, s = t.left + t.width / 2 - le + (t.top + t.height / 2) - ie + t.width + t.height, o = le + r * s, a = ie + n * s;
        return Je(le, ie, o, a, t.left, t.top, t.left, t.bottom) || Je(le, ie, o, a, t.left, t.top, t.right, t.top) || Je(le, ie, o, a, t.right, t.top, t.right, t.bottom) || Je(le, ie, o, a, t.left, t.bottom, t.right, t.bottom);
      }
      return !1;
    }
  },
  render() {
    return this.$slots.default(this.slotData);
  }
});
typeof document < "u" && typeof window < "u" && (mn ? (document.addEventListener("touchstart", zr, Se ? {
  passive: !0,
  capture: !0
} : !0), document.addEventListener("touchend", Wd, Se ? {
  passive: !0,
  capture: !0
} : !0)) : (window.addEventListener("mousedown", zr, !0), window.addEventListener("click", jd, !0)), window.addEventListener("resize", Ud));
function zr(e) {
  for (let t = 0; t < J.length; t++) {
    const r = J[t];
    try {
      const n = r.popperNode();
      r.$_mouseDownContains = n.contains(e.target);
    } catch {
    }
  }
}
function jd(e) {
  vn(e);
}
function Wd(e) {
  vn(e, !0);
}
function vn(e, t = !1) {
  const r = {};
  for (let n = J.length - 1; n >= 0; n--) {
    const s = J[n];
    try {
      const o = s.$_containsGlobalTarget = Gd(s, e);
      s.$_pendingHide = !1, requestAnimationFrame(() => {
        if (s.$_pendingHide = !1, !r[s.randomId] && Or(s, o, e)) {
          if (s.$_handleGlobalClose(e, t), !e.closeAllPopover && e.closePopover && o) {
            let l = s.parentPopper;
            for (; l; )
              r[l.randomId] = !0, l = l.parentPopper;
            return;
          }
          let a = s.parentPopper;
          for (; a && Or(a, a.$_containsGlobalTarget, e); )
            a.$_handleGlobalClose(e, t), a = a.parentPopper;
        }
      });
    } catch {
    }
  }
}
function Gd(e, t) {
  const r = e.popperNode();
  return e.$_mouseDownContains || r.contains(t.target);
}
function Or(e, t, r) {
  return r.closeAllPopover || r.closePopover && t || qd(e, r) && !t;
}
function qd(e, t) {
  if (typeof e.autoHide == "function") {
    const r = e.autoHide(t);
    return e.lastAutoHide = r, r;
  }
  return e.autoHide;
}
function Ud(e) {
  for (let t = 0; t < J.length; t++)
    J[t].$_computePosition(e);
}
let le = 0, ie = 0, Ve = 0, Re = 0;
typeof window < "u" && window.addEventListener("mousemove", (e) => {
  le = Ve, ie = Re, Ve = e.clientX, Re = e.clientY;
}, Se ? {
  passive: !0
} : void 0);
function Je(e, t, r, n, s, o, a, l) {
  const i = ((a - s) * (t - o) - (l - o) * (e - s)) / ((l - o) * (r - e) - (a - s) * (n - t)), f = ((r - e) * (t - o) - (n - t) * (e - s)) / ((l - o) * (r - e) - (a - s) * (n - t));
  return i >= 0 && i <= 1 && f >= 0 && f <= 1;
}
const Kd = {
  extends: bn()
}, jt = (e, t) => {
  const r = e.__vccOpts || e;
  for (const [n, s] of t)
    r[n] = s;
  return r;
};
function Yd(e, t, r, n, s, o) {
  return c(), g("div", {
    ref: "reference",
    class: w(["v-popper", {
      "v-popper--shown": e.slotData.isShown
    }])
  }, [
    x(e.$slots, "default", Ar(Ir(e.slotData)))
  ], 2);
}
const Jd = /* @__PURE__ */ jt(Kd, [["render", Yd]]);
function Qd() {
  var e = window.navigator.userAgent, t = e.indexOf("MSIE ");
  if (t > 0)
    return parseInt(e.substring(t + 5, e.indexOf(".", t)), 10);
  var r = e.indexOf("Trident/");
  if (r > 0) {
    var n = e.indexOf("rv:");
    return parseInt(e.substring(n + 3, e.indexOf(".", n)), 10);
  }
  var s = e.indexOf("Edge/");
  return s > 0 ? parseInt(e.substring(s + 5, e.indexOf(".", s)), 10) : -1;
}
let Xe;
function zt() {
  zt.init || (zt.init = !0, Xe = Qd() !== -1);
}
var bt = {
  name: "ResizeObserver",
  props: {
    emitOnMount: {
      type: Boolean,
      default: !1
    },
    ignoreWidth: {
      type: Boolean,
      default: !1
    },
    ignoreHeight: {
      type: Boolean,
      default: !1
    }
  },
  emits: [
    "notify"
  ],
  mounted() {
    zt(), Ot(() => {
      this._w = this.$el.offsetWidth, this._h = this.$el.offsetHeight, this.emitOnMount && this.emitSize();
    });
    const e = document.createElement("object");
    this._resizeObject = e, e.setAttribute("aria-hidden", "true"), e.setAttribute("tabindex", -1), e.onload = this.addResizeHandlers, e.type = "text/html", Xe && this.$el.appendChild(e), e.data = "about:blank", Xe || this.$el.appendChild(e);
  },
  beforeUnmount() {
    this.removeResizeHandlers();
  },
  methods: {
    compareAndNotify() {
      (!this.ignoreWidth && this._w !== this.$el.offsetWidth || !this.ignoreHeight && this._h !== this.$el.offsetHeight) && (this._w = this.$el.offsetWidth, this._h = this.$el.offsetHeight, this.emitSize());
    },
    emitSize() {
      this.$emit("notify", {
        width: this._w,
        height: this._h
      });
    },
    addResizeHandlers() {
      this._resizeObject.contentDocument.defaultView.addEventListener("resize", this.compareAndNotify), this.compareAndNotify();
    },
    removeResizeHandlers() {
      this._resizeObject && this._resizeObject.onload && (!Xe && this._resizeObject.contentDocument && this._resizeObject.contentDocument.defaultView.removeEventListener("resize", this.compareAndNotify), this.$el.removeChild(this._resizeObject), this._resizeObject.onload = null, this._resizeObject = null);
    }
  }
};
const Xd = /* @__PURE__ */ En("data-v-b329ee4c");
Er("data-v-b329ee4c");
const Zd = {
  class: "resize-observer",
  tabindex: "-1"
};
Lr();
const ec = /* @__PURE__ */ Xd((e, t, r, n, s, o) => (c(), D("div", Zd)));
bt.render = ec;
bt.__scopeId = "data-v-b329ee4c";
bt.__file = "src/components/ResizeObserver.vue";
const yn = (e = "theme") => ({
  computed: {
    themeClass() {
      return Dd(this[e]);
    }
  }
}), tc = $({
  name: "VPopperContent",
  components: {
    ResizeObserver: bt
  },
  mixins: [
    yn()
  ],
  props: {
    popperId: String,
    theme: String,
    shown: Boolean,
    mounted: Boolean,
    skipTransition: Boolean,
    autoHide: Boolean,
    handleResize: Boolean,
    classes: Object,
    result: Object
  },
  emits: [
    "hide",
    "resize"
  ],
  methods: {
    toPx(e) {
      return e != null && !isNaN(e) ? `${e}px` : null;
    }
  }
}), rc = ["id", "aria-hidden", "tabindex", "data-popper-placement"], nc = {
  ref: "inner",
  class: "v-popper__inner"
}, sc = /* @__PURE__ */ m("div", { class: "v-popper__arrow-outer" }, null, -1), oc = /* @__PURE__ */ m("div", { class: "v-popper__arrow-inner" }, null, -1), ac = [
  sc,
  oc
];
function lc(e, t, r, n, s, o) {
  const a = et("ResizeObserver");
  return c(), g("div", {
    id: e.popperId,
    ref: "popover",
    class: w(["v-popper__popper", [
      e.themeClass,
      e.classes.popperClass,
      {
        "v-popper__popper--shown": e.shown,
        "v-popper__popper--hidden": !e.shown,
        "v-popper__popper--show-from": e.classes.showFrom,
        "v-popper__popper--show-to": e.classes.showTo,
        "v-popper__popper--hide-from": e.classes.hideFrom,
        "v-popper__popper--hide-to": e.classes.hideTo,
        "v-popper__popper--skip-transition": e.skipTransition,
        "v-popper__popper--arrow-overflow": e.result && e.result.arrow.overflow,
        "v-popper__popper--no-positioning": !e.result
      }
    ]]),
    style: Le(e.result ? {
      position: e.result.strategy,
      transform: `translate3d(${Math.round(e.result.x)}px,${Math.round(e.result.y)}px,0)`
    } : void 0),
    "aria-hidden": e.shown ? "false" : "true",
    tabindex: e.autoHide ? 0 : void 0,
    "data-popper-placement": e.result ? e.result.placement : void 0,
    onKeyup: t[2] || (t[2] = Nr((l) => e.autoHide && e.$emit("hide"), ["esc"]))
  }, [
    m("div", {
      class: "v-popper__backdrop",
      onClick: t[0] || (t[0] = (l) => e.autoHide && e.$emit("hide"))
    }),
    m("div", {
      class: "v-popper__wrapper",
      style: Le(e.result ? {
        transformOrigin: e.result.transformOrigin
      } : void 0)
    }, [
      m("div", nc, [
        e.mounted ? (c(), g(U, { key: 0 }, [
          m("div", null, [
            x(e.$slots, "default")
          ]),
          e.handleResize ? (c(), D(a, {
            key: 0,
            onNotify: t[1] || (t[1] = (l) => e.$emit("resize", l))
          })) : S("", !0)
        ], 64)) : S("", !0)
      ], 512),
      m("div", {
        ref: "arrow",
        class: "v-popper__arrow-container",
        style: Le(e.result ? {
          left: e.toPx(e.result.arrow.x),
          top: e.toPx(e.result.arrow.y)
        } : void 0)
      }, ac, 4)
    ], 4)
  ], 46, rc);
}
const wn = /* @__PURE__ */ jt(tc, [["render", lc]]), xn = {
  methods: {
    show(...e) {
      return this.$refs.popper.show(...e);
    },
    hide(...e) {
      return this.$refs.popper.hide(...e);
    },
    dispose(...e) {
      return this.$refs.popper.dispose(...e);
    },
    onResize(...e) {
      return this.$refs.popper.onResize(...e);
    }
  }
}, ic = $({
  name: "VPopperWrapper",
  components: {
    Popper: Jd,
    PopperContent: wn
  },
  mixins: [
    xn,
    yn("finalTheme")
  ],
  props: {
    theme: {
      type: String,
      default: null
    }
  },
  computed: {
    finalTheme() {
      var e;
      return (e = this.theme) != null ? e : this.$options.vPopperTheme;
    }
  },
  methods: {
    getTargetNodes() {
      return Array.from(this.$el.children).filter((e) => e !== this.$refs.popperContent.$el);
    }
  }
});
function uc(e, t, r, n, s, o) {
  const a = et("PopperContent"), l = et("Popper");
  return c(), D(l, {
    ref: "popper",
    theme: e.finalTheme,
    "target-nodes": e.getTargetNodes,
    "popper-node": () => e.$refs.popperContent.$el,
    class: w([
      e.themeClass
    ])
  }, {
    default: H(({
      popperId: i,
      isShown: f,
      shouldMountContent: p,
      skipTransition: v,
      autoHide: y,
      show: h,
      hide: b,
      handleResize: C,
      onResize: _,
      classes: k,
      result: T
    }) => [
      x(e.$slots, "default", {
        shown: f,
        show: h,
        hide: b
      }),
      Ce(a, {
        ref: "popperContent",
        "popper-id": i,
        theme: e.finalTheme,
        shown: f,
        mounted: p,
        "skip-transition": v,
        "auto-hide": y,
        "handle-resize": C,
        classes: k,
        result: T,
        onHide: b,
        onResize: _
      }, {
        default: H(() => [
          x(e.$slots, "popper", {
            shown: f,
            hide: b
          })
        ]),
        _: 2
      }, 1032, ["popper-id", "theme", "shown", "mounted", "skip-transition", "auto-hide", "handle-resize", "classes", "result", "onHide", "onResize"])
    ]),
    _: 3
  }, 8, ["theme", "target-nodes", "popper-node", "class"]);
}
const Wt = /* @__PURE__ */ jt(ic, [["render", uc]]);
({
  ...Wt
});
({
  ...Wt
});
const dc = {
  ...Wt,
  name: "VTooltip",
  vPopperTheme: "tooltip"
};
$({
  name: "VTooltipDirective",
  components: {
    Popper: bn(),
    PopperContent: wn
  },
  mixins: [
    xn
  ],
  inheritAttrs: !1,
  props: {
    theme: {
      type: String,
      default: "tooltip"
    },
    html: {
      type: Boolean,
      default: (e) => Tt(e.theme, "html")
    },
    content: {
      type: [String, Number, Function],
      default: null
    },
    loadingContent: {
      type: String,
      default: (e) => Tt(e.theme, "loadingContent")
    },
    targetNodes: {
      type: Function,
      required: !0
    }
  },
  data() {
    return {
      asyncContent: null
    };
  },
  computed: {
    isContentAsync() {
      return typeof this.content == "function";
    },
    loading() {
      return this.isContentAsync && this.asyncContent == null;
    },
    finalContent() {
      return this.isContentAsync ? this.loading ? this.loadingContent : this.asyncContent : this.content;
    }
  },
  watch: {
    content: {
      handler() {
        this.fetchContent(!0);
      },
      immediate: !0
    },
    async finalContent() {
      await this.$nextTick(), this.$refs.popper.onResize();
    }
  },
  created() {
    this.$_fetchId = 0;
  },
  methods: {
    fetchContent(e) {
      if (typeof this.content == "function" && this.$_isShown && (e || !this.$_loading && this.asyncContent == null)) {
        this.asyncContent = null, this.$_loading = !0;
        const t = ++this.$_fetchId, r = this.content(this);
        r.then ? r.then((n) => this.onResult(t, n)) : this.onResult(t, r);
      }
    },
    onResult(e, t) {
      e === this.$_fetchId && (this.$_loading = !1, this.asyncContent = t);
    },
    onShow() {
      this.$_isShown = !0, this.fetchContent();
    },
    onHide() {
      this.$_isShown = !1;
    }
  }
});
const cc = dc;
const fc = { class: "flex items-start" }, sf = /* @__PURE__ */ $({
  __name: "Tooltip",
  props: {
    placement: { default: "top" },
    theme: { default: "dark" },
    trigger: { default: "hover" }
  },
  setup(e) {
    const t = e, r = u(() => ({
      light: "tooltip-light",
      dark: "tooltip-dark"
    })[t.theme]);
    return (n, s) => (c(), g("div", fc, [
      Ce(d(cc), {
        placement: e.placement,
        triggers: [e.trigger],
        theme: d(r),
        "auto-hide": ""
      }, {
        popper: H(() => [
          x(n.$slots, "content")
        ]),
        default: H(() => [
          x(n.$slots, "trigger")
        ]),
        _: 3
      }, 8, ["placement", "triggers", "theme"])
    ]));
  }
});
const gc = "block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300", pc = "bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500", hc = "cursor-not-allowed bg-gray-100", mc = {
  lg: "p-4",
  md: "p-2.5 text-sm",
  sm: "p-2 text-sm"
};
function bc(e) {
  const t = u(() => j(pc, mc[e.size.value], e.disabled.value ? hc : "")), r = u(() => gc);
  return {
    inputClasses: t,
    labelClasses: r
  };
}
const vc = { class: "flex relative" }, yc = {
  key: 0,
  class: "w-10 flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none overflow-hidden"
}, wc = ["disabled", "type", "required"], xc = {
  key: 1,
  class: "absolute right-2.5 bottom-2.5"
}, kc = {
  key: 1,
  class: "mt-2 text-sm text-gray-500 dark:text-gray-400"
}, of = /* @__PURE__ */ $({
  __name: "Input",
  props: {
    label: { default: "" },
    disabled: { type: Boolean, default: !1 },
    type: { default: "text" },
    size: { default: "md" },
    required: { type: Boolean, default: !1 },
    modelValue: { default: "" }
  },
  setup(e) {
    const t = e, r = ho(t, "modelValue"), { inputClasses: n, labelClasses: s } = bc(N(t));
    return (o, a) => (c(), g("div", null, [
      e.label ? (c(), g("label", {
        key: 0,
        class: w(d(s))
      }, B(e.label), 3)) : S("", !0),
      m("div", vc, [
        o.$slots.prefix ? (c(), g("div", yc, [
          x(o.$slots, "prefix")
        ])) : S("", !0),
        ye(m("input", Vr(o.$attrs, {
          "onUpdate:modelValue": a[0] || (a[0] = (l) => we(r) ? r.value = l : null),
          disabled: e.disabled,
          type: e.type,
          required: e.required,
          class: [d(n), o.$slots.prefix ? "pl-10" : ""]
        }), null, 16, wc), [
          [Ln, d(r)]
        ]),
        o.$slots.suffix ? (c(), g("div", xc, [
          x(o.$slots, "suffix")
        ])) : S("", !0)
      ]),
      o.$slots.helper ? (c(), g("p", kc, [
        x(o.$slots, "helper")
      ])) : S("", !0)
    ]));
  }
}), _c = "block text-sm font-medium text-gray-900 dark:text-gray-300", Cc = "w-4 h-4 rounded bg-gray-100 border-gray-300 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-600 dark:border-gray-500";
function $c() {
  const e = u(() => j(Cc)), t = u(() => _c);
  return {
    checkboxClasses: e,
    labelClasses: t
  };
}
const Sc = { class: "flex gap-3 items-center justify-start" }, Tc = ["disabled"], af = /* @__PURE__ */ $({
  __name: "Checkbox",
  props: {
    modelValue: { type: Boolean, default: !1 },
    label: { default: "" },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = u({
      get() {
        return r.modelValue;
      },
      set(a) {
        t("update:modelValue", a);
      }
    }), { checkboxClasses: s, labelClasses: o } = $c();
    return (a, l) => (c(), g("label", Sc, [
      ye(m("input", {
        "onUpdate:modelValue": l[0] || (l[0] = (i) => we(n) ? n.value = i : null),
        type: "checkbox",
        disabled: e.disabled,
        class: w(d(s))
      }, null, 10, Tc), [
        [Rr, d(n)]
      ]),
      e.label ? (c(), g("span", {
        key: 0,
        class: w(d(o))
      }, B(e.label), 3)) : S("", !0),
      x(a.$slots, "default")
    ]));
  }
}), Pc = "block mb-2 text-sm font-medium text-gray-900 dark:text-gray-300", zc = "w-full text-gray-900 bg-gray-50 border-[1px] border-gray-300 rounded-lg focus:ring-primary-500 focus:border-primary-500 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-primary-500 dark:focus:border-primary-500", Oc = "cursor-not-allowed bg-gray-100", Ac = "bg-transparent dark:bg-transparent border-0 border-b-2 border-gray-200 appearance-none dark:border-gray-700 focus:outline-none focus:ring-0 focus:border-gray-200 peer rounded-none", Ic = {
  lg: "p-4",
  md: "p-2.5 text-sm",
  sm: "p-2 text-sm"
};
function Bc(e) {
  const t = u(() => j(zc, Ic[e.size.value], e.disabled.value ? Oc : "")), r = u(() => Ac), n = u(() => Pc);
  return {
    selectClasses: t,
    underlineClasses: r,
    labelClasses: n
  };
}
const Ec = ["disabled"], Lc = {
  disabled: "",
  selected: "",
  value: ""
}, Mc = ["value"], lf = /* @__PURE__ */ $({
  __name: "Select",
  props: {
    modelValue: { default: "" },
    label: { default: "" },
    options: { default: () => [] },
    placeholder: { default: "Please select one" },
    disabled: { type: Boolean, default: !1 },
    underline: { type: Boolean, default: !1 },
    size: { default: "md" }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = u({
      get() {
        return r.modelValue;
      },
      set(l) {
        t("update:modelValue", l);
      }
    }), { selectClasses: s, underlineClasses: o, labelClasses: a } = Bc(N(r));
    return (l, i) => (c(), g("div", null, [
      e.label ? (c(), g("label", {
        key: 0,
        class: w(d(a))
      }, B(e.label), 3)) : S("", !0),
      ye(m("select", {
        "onUpdate:modelValue": i[0] || (i[0] = (f) => we(n) ? n.value = f : null),
        disabled: e.disabled,
        class: w([d(s), e.underline ? d(o) : ""])
      }, [
        m("option", Lc, B(e.placeholder), 1),
        (c(!0), g(U, null, be(e.options, (f, p) => (c(), g("option", {
          value: f.value,
          key: p
        }, B(f.name), 9, Mc))), 128))
      ], 10, Ec), [
        [Mn, d(n)]
      ])
    ]));
  }
}), Nc = "w-fit relative inline-flex items-center cursor-pointer", Vc = 'bg-gray-200 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-blue-300 dark:peer-focus:ring-blue-800 rounded-full peer dark:bg-gray-700 peer-checked:after:translate-x-full peer-checked:after:border-white after:content-[""] after:absolute after:top-[2px] after:left-[2px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-blue-600', Rc = "ml-3 text-sm font-medium text-gray-900 dark:text-gray-300", Hc = {
  lg: "w-14 h-7 after:top-0.5 after:left-[4px] after:h-6 after:w-6",
  md: "w-11 h-6 after:top-[2px] after:left-[2px] after:h-5 after:w-5",
  sm: "w-9 h-5 after:top-[2px] after:left-[2px] after:h-4 after:w-4"
}, Dc = {
  red: "peer-focus:ring-red-300 dark:peer-focus:ring-red-800 peer-checked:bg-red-600",
  green: "peer-focus:ring-green-300 dark:peer-focus:ring-green-800 peer-checked:bg-green-600",
  purple: "peer-focus:ring-purple-300 dark:peer-focus:ring-purple-800 peer-checked:bg-purple-600",
  yellow: "peer-focus:ring-yellow-300 dark:peer-focus:ring-yellow-800 peer-checked:bg-yellow-400",
  teal: "peer-focus:ring-teal-300 dark:peer-focus:ring-teal-800 peer-checked:bg-teal-600",
  orange: "peer-focus:ring-orange-300 dark:peer-focus:ring-orange-800 peer-checked:bg-orange-500"
};
function Fc(e) {
  const t = u(() => Nc), r = u(() => Vc), n = u(() => Hc[e.size.value]), s = u(() => Dc[e.color.value]), o = u(() => Rc);
  return {
    labelClasses: t,
    toggleSize: n,
    toggleClasses: r,
    toggleColor: s,
    toggleBallClasses: o
  };
}
const jc = ["disabled"], uf = /* @__PURE__ */ $({
  __name: "Toggle",
  props: {
    modelValue: { type: Boolean, default: !1 },
    label: { default: "" },
    color: { default: "" },
    size: { default: "md" },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = u({
      get() {
        return r.modelValue;
      },
      set(f) {
        t("update:modelValue", f);
      }
    }), { labelClasses: s, toggleSize: o, toggleClasses: a, toggleColor: l, toggleBallClasses: i } = Fc(N(r));
    return (f, p) => (c(), g("label", {
      class: w(d(s))
    }, [
      ye(m("input", {
        disabled: e.disabled,
        "onUpdate:modelValue": p[0] || (p[0] = (v) => we(n) ? n.value = v : null),
        type: "checkbox",
        class: "sr-only peer"
      }, null, 8, jc), [
        [Rr, d(n)]
      ]),
      m("span", {
        class: w([d(a), d(o), d(l)])
      }, null, 2),
      m("span", {
        class: w(d(i))
      }, B(e.label), 3)
    ], 2));
  }
}), Wc = "w-full h-2 bg-gray-200 rounded-lg appearance-none cursor-pointer dark:bg-gray-700", Gc = "block mb-2 text-sm font-medium text-gray-900 dark:text-white", qc = {
  lg: "h-3 range-lg",
  md: "h-2 range-md",
  sm: "h-1 range-sm"
};
function Uc(e) {
  const t = u(() => j(Wc, qc[e.size.value])), r = u(() => Gc);
  return {
    rangeClasses: t,
    labelClasses: r
  };
}
const Kc = ["step", "min", "max", "disabled"], df = /* @__PURE__ */ $({
  __name: "Range",
  props: {
    modelValue: { default: "50" },
    label: { default: "Range slider" },
    disabled: { type: Boolean, default: !1 },
    min: { default: "0" },
    max: { default: "100" },
    size: { default: "md" },
    steps: { default: "1" }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = u({
      get() {
        return r.modelValue;
      },
      set(a) {
        t("update:modelValue", a);
      }
    }), { rangeClasses: s, labelClasses: o } = Uc(N(r));
    return (a, l) => (c(), g("div", null, [
      m("label", {
        for: "default-range",
        class: w(d(o))
      }, B(e.label), 3),
      ye(m("input", {
        step: e.steps,
        "onUpdate:modelValue": l[0] || (l[0] = (i) => we(n) ? n.value = i : null),
        min: e.min,
        max: e.max,
        disabled: e.disabled,
        type: "range",
        class: w(d(s))
      }, null, 10, Kc), [
        [Nn, d(n)]
      ])
    ]));
  }
}), Yc = "w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600", Jc = "m-2 mr-0 text-sm font-medium text-gray-900 dark:text-gray-300";
function Qc() {
  const e = u(() => Yc), t = u(() => Jc);
  return {
    radioClasses: e,
    labelClasses: t
  };
}
const Xc = { class: "flex w-[100%] items-center" }, Zc = ["disabled", "name", "value"], cf = /* @__PURE__ */ $({
  __name: "Radio",
  props: {
    modelValue: null,
    name: { default: "" },
    value: { default: "" },
    label: { default: "" },
    disabled: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = u({
      get() {
        return r.modelValue;
      },
      set(a) {
        t("update:modelValue", a);
      }
    }), { radioClasses: s, labelClasses: o } = Qc();
    return (a, l) => (c(), g("label", Xc, [
      ye(m("input", {
        type: "radio",
        "onUpdate:modelValue": l[0] || (l[0] = (i) => we(n) ? n.value = i : null),
        disabled: e.disabled,
        name: e.name,
        value: e.value,
        class: w(d(s))
      }, null, 10, Zc), [
        [Vn, d(n)]
      ]),
      m("span", {
        class: w(d(o))
      }, B(e.label), 3),
      x(a.$slots, "default")
    ]));
  }
}), e0 = "block w-full text-sm text-gray-900 border-[1px] border-gray-300 rounded-lg cursor-pointer bg-gray-50 dark:text-gray-400 focus:outline-none dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400", t0 = "block mb-2 text-sm font-medium text-gray-900 dark:text-white", r0 = "flex flex-col items-center justify-center w-full h-64 border-2 border-gray-300 border-dashed rounded-lg cursor-pointer bg-gray-50 dark:hover:bg-bray-800 dark:bg-gray-700 hover:bg-gray-100 dark:border-gray-600 dark:hover:border-gray-500 dark:hover:bg-gray-600", n0 = "flex flex-col items-center justify-center pt-5 pb-6", s0 = "!-mb-2 text-sm text-gray-500 dark:text-gray-400";
function o0(e) {
  const t = u(() => j(e0, "text-" + e)), r = u(() => t0), n = u(() => r0), s = u(() => n0), o = u(() => s0);
  return {
    fileInpClasses: t,
    labelClasses: r,
    dropzoneLabelClasses: n,
    dropzoneWrapClasses: s,
    dropzoneTextClasses: o
  };
}
const a0 = { key: 0 }, l0 = ["multiple"], i0 = /* @__PURE__ */ m("svg", {
  class: "w-8 h-8 text-gray-500 dark:text-gray-400",
  "aria-hidden": "true",
  xmlns: "http://www.w3.org/2000/svg",
  fill: "none",
  viewBox: "0 0 20 16"
}, [
  /* @__PURE__ */ m("path", {
    stroke: "currentColor",
    "stroke-linecap": "round",
    "stroke-linejoin": "round",
    "stroke-width": "2",
    d: "M13 13h3a3 3 0 0 0 0-6h-.025A5.56 5.56 0 0 0 16 6.5 5.5 5.5 0 0 0 5.207 5.021C5.137 5.017 5.071 5 5 5a4 4 0 0 0 0 8h2.167M10 15V6m0 0L8 8m2-2 2 2"
  })
], -1), u0 = { key: 0 }, d0 = /* @__PURE__ */ m("span", { class: "font-semibold" }, " Click to upload ", -1), c0 = { key: 1 }, f0 = /* @__PURE__ */ m("input", {
  type: "file",
  class: "hidden"
}, null, -1), ff = /* @__PURE__ */ $({
  __name: "FileInput",
  props: {
    modelValue: { default: null },
    label: { default: "" },
    size: { default: "sm" },
    dropzone: { type: Boolean, default: !1 },
    multiple: { type: Boolean, default: !1 }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t }) {
    const r = e, n = u(() => ze(r.modelValue) ? r.modelValue.map((h) => h.name).join(", ") : r.modelValue instanceof FileList ? Array.from(r.modelValue).map((h) => h.name).join(",") : r.modelValue instanceof File && r.modelValue.name || ""), s = u({
      get() {
        return r.modelValue;
      },
      set(h) {
        t("update:modelValue", h);
      }
    }), o = (h) => {
      var C, _, k;
      const b = h.target;
      r.multiple ? s.value = Array.from((C = b.files) != null ? C : []) : s.value = (k = (_ = b.files) == null ? void 0 : _[0]) != null ? k : null;
    }, a = (h) => {
      var C, _;
      h.preventDefault();
      const b = [];
      (C = h.dataTransfer) != null && C.items ? (Object.values(h.dataTransfer.items).forEach((k) => {
        k.kind === "file" && b.push(k.getAsFile());
      }), r.multiple ? s.value = b : s.value = b[0]) : (_ = h.dataTransfer) != null && _.files && Object.values(h.dataTransfer.files).forEach((k) => {
        s.value = k;
      });
    }, l = (h) => {
      h.preventDefault();
    }, { fileInpClasses: i, labelClasses: f, dropzoneLabelClasses: p, dropzoneWrapClasses: v, dropzoneTextClasses: y } = o0(r.size);
    return (h, b) => (c(), g("div", null, [
      e.dropzone ? (c(), g("div", {
        key: 1,
        onChange: o,
        onDrop: a,
        onDragover: l,
        class: "flex items-center justify-center"
      }, [
        m("label", {
          class: w(d(p))
        }, [
          m("div", {
            class: w(d(v))
          }, [
            i0,
            d(s) ? (c(), g("p", c0, "File: " + B(d(n)), 1)) : (c(), g("div", u0, [
              m("p", {
                class: w(d(y))
              }, [
                d0,
                ne(" or drag and drop ")
              ], 2),
              x(h.$slots, "default")
            ]))
          ], 2),
          f0
        ], 2)
      ], 32)) : (c(), g("div", a0, [
        m("label", null, [
          m("span", {
            class: w(d(f))
          }, B(e.label), 3),
          m("input", {
            class: w(d(i)),
            multiple: e.multiple,
            onChange: o,
            type: "file"
          }, null, 42, l0)
        ]),
        x(h.$slots, "default")
      ]))
    ]));
  }
});
function gf() {
  const e = W(sn, null);
  return e === null && console.warn("Cannot use useToast outside <toast-provider> component. Please wrap your component with <toast-provider>"), {
    add: (s) => e ? e == null ? void 0 : e.add(s) : "",
    remove: (s) => e ? e == null ? void 0 : e.remove(s) : !1,
    pop: () => e ? e == null ? void 0 : e.pop() : ""
  };
}
export {
  p0 as Accordion,
  b0 as AccordionContent,
  m0 as AccordionHeader,
  h0 as AccordionPanel,
  y0 as Alert,
  S0 as Avatar,
  z0 as Badge,
  C0 as Breadcrumb,
  $0 as BreadcrumbItem,
  gs as Button,
  v0 as ButtonGroup,
  A0 as Carousel,
  af as Checkbox,
  k0 as Dropdown,
  ff as FileInput,
  _0 as FlowbiteThemable,
  Wr as FlowbiteThemableChild,
  I0 as Footer,
  of as Input,
  B0 as ListGroup,
  E0 as ListGroupItem,
  L0 as Modal,
  M0 as Navbar,
  V0 as NavbarCollapse,
  R0 as NavbarLink,
  N0 as NavbarLogo,
  H0 as Pagination,
  D0 as Progress,
  cf as Radio,
  df as Range,
  F0 as Rating,
  lf as Select,
  j0 as Sidebar,
  ur as SlotListener,
  qe as Spinner,
  T0 as StackedAvatars,
  P0 as StackedAvatarsCounter,
  x0 as Tab,
  W0 as Table,
  q0 as TableBody,
  Y0 as TableCell,
  G0 as TableHead,
  U0 as TableHeadCell,
  K0 as TableRow,
  w0 as Tabs,
  O0 as TheCard,
  J0 as Timeline,
  rf as TimelineBody,
  Z0 as TimelineContent,
  Q0 as TimelineItem,
  X0 as TimelinePoint,
  ef as TimelineTime,
  tf as TimelineTitle,
  hr as Toast,
  nf as ToastProvider,
  uf as Toggle,
  sf as Tooltip,
  gf as useToast
};
